<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Corrmotif overlap</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CX5461_Project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Corrmotif overlap</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-03-04
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>CX5461_Project/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, youâ€™ll want to first commit
it to the Git repo. If youâ€™re still working on the analysis, you can
ignore this warning. When youâ€™re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity itâ€™s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250129code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250129)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250129code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250129)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttree701caaebc6b0ea903a62a571e29ba6cfb3b04fc9targetblank701caaea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sayanpaul01/CX5461_Project/tree/701caaebc6b0ea903a62a571e29ba6cfb3b04fc9" target="_blank">701caae</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttree701caaebc6b0ea903a62a571e29ba6cfb3b04fc9targetblank701caaea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sayanpaul01/CX5461_Project/tree/701caaebc6b0ea903a62a571e29ba6cfb3b04fc9" target="_blank">701caae</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/

Untracked files:
    Untracked:  data/GO_time/

Unstaged changes:
    Modified:   Enriched_BP_Terms.csv
    Modified:   Enriched_CC_Terms.csv
    Modified:   Enriched_MF_Terms.csv
    Modified:   analysis/Corrmotif_overlap.Rmd

</code></pre>
<p>
Note that any generated files, e.g.Â HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Corrmotif_overlap.Rmd</code>) and
HTML (<code>docs/Corrmotif_overlap.html</code>) files. If youâ€™ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/701caaebc6b0ea903a62a571e29ba6cfb3b04fc9/analysis/Corrmotif_overlap.Rmd" target="_blank">701caae</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-04
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/701caaebc6b0ea903a62a571e29ba6cfb3b04fc9/docs/Corrmotif_overlap.html" target="_blank">701caae</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-04
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/56e1527e2bafdcf8961406466d74b2890e9aa04f/analysis/Corrmotif_overlap.Rmd" target="_blank">56e1527</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-04
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/56e1527e2bafdcf8961406466d74b2890e9aa04f/docs/Corrmotif_overlap.html" target="_blank">56e1527</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-04
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/63ae929740c542c542320c42ac26e7d93dbe842e/analysis/Corrmotif_overlap.Rmd" target="_blank">63ae929</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-04
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/CX5461_Project/63ae929740c542c542320c42ac26e7d93dbe842e/docs/Corrmotif_overlap.html" target="_blank">63ae929</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-04
</td>
<td>
Commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div
id="overlap-of-go-functions-between-corrmotif-all-and-corrmotif-conc."
class="section level2">
<h2><strong>ðŸ“Œ Overlap of GO functions between Corrmotif all and
corrmotif Conc.</strong></h2>
<pre class="r"><code>library(UpSetR)
library(dplyr)
library(tools)
library(GO.db)

# Set the folder path
folder_path &lt;- &quot;data/all_GO&quot;

# Get a list of all CSV files in the folder
csv_files &lt;- list.files(folder_path, pattern = &quot;\\.csv$&quot;, full.names = TRUE)

# Loop through each file and assign it as a variable in the global environment
for (file in csv_files) {
  # Generate a valid R variable name from the file name (remove extension and replace spaces)
  file_name &lt;- tools::file_path_sans_ext(basename(file))
  file_name &lt;- gsub(&quot; &quot;, &quot;_&quot;, file_name)  # Replace spaces with underscores
  file_name &lt;- make.names(file_name)  # Ensure the name is valid in R
  
  # Assign the CSV file as a variable in the environment
  assign(file_name, read.csv(file, stringsAsFactors = FALSE))
}

# Define datasets (lists of Entrez Gene IDs)
sets &lt;- list(
  &quot;Non response all&quot; = prob_all_1$ID,
  &quot;CX_DOX shared late response all&quot; = prob_all_2$ID,
  &quot;Dox specific response all&quot; = prob_all_3$ID,
  &quot;Late high dose DOX specific response all&quot; = prob_all_4$ID,
  
  &quot;Non response (0.1)&quot; = prob_1_0.1$ID,
  &quot;DOX only mid-late (0.1)&quot; = prob_2_0.1$ID,
  &quot;CX_DOX mid-late (0.1)&quot; = prob_3_0.1$ID,
  
  &quot;Non response (0.5)&quot; = prob_1_0.5$ID,
  &quot;DOX only early-mid (0.5)&quot; = prob_2_0.5$ID,
  &quot;DOX only mid-late (0.5)&quot; = prob_3_0.5$ID,
  &quot;CX only mid-late (0.5)&quot; = prob_4_0.5$ID,
  &quot;CX_DOX mid-late (0.5)&quot; = prob_5_0.5$ID
)

# Create a binary matrix for UpSet plot
all_genes &lt;- unique(unlist(sets))  # Get all unique Entrez Gene IDs
binary_matrix &lt;- data.frame(Gene_ID = all_genes)  # Initialize DataFrame

# Convert gene lists into a presence/absence matrix (1 = present, 0 = absent)
for (set_name in names(sets)) {
  binary_matrix[[set_name]] &lt;- as.integer(all_genes %in% sets[[set_name]])
}

# Remove Gene_ID column as UpSetR only needs the binary matrix
binary_matrix &lt;- binary_matrix[, -1]

upset(binary_matrix,
      sets = names(sets),
      order.by = &quot;freq&quot;,
      sets.bar.color = &quot;#56B4E9&quot;,  # Blue bars for set sizes
      mainbar.y.label = &quot;Number of Shared Functions&quot;,
      sets.x.label = &quot;GO terms per set&quot;,
      text.scale = 1.2,
      nintersects = 30)</code></pre>
<p><img src="figure/Corrmotif_overlap.Rmd/0verlap-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-0verlap-1">
Past versions of 0verlap-1.png
</button>
</p>
<div id="fig-0verlap-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/CX5461_Project/blob/63ae929740c542c542320c42ac26e7d93dbe842e/docs/figure/Corrmotif_overlap.Rmd/0verlap-1.png" target="_blank">63ae929</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-03-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="identify-unique-go-terms-for-each-response-group"
class="section level2">
<h2><strong>ðŸ“Œ Identify Unique GO Terms for Each Response
Group</strong></h2>
<pre class="r"><code># Create a list to store unique GO terms per category
unique_go_terms &lt;- list()

# Loop through each set to find unique GO terms
for (set_name in names(sets)) {
  # Get the GO terms for the current set
  current_go_terms &lt;- sets[[set_name]]
  
  # Find GO terms that appear **only** in this set and not in others
  unique_terms &lt;- current_go_terms[!(current_go_terms %in% unlist(sets[names(sets) != set_name]))]
  
  # Store in the list if there are any unique terms
  if (length(unique_terms) &gt; 0) {
    unique_go_terms[[set_name]] &lt;- unique_terms
  }
}

# Display unique GO terms for each category
unique_go_terms</code></pre>
<pre><code>$`Non response all`
[1] &quot;GO:0071339&quot; &quot;GO:1990204&quot; &quot;GO:0101031&quot; &quot;GO:0070469&quot; &quot;GO:0070971&quot;
[6] &quot;GO:0044665&quot;

$`CX_DOX shared late response all`
[1] &quot;GO:0009132&quot; &quot;GO:0097421&quot; &quot;GO:0140719&quot; &quot;GO:0140299&quot; &quot;GO:0035064&quot;
[6] &quot;GO:0140034&quot; &quot;GO:1990498&quot; &quot;GO:0005682&quot; &quot;GO:0030532&quot;

$`Dox specific response all`
[1] &quot;GO:0045177&quot; &quot;GO:0016323&quot; &quot;GO:0009925&quot;

$`Non response (0.1)`
 [1] &quot;GO:0034470&quot; &quot;GO:0042254&quot; &quot;GO:0022613&quot; &quot;GO:0140053&quot; &quot;GO:0006413&quot;
 [6] &quot;GO:0006364&quot; &quot;GO:0032543&quot; &quot;GO:0048193&quot; &quot;GO:0033108&quot; &quot;GO:0016072&quot;
[11] &quot;GO:0008135&quot; &quot;GO:0090079&quot; &quot;GO:0030684&quot; &quot;GO:0005759&quot; &quot;GO:0098800&quot;
[16] &quot;GO:0098803&quot; &quot;GO:0010494&quot; &quot;GO:0000313&quot; &quot;GO:0005761&quot; &quot;GO:0035770&quot;

$`DOX only mid-late (0.1)`
 [1] &quot;GO:0009162&quot; &quot;GO:0009130&quot; &quot;GO:0006999&quot; &quot;GO:0006221&quot; &quot;GO:0007096&quot;
 [6] &quot;GO:0045859&quot; &quot;GO:0006978&quot; &quot;GO:0009129&quot; &quot;GO:0046785&quot; &quot;GO:0002562&quot;
[11] &quot;GO:0016444&quot; &quot;GO:0042772&quot; &quot;GO:0010458&quot; &quot;GO:0071900&quot; &quot;GO:0051292&quot;
[16] &quot;GO:0006289&quot; &quot;GO:0008584&quot; &quot;GO:0008406&quot; &quot;GO:0046546&quot; &quot;GO:0045739&quot;
[21] &quot;GO:0045137&quot; &quot;GO:0043549&quot; &quot;GO:0009124&quot; &quot;GO:0048144&quot; &quot;GO:0008301&quot;
[26] &quot;GO:0017056&quot; &quot;GO:0000803&quot; &quot;GO:0043240&quot;

$`CX_DOX mid-late (0.1)`
[1] &quot;GO:0005402&quot;

$`DOX only early-mid (0.5)`
[1] &quot;GO:0140297&quot; &quot;GO:0061629&quot; &quot;GO:0090575&quot; &quot;GO:0005667&quot; &quot;GO:0097550&quot;
[6] &quot;GO:0005669&quot;

$`DOX only mid-late (0.5)`
 [1] &quot;GO:0007186&quot; &quot;GO:0048738&quot; &quot;GO:0014706&quot; &quot;GO:0099084&quot; &quot;GO:0099173&quot;
 [6] &quot;GO:0045598&quot; &quot;GO:0046486&quot; &quot;GO:0003013&quot; &quot;GO:0045444&quot; &quot;GO:0046620&quot;
[11] &quot;GO:0016236&quot;

$`CX only mid-late (0.5)`
 [1] &quot;GO:0048599&quot; &quot;GO:0090305&quot; &quot;GO:0009994&quot; &quot;GO:2000243&quot; &quot;GO:0018105&quot;
 [6] &quot;GO:0018209&quot; &quot;GO:0035561&quot; &quot;GO:0090657&quot; &quot;GO:0048477&quot; &quot;GO:0071732&quot;
[11] &quot;GO:0006264&quot; &quot;GO:0000722&quot; &quot;GO:0071731&quot; &quot;GO:1902170&quot; &quot;GO:0004518&quot;
[16] &quot;GO:0019205&quot;

$`CX_DOX mid-late (0.5)`
 [1] &quot;GO:0048146&quot; &quot;GO:0030865&quot; &quot;GO:0051493&quot; &quot;GO:0051782&quot; &quot;GO:0090399&quot;
 [6] &quot;GO:2000279&quot; &quot;GO:0030010&quot; &quot;GO:1901875&quot; &quot;GO:0032147&quot; &quot;GO:0031398&quot;
[11] &quot;GO:0007163&quot; &quot;GO:0051972&quot; &quot;GO:0000235&quot; &quot;GO:0005818&quot; &quot;GO:0101019&quot;</code></pre>
</div>
<div id="map-go-ids-to-function-names-using-go.db"
class="section level2">
<h2><strong>ðŸ“Œ Map GO IDs to Function Names Using GO.db</strong></h2>
<pre class="r"><code>map_go_terms_local &lt;- function(go_ids) {
  go_names &lt;- unlist(mget(go_ids, GOTERM, ifnotfound = NA))  # Retrieve function names
  go_descriptions &lt;- sapply(go_names, function(x) if (!is.na(x)) Term(x) else NA)
  
  # Create a dataframe
  go_mapping &lt;- data.frame(GO_ID = go_ids, Function = go_descriptions, stringsAsFactors = FALSE)
  
  # Remove NAs (unrecognized GO IDs)
  go_mapping &lt;- go_mapping[!is.na(go_mapping$Function), ]
  
  return(go_mapping)
}
mapped_unique_go_terms &lt;- list()

for (set_name in names(unique_go_terms)) {
  if (length(unique_go_terms[[set_name]]) &gt; 0) {
    go_data &lt;- map_go_terms_local(unique_go_terms[[set_name]])
    mapped_unique_go_terms[[set_name]] &lt;- go_data
  }
}</code></pre>
<pre><code>Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;</code></pre>
<pre class="r"><code># Display mapped GO terms
mapped_unique_go_terms</code></pre>
<pre><code>$`Non response all`
                GO_ID                          Function
GO:0071339 GO:0071339                      MLL1 complex
GO:1990204 GO:1990204            oxidoreductase complex
GO:0101031 GO:0101031 protein folding chaperone complex
GO:0070469 GO:0070469                       respirasome
GO:0070971 GO:0070971   endoplasmic reticulum exit site
GO:0044665 GO:0044665                    MLL1/2 complex

$`CX_DOX shared late response all`
                GO_ID                                 Function
GO:0009132 GO:0009132 nucleoside diphosphate metabolic process
GO:0097421 GO:0097421                       liver regeneration
GO:0140719 GO:0140719   constitutive heterochromatin formation
GO:0140299 GO:0140299           small molecule sensor activity
GO:0035064 GO:0035064               methylated histone binding
GO:0140034 GO:0140034    methylation-dependent protein binding
GO:1990498 GO:1990498              mitotic spindle microtubule
GO:0005682 GO:0005682                                 U5 snRNP
GO:0030532 GO:0030532  small nuclear ribonucleoprotein complex

$`Dox specific response all`
                GO_ID                    Function
GO:0045177 GO:0045177         apical part of cell
GO:0016323 GO:0016323 basolateral plasma membrane
GO:0009925 GO:0009925       basal plasma membrane

$`Non response (0.1)`
                GO_ID                                             Function
GO:0034470 GO:0034470                                     ncRNA processing
GO:0042254 GO:0042254                                  ribosome biogenesis
GO:0022613 GO:0022613                 ribonucleoprotein complex biogenesis
GO:0140053 GO:0140053                        mitochondrial gene expression
GO:0006413 GO:0006413                             translational initiation
GO:0006364 GO:0006364                                      rRNA processing
GO:0032543 GO:0032543                            mitochondrial translation
GO:0048193 GO:0048193                              Golgi vesicle transport
GO:0033108 GO:0033108     mitochondrial respiratory chain complex assembly
GO:0016072 GO:0016072                               rRNA metabolic process
GO:0008135 GO:0008135             translation factor activity, RNA binding
GO:0090079 GO:0090079 translation regulator activity, nucleic acid binding
GO:0030684 GO:0030684                                          preribosome
GO:0005759 GO:0005759                                 mitochondrial matrix
GO:0098800 GO:0098800         inner mitochondrial membrane protein complex
GO:0098803 GO:0098803                            respiratory chain complex
GO:0010494 GO:0010494                           cytoplasmic stress granule
GO:0000313 GO:0000313                                  organellar ribosome
GO:0005761 GO:0005761                               mitochondrial ribosome
GO:0035770 GO:0035770                            ribonucleoprotein granule

$`DOX only mid-late (0.1)`
                GO_ID
GO:0009162 GO:0009162
GO:0009130 GO:0009130
GO:0006999 GO:0006999
GO:0006221 GO:0006221
GO:0007096 GO:0007096
GO:0045859 GO:0045859
GO:0006978 GO:0006978
GO:0009129 GO:0009129
GO:0046785 GO:0046785
GO:0002562 GO:0002562
GO:0016444 GO:0016444
GO:0042772 GO:0042772
GO:0010458 GO:0010458
GO:0071900 GO:0071900
GO:0051292 GO:0051292
GO:0006289 GO:0006289
GO:0008584 GO:0008584
GO:0008406 GO:0008406
GO:0046546 GO:0046546
GO:0045739 GO:0045739
GO:0045137 GO:0045137
GO:0043549 GO:0043549
GO:0009124 GO:0009124
GO:0048144 GO:0048144
GO:0008301 GO:0008301
GO:0017056 GO:0017056
GO:0000803 GO:0000803
GO:0043240 GO:0043240
                                                                                                                  Function
GO:0009162                                                             deoxyribonucleoside monophosphate metabolic process
GO:0009130                                                        pyrimidine nucleoside monophosphate biosynthetic process
GO:0006999                                                                                       nuclear pore organization
GO:0006221                                                                      pyrimidine nucleotide biosynthetic process
GO:0007096                                                                                 regulation of exit from mitosis
GO:0045859                                                                           regulation of protein kinase activity
GO:0006978 DNA damage response, signal transduction by p53 class mediator resulting in transcription of p21 class mediator
GO:0009129                                                           pyrimidine nucleoside monophosphate metabolic process
GO:0046785                                                                                      microtubule polymerization
GO:0002562                    somatic diversification of immune receptors via germline recombination within a single locus
GO:0016444                                                                                  somatic cell DNA recombination
GO:0042772                                             DNA damage response, signal transduction resulting in transcription
GO:0010458                                                                                               exit from mitosis
GO:0071900                                                          regulation of protein serine/threonine kinase activity
GO:0051292                                                                                   nuclear pore complex assembly
GO:0006289                                                                                      nucleotide-excision repair
GO:0008584                                                                                          male gonad development
GO:0008406                                                                                               gonad development
GO:0046546                                                              development of primary male sexual characteristics
GO:0045739                                                                               positive regulation of DNA repair
GO:0045137                                                                   development of primary sexual characteristics
GO:0043549                                                                                   regulation of kinase activity
GO:0009124                                                                   nucleoside monophosphate biosynthetic process
GO:0048144                                                                                        fibroblast proliferation
GO:0008301                                                                                            DNA binding, bending
GO:0017056                                                                          structural constituent of nuclear pore
GO:0000803                                                                                                  sex chromosome
GO:0043240                                                                                 Fanconi anaemia nuclear complex

$`CX_DOX mid-late (0.1)`
                GO_ID                                          Function
GO:0005402 GO:0005402 carbohydrate:monoatomic cation symporter activity

$`DOX only early-mid (0.5)`
                GO_ID
GO:0140297 GO:0140297
GO:0061629 GO:0061629
GO:0090575 GO:0090575
GO:0005667 GO:0005667
GO:0097550 GO:0097550
GO:0005669 GO:0005669
                                                                      Function
GO:0140297                            DNA-binding transcription factor binding
GO:0061629 RNA polymerase II-specific DNA-binding transcription factor binding
GO:0090575                   RNA polymerase II transcription regulator complex
GO:0005667                                     transcription regulator complex
GO:0097550                                 transcription preinitiation complex
GO:0005669                                  transcription factor TFIID complex

$`DOX only mid-late (0.5)`
                GO_ID                                     Function
GO:0007186 GO:0007186 G protein-coupled receptor signaling pathway
GO:0048738 GO:0048738            cardiac muscle tissue development
GO:0014706 GO:0014706           striated muscle tissue development
GO:0099084 GO:0099084     postsynaptic specialization organization
GO:0099173 GO:0099173                     postsynapse organization
GO:0045598 GO:0045598       regulation of fat cell differentiation
GO:0046486 GO:0046486               glycerolipid metabolic process
GO:0003013 GO:0003013                   circulatory system process
GO:0045444 GO:0045444                     fat cell differentiation
GO:0046620 GO:0046620                   regulation of organ growth
GO:0016236 GO:0016236                               macroautophagy

$`CX only mid-late (0.5)`
                GO_ID                                       Function
GO:0048599 GO:0048599                             oocyte development
GO:0090305 GO:0090305    nucleic acid phosphodiester bond hydrolysis
GO:0009994 GO:0009994                         oocyte differentiation
GO:2000243 GO:2000243    positive regulation of reproductive process
GO:0018105 GO:0018105                peptidyl-serine phosphorylation
GO:0018209 GO:0018209                   peptidyl-serine modification
GO:0035561 GO:0035561                regulation of chromatin binding
GO:0090657 GO:0090657                     telomeric loop disassembly
GO:0048477 GO:0048477                                      oogenesis
GO:0071732 GO:0071732              cellular response to nitric oxide
GO:0006264 GO:0006264                  mitochondrial DNA replication
GO:0000722 GO:0000722         telomere maintenance via recombination
GO:0071731 GO:0071731                       response to nitric oxide
GO:1902170 GO:1902170 cellular response to reactive nitrogen species
GO:0004518 GO:0004518                              nuclease activity
GO:0019205 GO:0019205 nucleobase-containing compound kinase activity

$`CX_DOX mid-late (0.5)`
                GO_ID
GO:0048146 GO:0048146
GO:0030865 GO:0030865
GO:0051493 GO:0051493
GO:0051782 GO:0051782
GO:0090399 GO:0090399
GO:2000279 GO:2000279
GO:0030010 GO:0030010
GO:1901875 GO:1901875
GO:0032147 GO:0032147
GO:0031398 GO:0031398
GO:0007163 GO:0007163
GO:0051972 GO:0051972
GO:0000235 GO:0000235
GO:0005818 GO:0005818
GO:0101019 GO:0101019
                                                                 Function
GO:0048146                positive regulation of fibroblast proliferation
GO:0030865                             cortical cytoskeleton organization
GO:0051493                        regulation of cytoskeleton organization
GO:0051782                           negative regulation of cell division
GO:0090399                                         replicative senescence
GO:2000279                negative regulation of DNA biosynthetic process
GO:0030010                                 establishment of cell polarity
GO:1901875 positive regulation of post-translational protein modification
GO:0032147                          activation of protein kinase activity
GO:0031398                  positive regulation of protein ubiquitination
GO:0007163                  establishment or maintenance of cell polarity
GO:0051972                              regulation of telomerase activity
GO:0000235                                             astral microtubule
GO:0005818                                                          aster
GO:0101019                              nucleolar exosome (RNase complex)</code></pre>
</div>
<div id="overlap-of-go-functions-between-degs-and-corrmotif-timepoints"
class="section level2">
<h2><strong>ðŸ“Œ Overlap of GO functions between DEGs and corrmotif
timepoints</strong></h2>
<pre class="r"><code>library(UpSetR)
library(dplyr)
library(tools)
library(GO.db)

# Set the folder path
folder_path &lt;- &quot;data/GO_time&quot;

# Get a list of all CSV files in the folder
csv_files &lt;- list.files(folder_path, pattern = &quot;\\.csv$&quot;, full.names = TRUE)

# Loop through each file and assign it as a variable in the global environment
for (file in csv_files) {
  # Generate a valid R variable name from the file name (remove extension and replace spaces)
  file_name &lt;- tools::file_path_sans_ext(basename(file))
  file_name &lt;- gsub(&quot; &quot;, &quot;_&quot;, file_name)  # Replace spaces with underscores
  file_name &lt;- make.names(file_name)  # Ensure the name is valid in R
  
  # Assign the CSV file as a variable in the environment
  assign(file_name, read.csv(file, stringsAsFactors = FALSE))
}

sets &lt;- list(
  &quot;CX_0.1_24&quot; = CX_0.1_24_GO$ID,
  &quot;CX_0.1_48&quot; = CX_0.1_48_GO$ID,
  &quot;CX_0.5_3&quot; = CX_0.5_3_GO$ID,
  &quot;CX_0.5_24&quot; = CX_0.5_24_GO$ID,
  &quot;CX_0.5_48&quot; = CX_0.5_48_GO$ID,
  &quot;DOX_0.1_3&quot; = DOX_0.1_3_GO$ID,
  &quot;DOX_0.1_24&quot; = DOX_0.1_24_GO$ID,
  &quot;DOX_0.1_48&quot; = DOX_0.1_48_GO$ID,
  &quot;DOX_0.5_3&quot; = DOX_0.5_3_GO$ID,
  &quot;DOX_0.5_24&quot; = DOX_0.5_24_GO$ID,
  &quot;DOX_0.5_48&quot; = DOX_0.5_48_GO$ID,
  &quot;CX_low_non_response&quot; = prob_CX_1_0.1_GO$ID,
  &quot;CX_low_mid_late_response&quot; = prob_CX_2_0.1_GO$ID,
  &quot;CX_high_non_response&quot; = prob_CX_1_0.5_GO$ID,
  &quot;CX_high_mid_late_response&quot; = prob_CX_2_0.5_GO$ID,
  &quot;DOX_low_non_response&quot; = prob_DOX_1_0.1_GO$ID,
  &quot;DOX_low_mid_late_response&quot; = prob_DOX_2_0.1_GO$ID,
  &quot;DOX_high_non_response&quot; = prob_DOX_1_0.5_GO$ID,
  &quot;DOX_high_response&quot; = prob_DOX_2_0.5_GO$ID
)

# Create a binary matrix for UpSet plot
all_genes &lt;- unique(unlist(sets))  # Get all unique Entrez Gene IDs
binary_matrix &lt;- data.frame(Gene_ID = all_genes)  # Initialize DataFrame

# Convert gene lists into a presence/absence matrix (1 = present, 0 = absent)
for (set_name in names(sets)) {
  binary_matrix[[set_name]] &lt;- as.integer(all_genes %in% sets[[set_name]])
}

# Remove Gene_ID column as UpSetR only needs the binary matrix
binary_matrix &lt;- binary_matrix[, -1]

upset(binary_matrix,
      sets = names(sets),
      order.by = &quot;freq&quot;,
      sets.bar.color = &quot;#56B4E9&quot;,  # Blue bars for set sizes
      mainbar.y.label = &quot;Number of Shared Functions&quot;,
      sets.x.label = &quot;GO terms per set&quot;,
      text.scale = 1.2,
      nintersects = 30)</code></pre>
<p><img src="figure/Corrmotif_overlap.Rmd/0verlap2-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="identify-unique-go-terms-for-each-response-group-1"
class="section level2">
<h2><strong>ðŸ“Œ Identify Unique GO Terms for Each Response
Group</strong></h2>
<pre class="r"><code># Create a list to store unique GO terms per category
unique_go_terms &lt;- list()

# Loop through each set to find unique GO terms
for (set_name in names(sets)) {
  # Get the GO terms for the current set
  current_go_terms &lt;- sets[[set_name]]
  
  # Find GO terms that appear **only** in this set and not in others
  unique_terms &lt;- current_go_terms[!(current_go_terms %in% unlist(sets[names(sets) != set_name]))]
  
  # Store in the list if there are any unique terms
  if (length(unique_terms) &gt; 0) {
    unique_go_terms[[set_name]] &lt;- unique_terms
  }
}

# Display unique GO terms for each category
unique_go_terms</code></pre>
<pre><code>$CX_0.1_24
[1] &quot;GO:0032147&quot; &quot;GO:0061351&quot; &quot;GO:0021987&quot; &quot;GO:0004712&quot; &quot;GO:0000235&quot;
[6] &quot;GO:0005818&quot; &quot;GO:0016363&quot;

$CX_0.1_48
[1] &quot;GO:1902115&quot; &quot;GO:0070979&quot; &quot;GO:0046653&quot; &quot;GO:0042558&quot; &quot;GO:0009163&quot;
[6] &quot;GO:0034404&quot;

$CX_0.5_3
[1] &quot;GO:0060218&quot; &quot;GO:0002244&quot; &quot;GO:0048863&quot; &quot;GO:0007517&quot;

$CX_0.5_24
[1] &quot;GO:0036064&quot;

$CX_0.5_48
 [1] &quot;GO:0006305&quot; &quot;GO:0006306&quot; &quot;GO:0009129&quot; &quot;GO:0006544&quot; &quot;GO:0140719&quot;
 [6] &quot;GO:2001032&quot; &quot;GO:0009116&quot; &quot;GO:0045911&quot; &quot;GO:0051444&quot; &quot;GO:0045739&quot;

$DOX_0.1_3
[1] &quot;GO:1900034&quot; &quot;GO:0034605&quot; &quot;GO:0009408&quot; &quot;GO:0009266&quot;

$DOX_0.1_24
 [1] &quot;GO:0070372&quot; &quot;GO:0001759&quot; &quot;GO:0070371&quot; &quot;GO:0032609&quot; &quot;GO:0032649&quot;
 [6] &quot;GO:0008585&quot; &quot;GO:0061458&quot; &quot;GO:0010712&quot; &quot;GO:0048608&quot; &quot;GO:0046545&quot;
[11] &quot;GO:0032729&quot; &quot;GO:0010818&quot; &quot;GO:0070374&quot; &quot;GO:0040008&quot; &quot;GO:0014856&quot;
[16] &quot;GO:0046660&quot; &quot;GO:0033002&quot; &quot;GO:0042327&quot; &quot;GO:0010714&quot; &quot;GO:0009123&quot;
[21] &quot;GO:0032965&quot; &quot;GO:0048771&quot; &quot;GO:0031128&quot; &quot;GO:0032733&quot; &quot;GO:0009156&quot;
[26] &quot;GO:0002250&quot; &quot;GO:0003156&quot; &quot;GO:0032967&quot; &quot;GO:0001934&quot; &quot;GO:0043410&quot;
[31] &quot;GO:0048247&quot; &quot;GO:0009410&quot; &quot;GO:0110110&quot; &quot;GO:0031952&quot; &quot;GO:0002460&quot;
[36] &quot;GO:0009161&quot; &quot;GO:0034103&quot; &quot;GO:0050869&quot; &quot;GO:0060485&quot; &quot;GO:0002706&quot;
[41] &quot;GO:0003014&quot; &quot;GO:1901652&quot; &quot;GO:0030316&quot; &quot;GO:0030856&quot; &quot;GO:0008217&quot;
[46] &quot;GO:2000242&quot; &quot;GO:0030546&quot; &quot;GO:0048018&quot; &quot;GO:0030545&quot; &quot;GO:0008083&quot;
[51] &quot;GO:0016471&quot; &quot;GO:0030672&quot; &quot;GO:0099501&quot; &quot;GO:0099569&quot;

$DOX_0.1_48
 [1] &quot;GO:1901136&quot; &quot;GO:0006516&quot; &quot;GO:0008045&quot; &quot;GO:0030903&quot; &quot;GO:0086010&quot;
 [6] &quot;GO:0001508&quot; &quot;GO:0043062&quot; &quot;GO:0097205&quot; &quot;GO:0051899&quot; &quot;GO:0030198&quot;
[11] &quot;GO:0061337&quot; &quot;GO:0045229&quot; &quot;GO:0003094&quot; &quot;GO:0006026&quot; &quot;GO:0044409&quot;
[16] &quot;GO:0046718&quot; &quot;GO:0006027&quot; &quot;GO:0099638&quot; &quot;GO:0086012&quot; &quot;GO:0001570&quot;
[21] &quot;GO:0086001&quot; &quot;GO:0086002&quot; &quot;GO:0016798&quot; &quot;GO:0004888&quot; &quot;GO:0004553&quot;
[26] &quot;GO:0042734&quot; &quot;GO:0043202&quot; &quot;GO:0005775&quot; &quot;GO:0045211&quot; &quot;GO:0030315&quot;
[31] &quot;GO:0005796&quot;

$DOX_0.5_3
 [1] &quot;GO:0060411&quot; &quot;GO:0003007&quot; &quot;GO:0060840&quot; &quot;GO:0035270&quot; &quot;GO:0030098&quot;
 [6] &quot;GO:0003279&quot; &quot;GO:0045165&quot; &quot;GO:0090092&quot; &quot;GO:0003281&quot; &quot;GO:0048844&quot;
[11] &quot;GO:0003151&quot; &quot;GO:1903131&quot; &quot;GO:0071559&quot; &quot;GO:0060419&quot; &quot;GO:0017015&quot;
[16] &quot;GO:1903844&quot; &quot;GO:0035019&quot; &quot;GO:0035904&quot; &quot;GO:0030217&quot; &quot;GO:0071560&quot;
[21] &quot;GO:0060412&quot; &quot;GO:0003170&quot; &quot;GO:0055017&quot; &quot;GO:0060976&quot; &quot;GO:0046649&quot;
[26] &quot;GO:0002521&quot; &quot;GO:1902105&quot; &quot;GO:0030278&quot; &quot;GO:1903706&quot; &quot;GO:0035051&quot;
[31] &quot;GO:0042110&quot; &quot;GO:0002065&quot; &quot;GO:0033689&quot; &quot;GO:0003181&quot; &quot;GO:0009896&quot;
[36] &quot;GO:0035909&quot; &quot;GO:0007179&quot; &quot;GO:0017148&quot; &quot;GO:0048568&quot; &quot;GO:0002067&quot;
[41] &quot;GO:0060537&quot; &quot;GO:0005978&quot; &quot;GO:0009250&quot; &quot;GO:0045785&quot; &quot;GO:0003171&quot;
[46] &quot;GO:0003712&quot; &quot;GO:0140297&quot; &quot;GO:0061733&quot; &quot;GO:0034212&quot; &quot;GO:0008080&quot;
[51] &quot;GO:0001217&quot; &quot;GO:0003713&quot; &quot;GO:0004402&quot; &quot;GO:0061629&quot; &quot;GO:0016407&quot;
[56] &quot;GO:0001227&quot; &quot;GO:0016410&quot; &quot;GO:0046332&quot; &quot;GO:0003714&quot; &quot;GO:0050321&quot;
[61] &quot;GO:0035497&quot; &quot;GO:0048156&quot; &quot;GO:0010484&quot; &quot;GO:0005667&quot; &quot;GO:0090575&quot;

$DOX_0.5_48
 [1] &quot;GO:0051716&quot; &quot;GO:0050896&quot; &quot;GO:0023052&quot; &quot;GO:0008150&quot; &quot;GO:0007154&quot;
 [6] &quot;GO:0050789&quot; &quot;GO:0009987&quot; &quot;GO:0065007&quot; &quot;GO:0019725&quot; &quot;GO:0032501&quot;
[11] &quot;GO:0006942&quot; &quot;GO:0072359&quot; &quot;GO:0003015&quot; &quot;GO:0050794&quot; &quot;GO:0060047&quot;
[16] &quot;GO:0051128&quot; &quot;GO:0070887&quot; &quot;GO:0051179&quot; &quot;GO:0030003&quot; &quot;GO:1903522&quot;
[21] &quot;GO:0007165&quot; &quot;GO:0048856&quot; &quot;GO:0042221&quot; &quot;GO:0032502&quot; &quot;GO:0006937&quot;
[26] &quot;GO:0016236&quot; &quot;GO:0005488&quot; &quot;GO:0003674&quot; &quot;GO:0005515&quot; &quot;GO:0043168&quot;
[31] &quot;GO:0097367&quot; &quot;GO:0005524&quot; &quot;GO:0030554&quot; &quot;GO:1901363&quot; &quot;GO:0035639&quot;
[36] &quot;GO:0032559&quot; &quot;GO:0000166&quot; &quot;GO:1901265&quot; &quot;GO:0017076&quot; &quot;GO:0032555&quot;
[41] &quot;GO:0032553&quot; &quot;GO:0003824&quot; &quot;GO:0016817&quot; &quot;GO:0016818&quot; &quot;GO:0016462&quot;
[46] &quot;GO:0005575&quot; &quot;GO:0110165&quot; &quot;GO:0071944&quot; &quot;GO:0005886&quot; &quot;GO:0005737&quot;
[51] &quot;GO:0016020&quot; &quot;GO:0005622&quot; &quot;GO:0043229&quot; &quot;GO:0099512&quot; &quot;GO:0099081&quot;
[56] &quot;GO:0000323&quot; &quot;GO:0005764&quot;

$CX_low_non_response
[1] &quot;GO:0038023&quot; &quot;GO:0060089&quot; &quot;GO:0005770&quot; &quot;GO:0098978&quot; &quot;GO:0030055&quot;

$CX_low_mid_late_response
[1] &quot;GO:0051493&quot; &quot;GO:0070301&quot; &quot;GO:0006304&quot; &quot;GO:0009223&quot; &quot;GO:0048308&quot;
[6] &quot;GO:0048313&quot; &quot;GO:0006213&quot; &quot;GO:0032404&quot;

$CX_high_non_response
 [1] &quot;GO:0061564&quot; &quot;GO:0006397&quot; &quot;GO:0007409&quot; &quot;GO:0060627&quot; &quot;GO:0090066&quot;
 [6] &quot;GO:0007186&quot; &quot;GO:0005509&quot; &quot;GO:0046873&quot; &quot;GO:1902495&quot; &quot;GO:0035770&quot;
[11] &quot;GO:0030424&quot; &quot;GO:0031252&quot; &quot;GO:0036464&quot;

$CX_high_mid_late_response
 [1] &quot;GO:0000722&quot; &quot;GO:0006220&quot; &quot;GO:0009132&quot; &quot;GO:0090657&quot; &quot;GO:0032042&quot;
 [6] &quot;GO:0006264&quot; &quot;GO:0009185&quot; &quot;GO:0004518&quot; &quot;GO:0140299&quot; &quot;GO:0016894&quot;
[11] &quot;GO:0016796&quot; &quot;GO:0004527&quot; &quot;GO:1990498&quot;

$DOX_low_non_response
 [1] &quot;GO:0034470&quot; &quot;GO:0022613&quot; &quot;GO:0042254&quot; &quot;GO:0006364&quot; &quot;GO:0140053&quot;
 [6] &quot;GO:0016072&quot; &quot;GO:0006413&quot; &quot;GO:0032543&quot; &quot;GO:0008135&quot; &quot;GO:0030684&quot;
[11] &quot;GO:0098800&quot; &quot;GO:0098803&quot; &quot;GO:0019866&quot;

$DOX_low_mid_late_response
 [1] &quot;GO:0009416&quot; &quot;GO:0050673&quot; &quot;GO:0010955&quot; &quot;GO:1903318&quot; &quot;GO:0002251&quot;
 [6] &quot;GO:0014029&quot; &quot;GO:0097237&quot; &quot;GO:0055021&quot; &quot;GO:0031645&quot; &quot;GO:0034350&quot;
[11] &quot;GO:0055023&quot; &quot;GO:0010804&quot; &quot;GO:0046686&quot; &quot;GO:0005402&quot; &quot;GO:0008296&quot;

$DOX_high_non_response
[1] &quot;GO:0002181&quot; &quot;GO:0022626&quot; &quot;GO:0022625&quot; &quot;GO:0042788&quot;

$DOX_high_response
[1] &quot;GO:0045598&quot; &quot;GO:0045444&quot; &quot;GO:0009084&quot;</code></pre>
</div>
<div id="map-go-ids-to-function-names-using-go.db-1"
class="section level2">
<h2><strong>ðŸ“Œ Map GO IDs to Function Names Using GO.db</strong></h2>
<pre class="r"><code>map_go_terms_local &lt;- function(go_ids) {
  go_names &lt;- unlist(mget(go_ids, GOTERM, ifnotfound = NA))  # Retrieve function names
  go_descriptions &lt;- sapply(go_names, function(x) if (!is.na(x)) Term(x) else NA)
  
  # Create a dataframe
  go_mapping &lt;- data.frame(GO_ID = go_ids, Function = go_descriptions, stringsAsFactors = FALSE)
  
  # Remove NAs (unrecognized GO IDs)
  go_mapping &lt;- go_mapping[!is.na(go_mapping$Function), ]
  
  return(go_mapping)
}
mapped_unique_go_terms &lt;- list()

for (set_name in names(unique_go_terms)) {
  if (length(unique_go_terms[[set_name]]) &gt; 0) {
    go_data &lt;- map_go_terms_local(unique_go_terms[[set_name]])
    mapped_unique_go_terms[[set_name]] &lt;- go_data
  }
}</code></pre>
<pre><code>Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;
Warning in is.na(x): is.na() applied to non-(list or vector) of type &#39;S4&#39;</code></pre>
<pre class="r"><code># Display mapped GO terms
mapped_unique_go_terms</code></pre>
<pre><code>$CX_0.1_24
                GO_ID                                          Function
GO:0032147 GO:0032147             activation of protein kinase activity
GO:0061351 GO:0061351               neural precursor cell proliferation
GO:0021987 GO:0021987                       cerebral cortex development
GO:0004712 GO:0004712 protein serine/threonine/tyrosine kinase activity
GO:0000235 GO:0000235                                astral microtubule
GO:0005818 GO:0005818                                             aster
GO:0016363 GO:0016363                                    nuclear matrix

$CX_0.1_48
                GO_ID                                                  Function
GO:1902115 GO:1902115                          regulation of organelle assembly
GO:0070979 GO:0070979                         protein K11-linked ubiquitination
GO:0046653 GO:0046653                        tetrahydrofolate metabolic process
GO:0042558 GO:0042558           pteridine-containing compound metabolic process
GO:0009163 GO:0009163                           nucleoside biosynthetic process
GO:0034404 GO:0034404 nucleobase-containing small molecule biosynthetic process

$CX_0.5_3
                GO_ID                                      Function
GO:0060218 GO:0060218       hematopoietic stem cell differentiation
GO:0002244 GO:0002244 hematopoietic progenitor cell differentiation
GO:0048863 GO:0048863                     stem cell differentiation
GO:0007517 GO:0007517                      muscle organ development

$CX_0.5_24
                GO_ID           Function
GO:0036064 GO:0036064 ciliary basal body

$CX_0.5_48
                GO_ID
GO:0006305 GO:0006305
GO:0006306 GO:0006306
GO:0009129 GO:0009129
GO:0006544 GO:0006544
GO:0140719 GO:0140719
GO:2001032 GO:2001032
GO:0009116 GO:0009116
GO:0045911 GO:0045911
GO:0051444 GO:0051444
GO:0045739 GO:0045739
                                                                         Function
GO:0006305                                                         DNA alkylation
GO:0006306                                                        DNA methylation
GO:0009129                  pyrimidine nucleoside monophosphate metabolic process
GO:0006544                                              glycine metabolic process
GO:0140719                                 constitutive heterochromatin formation
GO:2001032 regulation of double-strand break repair via nonhomologous end joining
GO:0009116                                           nucleoside metabolic process
GO:0045911                               positive regulation of DNA recombination
GO:0051444          negative regulation of ubiquitin-protein transferase activity
GO:0045739                                      positive regulation of DNA repair

$DOX_0.1_3
                GO_ID                                Function
GO:1900034 GO:1900034 regulation of cellular response to heat
GO:0034605 GO:0034605               cellular response to heat
GO:0009408 GO:0009408                        response to heat
GO:0009266 GO:0009266        response to temperature stimulus

$DOX_0.1_24
                GO_ID
GO:0070372 GO:0070372
GO:0001759 GO:0001759
GO:0070371 GO:0070371
GO:0032609 GO:0032609
GO:0032649 GO:0032649
GO:0008585 GO:0008585
GO:0061458 GO:0061458
GO:0010712 GO:0010712
GO:0048608 GO:0048608
GO:0046545 GO:0046545
GO:0032729 GO:0032729
GO:0010818 GO:0010818
GO:0070374 GO:0070374
GO:0040008 GO:0040008
GO:0014856 GO:0014856
GO:0046660 GO:0046660
GO:0033002 GO:0033002
GO:0042327 GO:0042327
GO:0010714 GO:0010714
GO:0009123 GO:0009123
GO:0032965 GO:0032965
GO:0048771 GO:0048771
GO:0031128 GO:0031128
GO:0032733 GO:0032733
GO:0009156 GO:0009156
GO:0002250 GO:0002250
GO:0003156 GO:0003156
GO:0032967 GO:0032967
GO:0001934 GO:0001934
GO:0043410 GO:0043410
GO:0048247 GO:0048247
GO:0009410 GO:0009410
GO:0110110 GO:0110110
GO:0031952 GO:0031952
GO:0002460 GO:0002460
GO:0009161 GO:0009161
GO:0034103 GO:0034103
GO:0050869 GO:0050869
GO:0060485 GO:0060485
GO:0002706 GO:0002706
GO:0003014 GO:0003014
GO:1901652 GO:1901652
GO:0030316 GO:0030316
GO:0030856 GO:0030856
GO:0008217 GO:0008217
GO:2000242 GO:2000242
GO:0030546 GO:0030546
GO:0048018 GO:0048018
GO:0030545 GO:0030545
GO:0008083 GO:0008083
GO:0016471 GO:0016471
GO:0030672 GO:0030672
GO:0099501 GO:0099501
GO:0099569 GO:0099569
                                                                                                                            Function
GO:0070372                                                                                       regulation of ERK1 and ERK2 cascade
GO:0001759                                                                                                           organ induction
GO:0070371                                                                                                     ERK1 and ERK2 cascade
GO:0032609                                                                                             type II interferon production
GO:0032649                                                                               regulation of type II interferon production
GO:0008585                                                                                                  female gonad development
GO:0061458                                                                                           reproductive system development
GO:0010712                                                                                  regulation of collagen metabolic process
GO:0048608                                                                                        reproductive structure development
GO:0046545                                                                      development of primary female sexual characteristics
GO:0032729                                                                      positive regulation of type II interferon production
GO:0010818                                                                                                         T cell chemotaxis
GO:0070374                                                                              positive regulation of ERK1 and ERK2 cascade
GO:0040008                                                                                                      regulation of growth
GO:0014856                                                                                        skeletal muscle cell proliferation
GO:0046660                                                                                                female sex differentiation
GO:0033002                                                                                                 muscle cell proliferation
GO:0042327                                                                                    positive regulation of phosphorylation
GO:0010714                                                                         positive regulation of collagen metabolic process
GO:0009123                                                                                nucleoside monophosphate metabolic process
GO:0032965                                                                               regulation of collagen biosynthetic process
GO:0048771                                                                                                         tissue remodeling
GO:0031128                                                                                                   developmental induction
GO:0032733                                                                          positive regulation of interleukin-10 production
GO:0009156                                                                         ribonucleoside monophosphate biosynthetic process
GO:0002250                                                                                                  adaptive immune response
GO:0003156                                                                                      regulation of animal organ formation
GO:0032967                                                                      positive regulation of collagen biosynthetic process
GO:0001934                                                                            positive regulation of protein phosphorylation
GO:0043410                                                                                       positive regulation of MAPK cascade
GO:0048247                                                                                                     lymphocyte chemotaxis
GO:0009410                                                                                           response to xenobiotic stimulus
GO:0110110                                                                         positive regulation of animal organ morphogenesis
GO:0031952                                                                                 regulation of protein autophosphorylation
GO:0002460 adaptive immune response based on somatic recombination of immune receptors built from immunoglobulin superfamily domains
GO:0009161                                                                            ribonucleoside monophosphate metabolic process
GO:0034103                                                                                           regulation of tissue remodeling
GO:0050869                                                                                  negative regulation of B cell activation
GO:0060485                                                                                                    mesenchyme development
GO:0002706                                                                                regulation of lymphocyte mediated immunity
GO:0003014                                                                                                      renal system process
GO:1901652                                                                                                       response to peptide
GO:0030316                                                                                                osteoclast differentiation
GO:0030856                                                                             regulation of epithelial cell differentiation
GO:0008217                                                                                              regulation of blood pressure
GO:2000242                                                                               negative regulation of reproductive process
GO:0030546                                                                                     signaling receptor activator activity
GO:0048018                                                                                                  receptor ligand activity
GO:0030545                                                                                     signaling receptor regulator activity
GO:0008083                                                                                                    growth factor activity
GO:0016471                                                                        vacuolar proton-transporting V-type ATPase complex
GO:0030672                                                                                                 synaptic vesicle membrane
GO:0099501                                                                                                 exocytic vesicle membrane
GO:0099569                                                                                                  presynaptic cytoskeleton

$DOX_0.1_48
                GO_ID
GO:1901136 GO:1901136
GO:0006516 GO:0006516
GO:0008045 GO:0008045
GO:0030903 GO:0030903
GO:0086010 GO:0086010
GO:0001508 GO:0001508
GO:0043062 GO:0043062
GO:0097205 GO:0097205
GO:0051899 GO:0051899
GO:0030198 GO:0030198
GO:0061337 GO:0061337
GO:0045229 GO:0045229
GO:0003094 GO:0003094
GO:0006026 GO:0006026
GO:0044409 GO:0044409
GO:0046718 GO:0046718
GO:0006027 GO:0006027
GO:0099638 GO:0099638
GO:0086012 GO:0086012
GO:0001570 GO:0001570
GO:0086001 GO:0086001
GO:0086002 GO:0086002
GO:0016798 GO:0016798
GO:0004888 GO:0004888
GO:0004553 GO:0004553
GO:0042734 GO:0042734
GO:0043202 GO:0043202
GO:0005775 GO:0005775
GO:0045211 GO:0045211
GO:0030315 GO:0030315
GO:0005796 GO:0005796
                                                                      Function
GO:1901136                           carbohydrate derivative catabolic process
GO:0006516                                      glycoprotein catabolic process
GO:0008045                                          motor neuron axon guidance
GO:0030903                                               notochord development
GO:0086010                     membrane depolarization during action potential
GO:0001508                                                    action potential
GO:0043062                                extracellular structure organization
GO:0097205                                                    renal filtration
GO:0051899                                             membrane depolarization
GO:0030198                                   extracellular matrix organization
GO:0061337                                                  cardiac conduction
GO:0045229                       external encapsulating structure organization
GO:0003094                                               glomerular filtration
GO:0006026                                       aminoglycan catabolic process
GO:0044409                                                     entry into host
GO:0046718                                          viral entry into host cell
GO:0006027                                 glycosaminoglycan catabolic process
GO:0099638                       endosome to plasma membrane protein transport
GO:0086012 membrane depolarization during cardiac muscle cell action potential
GO:0001570                                                      vasculogenesis
GO:0086001                                cardiac muscle cell action potential
GO:0086002        cardiac muscle cell action potential involved in contraction
GO:0016798                        hydrolase activity, acting on glycosyl bonds
GO:0004888                           transmembrane signaling receptor activity
GO:0004553                hydrolase activity, hydrolyzing O-glycosyl compounds
GO:0042734                                                presynaptic membrane
GO:0043202                                                     lysosomal lumen
GO:0005775                                                      vacuolar lumen
GO:0045211                                               postsynaptic membrane
GO:0030315                                                            T-tubule
GO:0005796                                                         Golgi lumen

$DOX_0.5_3
                GO_ID
GO:0060411 GO:0060411
GO:0003007 GO:0003007
GO:0060840 GO:0060840
GO:0035270 GO:0035270
GO:0030098 GO:0030098
GO:0003279 GO:0003279
GO:0045165 GO:0045165
GO:0090092 GO:0090092
GO:0003281 GO:0003281
GO:0048844 GO:0048844
GO:0003151 GO:0003151
GO:1903131 GO:1903131
GO:0071559 GO:0071559
GO:0060419 GO:0060419
GO:0017015 GO:0017015
GO:1903844 GO:1903844
GO:0035019 GO:0035019
GO:0035904 GO:0035904
GO:0030217 GO:0030217
GO:0071560 GO:0071560
GO:0060412 GO:0060412
GO:0003170 GO:0003170
GO:0055017 GO:0055017
GO:0060976 GO:0060976
GO:0046649 GO:0046649
GO:0002521 GO:0002521
GO:1902105 GO:1902105
GO:0030278 GO:0030278
GO:1903706 GO:1903706
GO:0035051 GO:0035051
GO:0042110 GO:0042110
GO:0002065 GO:0002065
GO:0033689 GO:0033689
GO:0003181 GO:0003181
GO:0009896 GO:0009896
GO:0035909 GO:0035909
GO:0007179 GO:0007179
GO:0017148 GO:0017148
GO:0048568 GO:0048568
GO:0002067 GO:0002067
GO:0060537 GO:0060537
GO:0005978 GO:0005978
GO:0009250 GO:0009250
GO:0045785 GO:0045785
GO:0003171 GO:0003171
GO:0003712 GO:0003712
GO:0140297 GO:0140297
GO:0061733 GO:0061733
GO:0034212 GO:0034212
GO:0008080 GO:0008080
GO:0001217 GO:0001217
GO:0003713 GO:0003713
GO:0004402 GO:0004402
GO:0061629 GO:0061629
GO:0016407 GO:0016407
GO:0001227 GO:0001227
GO:0016410 GO:0016410
GO:0046332 GO:0046332
GO:0003714 GO:0003714
GO:0050321 GO:0050321
GO:0035497 GO:0035497
GO:0048156 GO:0048156
GO:0010484 GO:0010484
GO:0005667 GO:0005667
GO:0090575 GO:0090575
                                                                                         Function
GO:0060411                                                           cardiac septum morphogenesis
GO:0003007                                                                    heart morphogenesis
GO:0060840                                                                     artery development
GO:0035270                                                           endocrine system development
GO:0030098                                                             lymphocyte differentiation
GO:0003279                                                             cardiac septum development
GO:0045165                                                                   cell fate commitment
GO:0090092 regulation of transmembrane receptor protein serine/threonine kinase signaling pathway
GO:0003281                                                         ventricular septum development
GO:0048844                                                                   artery morphogenesis
GO:0003151                                                            outflow tract morphogenesis
GO:1903131                                                       mononuclear cell differentiation
GO:0071559                                            response to transforming growth factor beta
GO:0060419                                                                           heart growth
GO:0017015               regulation of transforming growth factor beta receptor signaling pathway
GO:1903844            regulation of cellular response to transforming growth factor beta stimulus
GO:0035019                                               somatic stem cell population maintenance
GO:0035904                                                                      aorta development
GO:0030217                                                                 T cell differentiation
GO:0071560                          cellular response to transforming growth factor beta stimulus
GO:0060412                                                       ventricular septum morphogenesis
GO:0003170                                                                heart valve development
GO:0055017                                                           cardiac muscle tissue growth
GO:0060976                                                       coronary vasculature development
GO:0046649                                                                  lymphocyte activation
GO:0002521                                                              leukocyte differentiation
GO:1902105                                                regulation of leukocyte differentiation
GO:0030278                                                             regulation of ossification
GO:1903706                                                              regulation of hemopoiesis
GO:0035051                                                             cardiocyte differentiation
GO:0042110                                                                      T cell activation
GO:0002065                                      columnar/cuboidal epithelial cell differentiation
GO:0033689                                        negative regulation of osteoblast proliferation
GO:0003181                                                   atrioventricular valve morphogenesis
GO:0009896                                               positive regulation of catabolic process
GO:0035909                                                                    aorta morphogenesis
GO:0007179                             transforming growth factor beta receptor signaling pathway
GO:0017148                                                     negative regulation of translation
GO:0048568                                                            embryonic organ development
GO:0002067                                              glandular epithelial cell differentiation
GO:0060537                                                              muscle tissue development
GO:0005978                                                          glycogen biosynthetic process
GO:0009250                                                            glucan biosynthetic process
GO:0045785                                                   positive regulation of cell adhesion
GO:0003171                                                     atrioventricular valve development
GO:0003712                                                     transcription coregulator activity
GO:0140297                                               DNA-binding transcription factor binding
GO:0061733                                            peptide-lysine-N-acetyltransferase activity
GO:0034212                                                   peptide N-acetyltransferase activity
GO:0008080                                                           N-acetyltransferase activity
GO:0001217                                           DNA-binding transcription repressor activity
GO:0003713                                                     transcription coactivator activity
GO:0004402                                                     histone acetyltransferase activity
GO:0061629                    RNA polymerase II-specific DNA-binding transcription factor binding
GO:0016407                                                             acetyltransferase activity
GO:0001227               DNA-binding transcription repressor activity, RNA polymerase II-specific
GO:0016410                                                             N-acyltransferase activity
GO:0046332                                                                           SMAD binding
GO:0003714                                                     transcription corepressor activity
GO:0050321                                                            tau-protein kinase activity
GO:0035497                                                          cAMP response element binding
GO:0048156                                                                    tau protein binding
GO:0010484                                                  histone H3 acetyltransferase activity
GO:0005667                                                        transcription regulator complex
GO:0090575                                      RNA polymerase II transcription regulator complex

$DOX_0.5_48
                GO_ID
GO:0051716 GO:0051716
GO:0050896 GO:0050896
GO:0023052 GO:0023052
GO:0008150 GO:0008150
GO:0007154 GO:0007154
GO:0050789 GO:0050789
GO:0009987 GO:0009987
GO:0065007 GO:0065007
GO:0019725 GO:0019725
GO:0032501 GO:0032501
GO:0006942 GO:0006942
GO:0072359 GO:0072359
GO:0003015 GO:0003015
GO:0050794 GO:0050794
GO:0060047 GO:0060047
GO:0051128 GO:0051128
GO:0070887 GO:0070887
GO:0051179 GO:0051179
GO:0030003 GO:0030003
GO:1903522 GO:1903522
GO:0007165 GO:0007165
GO:0048856 GO:0048856
GO:0042221 GO:0042221
GO:0032502 GO:0032502
GO:0006937 GO:0006937
GO:0016236 GO:0016236
GO:0005488 GO:0005488
GO:0003674 GO:0003674
GO:0005515 GO:0005515
GO:0043168 GO:0043168
GO:0097367 GO:0097367
GO:0005524 GO:0005524
GO:0030554 GO:0030554
GO:1901363 GO:1901363
GO:0035639 GO:0035639
GO:0032559 GO:0032559
GO:0000166 GO:0000166
GO:1901265 GO:1901265
GO:0017076 GO:0017076
GO:0032555 GO:0032555
GO:0032553 GO:0032553
GO:0003824 GO:0003824
GO:0016817 GO:0016817
GO:0016818 GO:0016818
GO:0016462 GO:0016462
GO:0005575 GO:0005575
GO:0110165 GO:0110165
GO:0071944 GO:0071944
GO:0005886 GO:0005886
GO:0005737 GO:0005737
GO:0016020 GO:0016020
GO:0005622 GO:0005622
GO:0043229 GO:0043229
GO:0099512 GO:0099512
GO:0099081 GO:0099081
GO:0000323 GO:0000323
GO:0005764 GO:0005764
                                                                                     Function
GO:0051716                                                      cellular response to stimulus
GO:0050896                                                               response to stimulus
GO:0023052                                                                          signaling
GO:0008150                                                                 biological_process
GO:0007154                                                                 cell communication
GO:0050789                                                   regulation of biological process
GO:0009987                                                                   cellular process
GO:0065007                                                              biological regulation
GO:0019725                                                               cellular homeostasis
GO:0032501                                                   multicellular organismal process
GO:0006942                                          regulation of striated muscle contraction
GO:0072359                                                     circulatory system development
GO:0003015                                                                      heart process
GO:0050794                                                     regulation of cellular process
GO:0060047                                                                  heart contraction
GO:0051128                                      regulation of cellular component organization
GO:0070887                                             cellular response to chemical stimulus
GO:0051179                                                                       localization
GO:0030003                                        intracellular monoatomic cation homeostasis
GO:1903522                                                    regulation of blood circulation
GO:0007165                                                                signal transduction
GO:0048856                                                   anatomical structure development
GO:0042221                                                               response to chemical
GO:0032502                                                              developmental process
GO:0006937                                                   regulation of muscle contraction
GO:0016236                                                                     macroautophagy
GO:0005488                                                                            binding
GO:0003674                                                                 molecular_function
GO:0005515                                                                    protein binding
GO:0043168                                                                      anion binding
GO:0097367                                                    carbohydrate derivative binding
GO:0005524                                                                        ATP binding
GO:0030554                                                          adenyl nucleotide binding
GO:1901363                                                      heterocyclic compound binding
GO:0035639                                         purine ribonucleoside triphosphate binding
GO:0032559                                                      adenyl ribonucleotide binding
GO:0000166                                                                 nucleotide binding
GO:1901265                                                       nucleoside phosphate binding
GO:0017076                                                          purine nucleotide binding
GO:0032555                                                      purine ribonucleotide binding
GO:0032553                                                             ribonucleotide binding
GO:0003824                                                                 catalytic activity
GO:0016817                                      hydrolase activity, acting on acid anhydrides
GO:0016818 hydrolase activity, acting on acid anhydrides, in phosphorus-containing anhydrides
GO:0016462                                                           pyrophosphatase activity
GO:0005575                                                                 cellular_component
GO:0110165                                                         cellular anatomical entity
GO:0071944                                                                     cell periphery
GO:0005886                                                                    plasma membrane
GO:0005737                                                                          cytoplasm
GO:0016020                                                                           membrane
GO:0005622                                                 intracellular anatomical structure
GO:0043229                                                            intracellular organelle
GO:0099512                                                               supramolecular fiber
GO:0099081                                                             supramolecular polymer
GO:0000323                                                                      lytic vacuole
GO:0005764                                                                           lysosome

$CX_low_non_response
                GO_ID                      Function
GO:0038023 GO:0038023   signaling receptor activity
GO:0060089 GO:0060089 molecular transducer activity
GO:0005770 GO:0005770                 late endosome
GO:0098978 GO:0098978         glutamatergic synapse
GO:0030055 GO:0030055       cell-substrate junction

$CX_low_mid_late_response
                GO_ID                                         Function
GO:0051493 GO:0051493          regulation of cytoskeleton organization
GO:0070301 GO:0070301           cellular response to hydrogen peroxide
GO:0006304 GO:0006304                                 DNA modification
GO:0009223 GO:0009223 pyrimidine deoxyribonucleotide catabolic process
GO:0048308 GO:0048308                            organelle inheritance
GO:0048313 GO:0048313                                Golgi inheritance
GO:0006213 GO:0006213          pyrimidine nucleoside metabolic process
GO:0032404 GO:0032404                  mismatch repair complex binding

$CX_high_non_response
                GO_ID                                     Function
GO:0061564 GO:0061564                             axon development
GO:0006397 GO:0006397                              mRNA processing
GO:0007409 GO:0007409                                 axonogenesis
GO:0060627 GO:0060627     regulation of vesicle-mediated transport
GO:0090066 GO:0090066      regulation of anatomical structure size
GO:0007186 GO:0007186 G protein-coupled receptor signaling pathway
GO:0005509 GO:0005509                          calcium ion binding
GO:0046873 GO:0046873 metal ion transmembrane transporter activity
GO:1902495 GO:1902495            transmembrane transporter complex
GO:0035770 GO:0035770                    ribonucleoprotein granule
GO:0030424 GO:0030424                                         axon
GO:0031252 GO:0031252                            cell leading edge
GO:0036464 GO:0036464        cytoplasmic ribonucleoprotein granule

$CX_high_mid_late_response
                GO_ID
GO:0000722 GO:0000722
GO:0006220 GO:0006220
GO:0009132 GO:0009132
GO:0090657 GO:0090657
GO:0032042 GO:0032042
GO:0006264 GO:0006264
GO:0009185 GO:0009185
GO:0004518 GO:0004518
GO:0140299 GO:0140299
GO:0016894 GO:0016894
GO:0016796 GO:0016796
GO:0004527 GO:0004527
GO:1990498 GO:1990498
                                                                                                               Function
GO:0000722                                                                       telomere maintenance via recombination
GO:0006220                                                                      pyrimidine nucleotide metabolic process
GO:0009132                                                                     nucleoside diphosphate metabolic process
GO:0090657                                                                                   telomeric loop disassembly
GO:0032042                                                                          mitochondrial DNA metabolic process
GO:0006264                                                                                mitochondrial DNA replication
GO:0009185                                                                 ribonucleoside diphosphate metabolic process
GO:0004518                                                                                            nuclease activity
GO:0140299                                                                               small molecule sensor activity
GO:0016894 endonuclease activity, active with either ribo- or deoxyribonucleic acids and producing 3&#39;-phosphomonoesters
GO:0016796  exonuclease activity, active with either ribo- or deoxyribonucleic acids and producing 5&#39;-phosphomonoesters
GO:0004527                                                                                         exonuclease activity
GO:1990498                                                                                  mitotic spindle microtubule

$DOX_low_non_response
                GO_ID                                     Function
GO:0034470 GO:0034470                             ncRNA processing
GO:0022613 GO:0022613         ribonucleoprotein complex biogenesis
GO:0042254 GO:0042254                          ribosome biogenesis
GO:0006364 GO:0006364                              rRNA processing
GO:0140053 GO:0140053                mitochondrial gene expression
GO:0016072 GO:0016072                       rRNA metabolic process
GO:0006413 GO:0006413                     translational initiation
GO:0032543 GO:0032543                    mitochondrial translation
GO:0008135 GO:0008135     translation factor activity, RNA binding
GO:0030684 GO:0030684                                  preribosome
GO:0098800 GO:0098800 inner mitochondrial membrane protein complex
GO:0098803 GO:0098803                    respiratory chain complex
GO:0019866 GO:0019866                     organelle inner membrane

$DOX_low_mid_late_response
                GO_ID
GO:0009416 GO:0009416
GO:0050673 GO:0050673
GO:0010955 GO:0010955
GO:1903318 GO:1903318
GO:0002251 GO:0002251
GO:0014029 GO:0014029
GO:0097237 GO:0097237
GO:0055021 GO:0055021
GO:0031645 GO:0031645
GO:0034350 GO:0034350
GO:0055023 GO:0055023
GO:0010804 GO:0010804
GO:0046686 GO:0046686
GO:0005402 GO:0005402
GO:0008296 GO:0008296
                                                                          Function
GO:0009416                                              response to light stimulus
GO:0050673                                           epithelial cell proliferation
GO:0010955                               negative regulation of protein processing
GO:1903318                               negative regulation of protein maturation
GO:0002251                                organ or tissue specific immune response
GO:0014029                                                  neural crest formation
GO:0097237                                    cellular response to toxic substance
GO:0055021                              regulation of cardiac muscle tissue growth
GO:0031645                           negative regulation of nervous system process
GO:0034350                              regulation of glial cell apoptotic process
GO:0055023                     positive regulation of cardiac muscle tissue growth
GO:0010804 negative regulation of tumor necrosis factor-mediated signaling pathway
GO:0046686                                                 response to cadmium ion
GO:0005402                       carbohydrate:monoatomic cation symporter activity
GO:0008296                                          3&#39;-5&#39;-DNA exonuclease activity

$DOX_high_non_response
                GO_ID                          Function
GO:0002181 GO:0002181           cytoplasmic translation
GO:0022626 GO:0022626                cytosolic ribosome
GO:0022625 GO:0022625 cytosolic large ribosomal subunit
GO:0042788 GO:0042788                polysomal ribosome

$DOX_high_response
                GO_ID                                         Function
GO:0045598 GO:0045598           regulation of fat cell differentiation
GO:0045444 GO:0045444                         fat cell differentiation
GO:0009084 GO:0009084 glutamine family amino acid biosynthetic process</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 22631)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats4    tools     stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
[1] GO.db_3.18.0         AnnotationDbi_1.64.1 IRanges_2.36.0      
[4] S4Vectors_0.40.1     Biobase_2.62.0       BiocGenerics_0.48.1 
[7] dplyr_1.1.4          UpSetR_1.4.0        

loaded via a namespace (and not attached):
 [1] KEGGREST_1.42.0         gtable_0.3.6            xfun_0.50              
 [4] bslib_0.8.0             ggplot2_3.5.1           vctrs_0.6.5            
 [7] bitops_1.0-7            generics_0.1.3          tibble_3.2.1           
[10] RSQLite_2.3.3           blob_1.2.4              pkgconfig_2.0.3        
[13] lifecycle_1.0.4         GenomeInfoDbData_1.2.11 farver_2.1.2           
[16] compiler_4.3.0          stringr_1.5.1           git2r_0.35.0           
[19] Biostrings_2.70.1       munsell_0.5.1           httpuv_1.6.15          
[22] GenomeInfoDb_1.38.8     htmltools_0.5.8.1       sass_0.4.9             
[25] RCurl_1.98-1.13         yaml_2.3.10             later_1.3.2            
[28] pillar_1.10.1           crayon_1.5.3            jquerylib_0.1.4        
[31] whisker_0.4.1           cachem_1.0.8            tidyselect_1.2.1       
[34] digest_0.6.34           stringi_1.8.3           labeling_0.4.3         
[37] rprojroot_2.0.4         fastmap_1.1.1           grid_4.3.0             
[40] colorspace_2.1-0        cli_3.6.1               magrittr_2.0.3         
[43] withr_3.0.2             scales_1.3.0            promises_1.3.0         
[46] bit64_4.0.5             rmarkdown_2.29          XVector_0.42.0         
[49] httr_1.4.7              bit_4.0.5               gridExtra_2.3          
[52] workflowr_1.7.1         png_0.1-8               memoise_2.0.1          
[55] evaluate_1.0.3          knitr_1.49              rlang_1.1.3            
[58] Rcpp_1.0.12             glue_1.7.0              DBI_1.2.3              
[61] rstudioapi_0.17.1       jsonlite_1.8.9          R6_2.5.1               
[64] plyr_1.8.9              fs_1.6.3                zlibbioc_1.48.0        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
