<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Corrmotif Analysis</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Paul_CX_2025</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Corrmotif Analysis</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-07-03
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>CX5461_Project/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity itâ€™s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250129code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250129)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250129code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250129)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01PaulCX2025treeeca9f052454582883f0843ae1151e67e21b76aadtargetblankeca9f05a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/tree/eca9f052454582883f0843ae1151e67e21b76aad" target="_blank">eca9f05</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01PaulCX2025treeeca9f052454582883f0843ae1151e67e21b76aadtargetblankeca9f05a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sayanpaul01/Paul_CX_2025/tree/eca9f052454582883f0843ae1151e67e21b76aad" target="_blank">eca9f05</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    0.1 box.svg
    Ignored:    Rplot04.svg

Untracked files:
    Untracked:  0.1 density.svg
    Untracked:  0.1.emf
    Untracked:  0.1.svg
    Untracked:  0.5 box.svg
    Untracked:  0.5 density.svg
    Untracked:  0.5.svg
    Untracked:  Additional/
    Untracked:  Autosome factors.svg
    Untracked:  CX_5461_Pattern_Genes_24hr.csv
    Untracked:  CX_5461_Pattern_Genes_3hr.csv
    Untracked:  Cell viability box plot.svg
    Untracked:  DEG GO terms.svg
    Untracked:  DNA damage associated GO terms.svg
    Untracked:  DRC1.svg
    Untracked:  Figure 1.jpeg
    Untracked:  Figure 1.pdf
    Untracked:  Figure_CM_Purity.pdf
    Untracked:  G Quadruplex DEGs.svg
    Untracked:  PC2 Vs PC3 Autosome.svg
    Untracked:  PCA autosome.svg
    Untracked:  Rplot 18.svg
    Untracked:  Rplot.svg
    Untracked:  Rplot01.svg
    Untracked:  Rplot02.svg
    Untracked:  Rplot03.svg
    Untracked:  Rplot05.svg
    Untracked:  Rplot06.svg
    Untracked:  Rplot07.svg
    Untracked:  Rplot08.jpeg
    Untracked:  Rplot08.svg
    Untracked:  Rplot09.svg
    Untracked:  Rplot10.svg
    Untracked:  Rplot11.svg
    Untracked:  Rplot12.svg
    Untracked:  Rplot13.svg
    Untracked:  Rplot14.svg
    Untracked:  Rplot15.svg
    Untracked:  Rplot16.svg
    Untracked:  Rplot17.svg
    Untracked:  Rplot18.svg
    Untracked:  Rplot19.svg
    Untracked:  Rplot20.svg
    Untracked:  Rplot21.svg
    Untracked:  Rplot22.svg
    Untracked:  Rplot23.svg
    Untracked:  Rplot24.svg
    Untracked:  S15.pdf
    Untracked:  TOP2B.bed
    Untracked:  TS HPA (Violin).svg
    Untracked:  TS HPA.svg
    Untracked:  TS_HA.svg
    Untracked:  TS_HV.svg
    Untracked:  Violin HA.svg
    Untracked:  Violin HV (CX vs DOX).svg
    Untracked:  Violin HV.svg
    Untracked:  data/AF.csv
    Untracked:  data/AF_Mapped.csv
    Untracked:  data/AF_genes.csv
    Untracked:  data/Annotated_DOX_Gene_Table.csv
    Untracked:  data/BP/
    Untracked:  data/CAD_genes.csv
    Untracked:  data/Cardiotox.csv
    Untracked:  data/Cardiotox_mapped.csv
    Untracked:  data/Col_DEG_proportion_data.csv
    Untracked:  data/Col_DEGs.csv
    Untracked:  data/Corrmotif_GO/
    Untracked:  data/DOX_Vald.csv
    Untracked:  data/DOX_Vald_Mapped.csv
    Untracked:  data/DOX_alt.csv
    Untracked:  data/Entrez_Cardiotox.csv
    Untracked:  data/Entrez_Cardiotox_Mapped.csv
    Untracked:  data/GWAS.xlsx
    Untracked:  data/GWAS_SNPs.bed
    Untracked:  data/HF.csv
    Untracked:  data/HF_Mapped.csv
    Untracked:  data/HF_genes.csv
    Untracked:  data/Hypertension_genes.csv
    Untracked:  data/MI_genes.csv
    Untracked:  data/P53_Target_mapped.csv
    Untracked:  data/Sample_annotated.csv
    Untracked:  data/Samples.csv
    Untracked:  data/Samples.xlsx
    Untracked:  data/TOP2A.bed
    Untracked:  data/TOP2A_target.csv
    Untracked:  data/TOP2A_target_lit.csv
    Untracked:  data/TOP2A_target_lit_mapped.csv
    Untracked:  data/TOP2A_target_mapped.csv
    Untracked:  data/TOP2B.bed
    Untracked:  data/TOP2B_target.csv
    Untracked:  data/TOP2B_target_heatmap.csv
    Untracked:  data/TOP2B_target_heatmap_mapped.csv
    Untracked:  data/TOP2B_target_mapped.csv
    Untracked:  data/TS.csv
    Untracked:  data/TS_HPA.csv
    Untracked:  data/TS_HPA_mapped.csv
    Untracked:  data/Toptable_CX_0.1_24.csv
    Untracked:  data/Toptable_CX_0.1_3.csv
    Untracked:  data/Toptable_CX_0.1_48.csv
    Untracked:  data/Toptable_CX_0.5_24.csv
    Untracked:  data/Toptable_CX_0.5_3.csv
    Untracked:  data/Toptable_CX_0.5_48.csv
    Untracked:  data/Toptable_DOX_0.1_24.csv
    Untracked:  data/Toptable_DOX_0.1_3.csv
    Untracked:  data/Toptable_DOX_0.1_48.csv
    Untracked:  data/Toptable_DOX_0.5_24.csv
    Untracked:  data/Toptable_DOX_0.5_3.csv
    Untracked:  data/Toptable_DOX_0.5_48.csv
    Untracked:  data/count.tsv
    Untracked:  data/heatmap.csv
    Untracked:  data/ts_data_mapped
    Untracked:  results/
    Untracked:  run_bedtools.bat

Unstaged changes:
    Deleted:    analysis/Actox.Rmd
    Modified:   data/DEGs/Toptable_DOX_0.5_3.csv
    Modified:   data/DOX_0.5_48 (Combined).csv
    Modified:   data/Total_number_of_Mapped_reads_by_Individuals.csv

</code></pre>
<p>
Note that any generated files, e.g.Â HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Corrmotif_all.Rmd</code>) and
HTML (<code>docs/Corrmotif_all.html</code>) files. If youâ€™ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/b08bc389e65825d6f94786c53af77f69811dec5c/analysis/Corrmotif_all.Rmd" target="_blank">b08bc38</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/b08bc389e65825d6f94786c53af77f69811dec5c/docs/Corrmotif_all.html" target="_blank">b08bc38</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/4e8a4bc605717eb9af559e2d88d0e720de5e798d/analysis/Corrmotif_all.Rmd" target="_blank">4e8a4bc</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-24
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/4e8a4bc605717eb9af559e2d88d0e720de5e798d/docs/Corrmotif_all.html" target="_blank">4e8a4bc</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-24
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/2862c42bbe271925d87c00cd50b08289692b2517/analysis/Corrmotif_all.Rmd" target="_blank">2862c42</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-24
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/2862c42bbe271925d87c00cd50b08289692b2517/docs/Corrmotif_all.html" target="_blank">2862c42</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-24
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/ebfddcd2544a6364ccb5f8df72be2a583bc7ab4b/analysis/Corrmotif_all.Rmd" target="_blank">ebfddcd</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-23
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/ebfddcd2544a6364ccb5f8df72be2a583bc7ab4b/docs/Corrmotif_all.html" target="_blank">ebfddcd</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-23
</td>
<td>
Commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/b55709531c179d5260046632aa517bd5a94898ad/docs/Corrmotif_all.html" target="_blank">b557095</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-23
</td>
<td>
Added CX_DOX shared late response (SVIP) boxplot
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/1cd333a3ea5597f1639f5d0e58bdc5270031b5c3/analysis/Corrmotif_all.Rmd" target="_blank">1cd333a</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-23
</td>
<td>
Added CX_DOX shared late response (SVIP) boxplot
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/e73d731caeb1fe0eb8f95488b5bcbd2d6049b253/docs/Corrmotif_all.html" target="_blank">e73d731</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/c4ed3e0b60e059fc645507b7232e0d9ad89ef780/analysis/Corrmotif_all.Rmd" target="_blank">c4ed3e0</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-23
</td>
<td>
Updated Corrmotif analysis and index file
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/b56a62ae3daececcbdeee937b8acc8e5fac7b4ff/docs/Corrmotif_all.html" target="_blank">b56a62a</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/bf0c4a3dac81065d5077b580f92949b6a8911bc1/analysis/Corrmotif_all.Rmd" target="_blank">bf0c4a3</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-21
</td>
<td>
Published updated Index and Corrmotif analysis with new
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="fit-limma-model-functions" class="section level2">
<h2><strong>ðŸ“Œ Fit Limma Model Functions</strong></h2>
<pre class="r"><code>## Fit limma model using code as it is found in the original cormotif code. It has
## only been modified to add names to the matrix of t values, as well as the
## limma fits

limmafit.default &lt;- function(exprs,groupid,compid) {
  limmafits  &lt;- list()
  compnum    &lt;- nrow(compid)
  genenum    &lt;- nrow(exprs)
  limmat     &lt;- matrix(0,genenum,compnum)
  limmas2    &lt;- rep(0,compnum)
  limmadf    &lt;- rep(0,compnum)
  limmav0    &lt;- rep(0,compnum)
  limmag1num &lt;- rep(0,compnum)
  limmag2num &lt;- rep(0,compnum)

  rownames(limmat)  &lt;- rownames(exprs)
  colnames(limmat)  &lt;- rownames(compid)
  names(limmas2)    &lt;- rownames(compid)
  names(limmadf)    &lt;- rownames(compid)
  names(limmav0)    &lt;- rownames(compid)
  names(limmag1num) &lt;- rownames(compid)
  names(limmag2num) &lt;- rownames(compid)

  for(i in 1:compnum) {
    selid1 &lt;- which(groupid == compid[i,1])
    selid2 &lt;- which(groupid == compid[i,2])
    eset   &lt;- new(&quot;ExpressionSet&quot;, exprs=cbind(exprs[,selid1],exprs[,selid2]))
    g1num  &lt;- length(selid1)
    g2num  &lt;- length(selid2)
    designmat &lt;- cbind(base=rep(1,(g1num+g2num)), delta=c(rep(0,g1num),rep(1,g2num)))
    fit &lt;- lmFit(eset,designmat)
    fit &lt;- eBayes(fit)
    limmat[,i] &lt;- fit$t[,2]
    limmas2[i] &lt;- fit$s2.prior
    limmadf[i] &lt;- fit$df.prior
    limmav0[i] &lt;- fit$var.prior[2]
    limmag1num[i] &lt;- g1num
    limmag2num[i] &lt;- g2num
    limmafits[[i]] &lt;- fit

    # log odds
    # w&lt;-sqrt(1+fit$var.prior[2]/(1/g1num+1/g2num))
    # log(0.99)+dt(fit$t[1,2],g1num+g2num-2+fit$df.prior,log=TRUE)-log(0.01)-dt(fit$t[1,2]/w, g1num+g2num-2+fit$df.prior, log=TRUE)+log(w)
  }
  names(limmafits) &lt;- rownames(compid)
  limmacompnum&lt;-nrow(compid)
  result&lt;-list(t       = limmat,
               v0      = limmav0,
               df0     = limmadf,
               s20     = limmas2,
               g1num   = limmag1num,
               g2num   = limmag2num,
               compnum = limmacompnum,
               fits    = limmafits)
}

limmafit.counts &lt;-
  function (exprs, groupid, compid, norm.factor.method = &quot;TMM&quot;, voom.normalize.method = &quot;none&quot;)
  {
    limmafits  &lt;- list()
    compnum    &lt;- nrow(compid)
    genenum    &lt;- nrow(exprs)
    limmat     &lt;- matrix(NA,genenum,compnum)
    limmas2    &lt;- rep(0,compnum)
    limmadf    &lt;- rep(0,compnum)
    limmav0    &lt;- rep(0,compnum)
    limmag1num &lt;- rep(0,compnum)
    limmag2num &lt;- rep(0,compnum)

    rownames(limmat)  &lt;- rownames(exprs)
    colnames(limmat)  &lt;- rownames(compid)
    names(limmas2)    &lt;- rownames(compid)
    names(limmadf)    &lt;- rownames(compid)
    names(limmav0)    &lt;- rownames(compid)
    names(limmag1num) &lt;- rownames(compid)
    names(limmag2num) &lt;- rownames(compid)

    for (i in 1:compnum) {
      message(paste(&quot;Running limma for comparision&quot;,i,&quot;/&quot;,compnum))
      selid1 &lt;- which(groupid == compid[i, 1])
      selid2 &lt;- which(groupid == compid[i, 2])
      # make a new count data frame
      counts &lt;- cbind(exprs[, selid1], exprs[, selid2])

      # remove NAs
      not.nas &lt;- which(apply(counts, 1, function(x) !any(is.na(x))) == TRUE)

      # runn voom/limma
      d &lt;- DGEList(counts[not.nas,])
      d &lt;- calcNormFactors(d, method = norm.factor.method)
      g1num &lt;- length(selid1)
      g2num &lt;- length(selid2)
      designmat &lt;- cbind(base = rep(1, (g1num + g2num)), delta = c(rep(0,
                                                                       g1num), rep(1, g2num)))

      y &lt;- voom(d, designmat, normalize.method = voom.normalize.method)
      fit &lt;- lmFit(y, designmat)
      fit &lt;- eBayes(fit)

      limmafits[[i]] &lt;- fit
      limmat[not.nas, i] &lt;- fit$t[, 2]
      limmas2[i] &lt;- fit$s2.prior
      limmadf[i] &lt;- fit$df.prior
      limmav0[i] &lt;- fit$var.prior[2]
      limmag1num[i] &lt;- g1num
      limmag2num[i] &lt;- g2num
    }
    limmacompnum &lt;- nrow(compid)
    names(limmafits) &lt;- rownames(compid)
    result &lt;- list(t       = limmat,
                   v0      = limmav0,
                   df0     = limmadf,
                   s20     = limmas2,
                   g1num   = limmag1num,
                   g2num   = limmag2num,
                   compnum = limmacompnum,
                   fits    = limmafits)
  }

limmafit.list &lt;-
  function (fitlist, cmp.idx=2)
  {
    compnum    &lt;- length(fitlist)

    genes &lt;- c()
    for (i in 1:compnum) genes &lt;- unique(c(genes, rownames(fitlist[[i]])))

    genenum    &lt;- length(genes)
    limmat     &lt;- matrix(NA,genenum,compnum)
    limmas2    &lt;- rep(0,compnum)
    limmadf    &lt;- rep(0,compnum)
    limmav0    &lt;- rep(0,compnum)
    limmag1num &lt;- rep(0,compnum)
    limmag2num &lt;- rep(0,compnum)

    rownames(limmat)  &lt;- genes
    colnames(limmat)  &lt;- names(fitlist)
    names(limmas2)    &lt;- names(fitlist)
    names(limmadf)    &lt;- names(fitlist)
    names(limmav0)    &lt;- names(fitlist)
    names(limmag1num) &lt;- names(fitlist)
    names(limmag2num) &lt;- names(fitlist)

    for (i in 1:compnum) {
      this.t &lt;- fitlist[[i]]$t[,cmp.idx]
      limmat[names(this.t),i] &lt;- this.t

      limmas2[i]    &lt;- fitlist[[i]]$s2.prior
      limmadf[i]    &lt;- fitlist[[i]]$df.prior
      limmav0[i]    &lt;- fitlist[[i]]$var.prior[cmp.idx]
      limmag1num[i] &lt;- sum(fitlist[[i]]$design[,cmp.idx]==0)
      limmag2num[i] &lt;- sum(fitlist[[i]]$design[,cmp.idx]==1)
    }

    limmacompnum &lt;- compnum
    result &lt;- list(t       = limmat,
                   v0      = limmav0,
                   df0     = limmadf,
                   s20     = limmas2,
                   g1num   = limmag1num,
                   g2num   = limmag2num,
                   compnum = limmacompnum,
                   fits    = limmafits)

  }

## Rank genes based on statistics
generank&lt;-function(x) {
  xcol&lt;-ncol(x)
  xrow&lt;-nrow(x)
  result&lt;-matrix(0,xrow,xcol)
  z&lt;-(1:1:xrow)
  for(i in 1:xcol) {
    y&lt;-sort(x[,i],decreasing=TRUE,na.last=TRUE)
    result[,i]&lt;-match(x[,i],y)
    result[,i]&lt;-order(result[,i])
  }
  result
}

## Log-likelihood for moderated t under H0
modt.f0.loglike&lt;-function(x,df) {
  a&lt;-dt(x, df, log=TRUE)
  result&lt;-as.vector(a)
  flag&lt;-which(is.na(result)==TRUE)
  result[flag]&lt;-0
  result
}

## Log-likelihood for moderated t under H1
## param=c(df,g1num,g2num,v0)
modt.f1.loglike&lt;-function(x,param) {
  df&lt;-param[1]
  g1num&lt;-param[2]
  g2num&lt;-param[3]
  v0&lt;-param[4]
  w&lt;-sqrt(1+v0/(1/g1num+1/g2num))
  dt(x/w, df, log=TRUE)-log(w)
  a&lt;-dt(x/w, df, log=TRUE)-log(w)
  result&lt;-as.vector(a)
  flag&lt;-which(is.na(result)==TRUE)
  result[flag]&lt;-0
  result
}

## Correlation Motif Fit
cmfit.X&lt;-function(x, type, K=1, tol=1e-3, max.iter=100) {
  ## initialize
  xrow &lt;- nrow(x)
  xcol &lt;- ncol(x)
  loglike0 &lt;- list()
  loglike1 &lt;- list()
  p &lt;- rep(1, K)/K
  q &lt;- matrix(runif(K * xcol), K, xcol)
  q[1, ] &lt;- rep(0.01, xcol)
  for (i in 1:xcol) {
    f0 &lt;- type[[i]][[1]]
    f0param &lt;- type[[i]][[2]]
    f1 &lt;- type[[i]][[3]]
    f1param &lt;- type[[i]][[4]]
    loglike0[[i]] &lt;- f0(x[, i], f0param)
    loglike1[[i]] &lt;- f1(x[, i], f1param)
  }
  condlike &lt;- list()
  for (i in 1:xcol) {
    condlike[[i]] &lt;- matrix(0, xrow, K)
  }
  loglike.old &lt;- -1e+10
  for (i.iter in 1:max.iter) {
    if ((i.iter%%50) == 0) {
      print(paste(&quot;We have run the first &quot;, i.iter, &quot; iterations for K=&quot;,
                  K, sep = &quot;&quot;))
    }
    err &lt;- tol + 1
    clustlike &lt;- matrix(0, xrow, K)
    #templike &lt;- matrix(0, xrow, 2)
    templike1 &lt;- rep(0, xrow)
    templike2 &lt;- rep(0, xrow)
    for (j in 1:K) {
      for (i in 1:xcol) {
        templike1 &lt;- log(q[j, i]) + loglike1[[i]]
        templike2 &lt;- log(1 - q[j, i]) + loglike0[[i]]
        tempmax &lt;- Rfast::Pmax(templike1, templike2)

        templike1 &lt;- exp(templike1 - tempmax)
        templike2 &lt;- exp(templike2 - tempmax)

        tempsum &lt;- templike1 + templike2
        clustlike[, j] &lt;- clustlike[, j] + tempmax +
          log(tempsum)
        condlike[[i]][, j] &lt;- templike1/tempsum
      }
      clustlike[, j] &lt;- clustlike[, j] + log(p[j])
    }
    #tempmax &lt;- apply(clustlike, 1, max)
    tempmax &lt;- Rfast::rowMaxs(clustlike, value=TRUE)
    for (j in 1:K) {
      clustlike[, j] &lt;- exp(clustlike[, j] - tempmax)
    }
    #tempsum &lt;- apply(clustlike, 1, sum)
    tempsum &lt;- Rfast::rowsums(clustlike)
    for (j in 1:K) {
      clustlike[, j] &lt;- clustlike[, j]/tempsum
    }
    #p.new &lt;- (apply(clustlike, 2, sum) + 1)/(xrow + K)
    p.new &lt;- (Rfast::colsums(clustlike) + 1)/(xrow + K)
    q.new &lt;- matrix(0, K, xcol)
    for (j in 1:K) {
      clustpsum &lt;- sum(clustlike[, j])
      for (i in 1:xcol) {
        q.new[j, i] &lt;- (sum(clustlike[, j] * condlike[[i]][,
                                                           j]) + 1)/(clustpsum + 2)
      }
    }
    err.p &lt;- max(abs(p.new - p)/p)
    err.q &lt;- max(abs(q.new - q)/q)
    err &lt;- max(err.p, err.q)
    loglike.new &lt;- (sum(tempmax + log(tempsum)) + sum(log(p.new)) +
                      sum(log(q.new) + log(1 - q.new)))/xrow
    p &lt;- p.new
    q &lt;- q.new
    loglike.old &lt;- loglike.new
    if (err &lt; tol) {
      break
    }
  }
  clustlike &lt;- matrix(0, xrow, K)
  for (j in 1:K) {
    for (i in 1:xcol) {
      templike1 &lt;- log(q[j, i]) + loglike1[[i]]
      templike2 &lt;- log(1 - q[j, i]) + loglike0[[i]]
      tempmax &lt;- Rfast::Pmax(templike1, templike2)

      templike1 &lt;- exp(templike1 - tempmax)
      templike2 &lt;- exp(templike2 - tempmax)

      tempsum &lt;- templike1 + templike2
      clustlike[, j] &lt;- clustlike[, j] + tempmax + log(tempsum)
      condlike[[i]][, j] &lt;- templike1/tempsum
    }
    clustlike[, j] &lt;- clustlike[, j] + log(p[j])
  }
  #tempmax &lt;- apply(clustlike, 1, max)
  tempmax &lt;- Rfast::rowMaxs(clustlike, value=TRUE)
  for (j in 1:K) {
    clustlike[, j] &lt;- exp(clustlike[, j] - tempmax)
  }
  #tempsum &lt;- apply(clustlike, 1, sum)
  tempsum &lt;- Rfast::rowsums(clustlike)
  for (j in 1:K) {
    clustlike[, j] &lt;- clustlike[, j]/tempsum
  }
  p.post &lt;- matrix(0, xrow, xcol)
  for (j in 1:K) {
    for (i in 1:xcol) {
      p.post[, i] &lt;- p.post[, i] + clustlike[, j] * condlike[[i]][,
                                                                  j]
    }
  }
  loglike.old &lt;- loglike.old - (sum(log(p)) + sum(log(q) +
                                                    log(1 - q)))/xrow
  loglike.old &lt;- loglike.old * xrow
  result &lt;- list(p.post = p.post, motif.prior = p, motif.q = q,
                 loglike = loglike.old, clustlike=clustlike, condlike=condlike)
}

## Fit using (0,0,...,0) and (1,1,...,1)
cmfitall&lt;-function(x, type, tol=1e-3, max.iter=100) {
  ## initialize
  xrow&lt;-nrow(x)
  xcol&lt;-ncol(x)
  loglike0&lt;-list()
  loglike1&lt;-list()
  p&lt;-0.01

  ## compute loglikelihood
  L0&lt;-matrix(0,xrow,1)
  L1&lt;-matrix(0,xrow,1)
  for(i in 1:xcol) {
    f0&lt;-type[[i]][[1]]
    f0param&lt;-type[[i]][[2]]
    f1&lt;-type[[i]][[3]]
    f1param&lt;-type[[i]][[4]]
    loglike0[[i]]&lt;-f0(x[,i],f0param)
    loglike1[[i]]&lt;-f1(x[,i],f1param)
    L0&lt;-L0+loglike0[[i]]
    L1&lt;-L1+loglike1[[i]]
  }


  ## EM algorithm to get MLE of p and q
  loglike.old &lt;- -1e10
  for(i.iter in 1:max.iter) {
    if((i.iter%%50) == 0) {
      print(paste(&quot;We have run the first &quot;, i.iter, &quot; iterations&quot;,sep=&quot;&quot;))
    }
    err&lt;-tol+1

    ## compute posterior cluster membership
    clustlike&lt;-matrix(0,xrow,2)
    clustlike[,1]&lt;-log(1-p)+L0
    clustlike[,2]&lt;-log(p)+L1

    tempmax&lt;-apply(clustlike,1,max)
    for(j in 1:2) {
      clustlike[,j]&lt;-exp(clustlike[,j]-tempmax)
    }
    tempsum&lt;-apply(clustlike,1,sum)

    ## update motif occurrence rate
    for(j in 1:2) {
      clustlike[,j]&lt;-clustlike[,j]/tempsum
    }

    p.new&lt;-(sum(clustlike[,2])+1)/(xrow+2)

    ## evaluate convergence
    err&lt;-abs(p.new-p)/p

    ## evaluate whether the log.likelihood increases
    loglike.new&lt;-(sum(tempmax+log(tempsum))+log(p.new)+log(1-p.new))/xrow

    loglike.old&lt;-loglike.new
    p&lt;-p.new

    if(err&lt;tol) {
      break;
    }
  }

  ## compute posterior p
  clustlike&lt;-matrix(0,xrow,2)
  clustlike[,1]&lt;-log(1-p)+L0
  clustlike[,2]&lt;-log(p)+L1

  tempmax&lt;-apply(clustlike,1,max)
  for(j in 1:2) {
    clustlike[,j]&lt;-exp(clustlike[,j]-tempmax)
  }
  tempsum&lt;-apply(clustlike,1,sum)

  for(j in 1:2) {
    clustlike[,j]&lt;-clustlike[,j]/tempsum
  }

  p.post&lt;-matrix(0,xrow,xcol)
  for(i in 1:xcol) {
    p.post[,i]&lt;-clustlike[,2]
  }

  ## return

  #calculate back loglikelihood
  loglike.old&lt;-loglike.old-(log(p)+log(1-p))/xrow
  loglike.old&lt;-loglike.old*xrow
  result&lt;-list(p.post=p.post, motif.prior=p, loglike=loglike.old)
}

## Fit each dataset separately
cmfitsep&lt;-function(x, type, tol=1e-3, max.iter=100) {
  ## initialize
  xrow&lt;-nrow(x)
  xcol&lt;-ncol(x)
  loglike0&lt;-list()
  loglike1&lt;-list()
  p&lt;-0.01*rep(1,xcol)
  loglike.final&lt;-rep(0,xcol)

  ## compute loglikelihood
  for(i in 1:xcol) {
    f0&lt;-type[[i]][[1]]
    f0param&lt;-type[[i]][[2]]
    f1&lt;-type[[i]][[3]]
    f1param&lt;-type[[i]][[4]]
    loglike0[[i]]&lt;-f0(x[,i],f0param)
    loglike1[[i]]&lt;-f1(x[,i],f1param)
  }

  p.post&lt;-matrix(0,xrow,xcol)

  ## EM algorithm to get MLE of p
  for(coli in 1:xcol) {
    loglike.old &lt;- -1e10
    for(i.iter in 1:max.iter) {
      if((i.iter%%50) == 0) {
        print(paste(&quot;We have run the first &quot;, i.iter, &quot; iterations&quot;,sep=&quot;&quot;))
      }
      err&lt;-tol+1

      ## compute posterior cluster membership
      clustlike&lt;-matrix(0,xrow,2)
      clustlike[,1]&lt;-log(1-p[coli])+loglike0[[coli]]
      clustlike[,2]&lt;-log(p[coli])+loglike1[[coli]]

      tempmax&lt;-apply(clustlike,1,max)
      for(j in 1:2) {
        clustlike[,j]&lt;-exp(clustlike[,j]-tempmax)
      }
      tempsum&lt;-apply(clustlike,1,sum)

      ## evaluate whether the log.likelihood increases
      loglike.new&lt;-sum(tempmax+log(tempsum))/xrow

      ## update motif occurrence rate
      for(j in 1:2) {
        clustlike[,j]&lt;-clustlike[,j]/tempsum
      }

      p.new&lt;-(sum(clustlike[,2]))/(xrow)

      ## evaluate convergence
      err&lt;-abs(p.new-p[coli])/p[coli]
      loglike.old&lt;-loglike.new
      p[coli]&lt;-p.new

      if(err&lt;tol) {
        break;
      }
    }

    ## compute posterior p
    clustlike&lt;-matrix(0,xrow,2)
    clustlike[,1]&lt;-log(1-p[coli])+loglike0[[coli]]
    clustlike[,2]&lt;-log(p[coli])+loglike1[[coli]]

    tempmax&lt;-apply(clustlike,1,max)
    for(j in 1:2) {
      clustlike[,j]&lt;-exp(clustlike[,j]-tempmax)
    }
    tempsum&lt;-apply(clustlike,1,sum)

    for(j in 1:2) {
      clustlike[,j]&lt;-clustlike[,j]/tempsum
    }

    p.post[,coli]&lt;-clustlike[,2]
    loglike.final[coli]&lt;-loglike.old
  }


  ## return
  loglike.final&lt;-loglike.final*xrow
  result&lt;-list(p.post=p.post, motif.prior=p, loglike=loglike.final)
}

## Fit the full model
cmfitfull&lt;-function(x, type, tol=1e-3, max.iter=100) {
  ## initialize
  xrow&lt;-nrow(x)
  xcol&lt;-ncol(x)
  loglike0&lt;-list()
  loglike1&lt;-list()
  K&lt;-2^xcol
  p&lt;-rep(1,K)/K
  pattern&lt;-rep(0,xcol)
  patid&lt;-matrix(0,K,xcol)

  ## compute loglikelihood
  for(i in 1:xcol) {
    f0&lt;-type[[i]][[1]]
    f0param&lt;-type[[i]][[2]]
    f1&lt;-type[[i]][[3]]
    f1param&lt;-type[[i]][[4]]
    loglike0[[i]]&lt;-f0(x[,i],f0param)
    loglike1[[i]]&lt;-f1(x[,i],f1param)
  }
  L&lt;-matrix(0,xrow,K)
  for(i in 1:K)
  {
    patid[i,]&lt;-pattern
    for(j in 1:xcol) {
      if(pattern[j] &lt; 0.5) {
        L[,i]&lt;-L[,i]+loglike0[[j]]
      } else {
        L[,i]&lt;-L[,i]+loglike1[[j]]
      }
    }

    if(i &lt; K) {
      pattern[xcol]&lt;-pattern[xcol]+1
      j&lt;-xcol
      while(pattern[j] &gt; 1) {
        pattern[j]&lt;-0
        j&lt;-j-1
        pattern[j]&lt;-pattern[j]+1
      }
    }
  }

  ## EM algorithm to get MLE of p and q
  loglike.old &lt;- -1e10
  for(i.iter in 1:max.iter) {
    if((i.iter%%50) == 0) {
      print(paste(&quot;We have run the first &quot;, i.iter, &quot; iterations&quot;,sep=&quot;&quot;))
    }
    err&lt;-tol+1

    ## compute posterior cluster membership
    clustlike&lt;-matrix(0,xrow,K)
    for(j in 1:K) {
      clustlike[,j]&lt;-log(p[j])+L[,j]
    }

    tempmax&lt;-apply(clustlike,1,max)
    for(j in 1:K) {
      clustlike[,j]&lt;-exp(clustlike[,j]-tempmax)
    }
    tempsum&lt;-apply(clustlike,1,sum)

    ## update motif occurrence rate
    for(j in 1:K) {
      clustlike[,j]&lt;-clustlike[,j]/tempsum
    }

    p.new&lt;-(apply(clustlike,2,sum)+1)/(xrow+K)

    ## evaluate convergence
    err&lt;-max(abs(p.new-p)/p)

    ## evaluate whether the log.likelihood increases
    loglike.new&lt;-(sum(tempmax+log(tempsum))+sum(log(p.new)))/xrow

    loglike.old&lt;-loglike.new
    p&lt;-p.new

    if(err&lt;tol) {
      break;
    }
  }

  ## compute posterior p
  clustlike&lt;-matrix(0,xrow,K)
  for(j in 1:K) {
    clustlike[,j]&lt;-log(p[j])+L[,j]
  }

  tempmax&lt;-apply(clustlike,1,max)
  for(j in 1:K) {
    clustlike[,j]&lt;-exp(clustlike[,j]-tempmax)
  }
  tempsum&lt;-apply(clustlike,1,sum)

  for(j in 1:K) {
    clustlike[,j]&lt;-clustlike[,j]/tempsum
  }

  p.post&lt;-matrix(0,xrow,xcol)
  for(j in 1:K) {
    for(i in 1:xcol) {
      if(patid[j,i] &gt; 0.5) {
        p.post[,i]&lt;-p.post[,i]+clustlike[,j]
      }
    }
  }

  ## return
  #calculate back loglikelihood
  loglike.old&lt;-loglike.old-sum(log(p))/xrow
  loglike.old&lt;-loglike.old*xrow
  result&lt;-list(p.post=p.post, motif.prior=p, loglike=loglike.old)
}

generatetype&lt;-function(limfitted)
{
  jtype&lt;-list()
  df&lt;-limfitted$g1num+limfitted$g2num-2+limfitted$df0
  for(j in 1:limfitted$compnum)
  {
    jtype[[j]]&lt;-list(f0=modt.f0.loglike, f0.param=df[j], f1=modt.f1.loglike, f1.param=c(df[j],limfitted$g1num[j],limfitted$g2num[j],limfitted$v0[j]))
  }
  jtype
}

cormotiffit &lt;- function(exprs, groupid=NULL, compid=NULL, K=1, tol=1e-3,
                        max.iter=100, BIC=TRUE, norm.factor.method=&quot;TMM&quot;,
                        voom.normalize.method = &quot;none&quot;, runtype=c(&quot;logCPM&quot;,&quot;counts&quot;,&quot;limmafits&quot;), each=3)
{
  # first I want to do some typechecking. Input can be either a normalized
  # matrix, a count matrix, or a list of limma fits. Dispatch the correct
  # limmafit accordingly.
  # todo: add some typechecking here
  limfitted &lt;- list()
  if (runtype==&quot;counts&quot;) {
    limfitted &lt;- limmafit.counts(exprs,groupid,compid, norm.factor.method, voom.normalize.method)
  } else if (runtype==&quot;logCPM&quot;) {
    limfitted &lt;- limmafit.default(exprs,groupid,compid)
  } else if (runtype==&quot;limmafits&quot;) {
    limfitted &lt;- limmafit.list(exprs)
  } else {
    stop(&quot;runtype must be one of &#39;logCPM&#39;, &#39;counts&#39;, or &#39;limmafits&#39;&quot;)
  }


  jtype&lt;-generatetype(limfitted)
  fitresult&lt;-list()
  ks &lt;- rep(K, each = each)
  fitresult &lt;- bplapply(1:length(ks), function(i, x, type, ks, tol, max.iter) {
    cmfit.X(x, type, K = ks[i], tol = tol, max.iter = max.iter)
  }, x=limfitted$t, type=jtype, ks=ks, tol=tol, max.iter=max.iter)

  best.fitresults &lt;- list()
  for (i in 1:length(K)) {
    w.k &lt;- which(ks==K[i])
    this.bic &lt;- c()
    for (j in w.k) this.bic[j] &lt;- -2 * fitresult[[j]]$loglike + (K[i] - 1 + K[i] * limfitted$compnum) * log(dim(limfitted$t)[1])
    w.min &lt;- which(this.bic == min(this.bic, na.rm = TRUE))[1]
    best.fitresults[[i]] &lt;- fitresult[[w.min]]
  }
  fitresult &lt;- best.fitresults

  bic &lt;- rep(0, length(K))
  aic &lt;- rep(0, length(K))
  loglike &lt;- rep(0, length(K))
  for (i in 1:length(K)) loglike[i] &lt;- fitresult[[i]]$loglike
  for (i in 1:length(K)) bic[i] &lt;- -2 * fitresult[[i]]$loglike + (K[i] - 1 + K[i] * limfitted$compnum) * log(dim(limfitted$t)[1])
  for (i in 1:length(K)) aic[i] &lt;- -2 * fitresult[[i]]$loglike + 2 * (K[i] - 1 + K[i] * limfitted$compnum)
  if(BIC==TRUE) {
    bestflag=which(bic==min(bic))
  }
  else {
    bestflag=which(aic==min(aic))
  }
  result&lt;-list(bestmotif=fitresult[[bestflag]],bic=cbind(K,bic),
               aic=cbind(K,aic),loglike=cbind(K,loglike), allmotifs=fitresult)

}

cormotiffitall&lt;-function(exprs,groupid,compid, tol=1e-3, max.iter=100)
{
  limfitted&lt;-limmafit(exprs,groupid,compid)
  jtype&lt;-generatetype(limfitted)
  fitresult&lt;-cmfitall(limfitted$t,type=jtype,tol=1e-3,max.iter=max.iter)
}

cormotiffitsep&lt;-function(exprs,groupid,compid, tol=1e-3, max.iter=100)
{
  limfitted&lt;-limmafit(exprs,groupid,compid)
  jtype&lt;-generatetype(limfitted)
  fitresult&lt;-cmfitsep(limfitted$t,type=jtype,tol=1e-3,max.iter=max.iter)
}

cormotiffitfull&lt;-function(exprs,groupid,compid, tol=1e-3, max.iter=100)
{
  limfitted&lt;-limmafit(exprs,groupid,compid)
  jtype&lt;-generatetype(limfitted)
  fitresult&lt;-cmfitfull(limfitted$t,type=jtype,tol=1e-3,max.iter=max.iter)
}

plotIC&lt;-function(fitted_cormotif)
{
  oldpar&lt;-par(mfrow=c(1,2))
  plot(fitted_cormotif$bic[,1], fitted_cormotif$bic[,2], type=&quot;b&quot;,xlab=&quot;Motif Number&quot;, ylab=&quot;BIC&quot;, main=&quot;BIC&quot;)
  plot(fitted_cormotif$aic[,1], fitted_cormotif$aic[,2], type=&quot;b&quot;,xlab=&quot;Motif Number&quot;, ylab=&quot;AIC&quot;, main=&quot;AIC&quot;)
}

plotMotif&lt;-function(fitted_cormotif,title=&quot;&quot;)
{
  layout(matrix(1:2,ncol=2))
  u&lt;-1:dim(fitted_cormotif$bestmotif$motif.q)[2]
  v&lt;-1:dim(fitted_cormotif$bestmotif$motif.q)[1]
  image(u,v,t(fitted_cormotif$bestmotif$motif.q),
        col=gray(seq(from=1,to=0,by=-0.1)),xlab=&quot;Study&quot;,yaxt = &quot;n&quot;,
        ylab=&quot;Corr. Motifs&quot;,main=paste(title,&quot;pattern&quot;,sep=&quot; &quot;))
  axis(2,at=1:length(v))
  for(i in 1:(length(u)+1))
  {
    abline(v=(i-0.5))
  }
  for(i in 1:(length(v)+1))
  {
    abline(h=(i-0.5))
  }
  Ng=10000
  if(is.null(fitted_cormotif$bestmotif$p.post)!=TRUE)
    Ng=nrow(fitted_cormotif$bestmotif$p.post)
  genecount=floor(fitted_cormotif$bestmotif$motif.p*Ng)
  NK=nrow(fitted_cormotif$bestmotif$motif.q)
  plot(0,0.7,pch=&quot;.&quot;,xlim=c(0,1.2),ylim=c(0.75,NK+0.25),
       frame.plot=FALSE,axes=FALSE,xlab=&quot;No. of genes&quot;,ylab=&quot;&quot;, main=paste(title,&quot;frequency&quot;,sep=&quot; &quot;))
  segments(0,0.7,fitted_cormotif$bestmotif$motif.p[1],0.7)
  rect(0,1:NK-0.3,fitted_cormotif$bestmotif$motif.p,1:NK+0.3,
       col=&quot;dark grey&quot;)
  mtext(1:NK,at=1:NK,side=2,cex=0.8)
  text(fitted_cormotif$bestmotif$motif.p+0.15,1:NK,
       labels=floor(fitted_cormotif$bestmotif$motif.p*Ng))
}</code></pre>
</div>
<div id="load-required-libraries" class="section level2">
<h2><strong>ðŸ“Œ Load Required Libraries</strong></h2>
<pre class="r"><code>library(tidyverse)
library(ggplot2)
library(tidyr)
library(dplyr)
library(reshape2)
library(clusterProfiler)
library(AnnotationDbi)
library(org.Hs.eg.db)
library(RColorBrewer)
library(gprofiler2)
library(pheatmap)
library(ggpubr)
library(corrplot)
library(Cormotif)
library(Rfast)
library(BiocParallel)</code></pre>
</div>
<div id="load-corrmotif-data" class="section level2">
<h2><strong>ðŸ“Œ Load Corrmotif Data</strong></h2>
<pre class="r"><code># Read the Corrmotif Results
Corrmotif &lt;- read.csv(&quot;data/Corrmotif/CX5461.csv&quot;)
Corrmotif_df &lt;- data.frame(Corrmotif)

rownames(Corrmotif_df) &lt;- Corrmotif_df$Gene
exprs.corrmotif &lt;- as.matrix(Corrmotif_df[,2:109])

# Read group and comparison IDs
groupid_df &lt;- read.csv(&quot;data/Corrmotif/groupid.csv&quot;)
compid_df &lt;- read.csv(&quot;data/Corrmotif/Compid.csv&quot;)</code></pre>
</div>
<div id="fit-corrmotif-model-k18" class="section level2">
<h2><strong>ðŸ“Œ Fit Corrmotif Model (K=1:8)</strong></h2>
<pre class="r"><code>set.seed(11111)
# Fit Corrmotif Model (K = 1 to 8)
set.seed(11111)
motif.fitted &lt;- cormotiffit(
  exprs = exprs.corrmotif,
  groupid = groupid_df,
  compid = compid_df,
  K = 1:8,
  max.iter = 1000,
  BIC = TRUE,
  runtype = &quot;logCPM&quot;
)

gene_prob_all &lt;- motif.fitted$bestmotif$p.post
rownames(gene_prob_all) &lt;- rownames(Corrmotif_df)
motif_prob &lt;- motif.fitted$bestmotif$clustlike
rownames(motif_prob) &lt;- rownames(gene_prob_all)
write.csv(motif_prob,&quot;data/cormotif_probability_genelist_all.csv&quot;)</code></pre>
</div>
<div id="plot-motif" class="section level2">
<h2><strong>ðŸ“Œ Plot motif</strong></h2>
<pre class="r"><code>cormotif_all &lt;- readRDS(&quot;data/Corrmotif/cormotif_all.RDS&quot;)
cormotif_all$bic</code></pre>
<pre><code>     K      bic
[1,] 1 643837.2
[2,] 2 632515.2
[3,] 3 626572.5
[4,] 4 624577.2
[5,] 5 624656.6
[6,] 6 624667.6
[7,] 7 624660.0
[8,] 8 624786.5</code></pre>
<pre class="r"><code>plotIC(cormotif_all)</code></pre>
<p><img src="figure/Corrmotif_all.Rmd/Motif-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Motif-1">
Past versions of Motif-1.png
</button>
</p>
<div id="fig-Motif-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/b08bc389e65825d6f94786c53af77f69811dec5c/docs/figure/Corrmotif_all.Rmd/Motif-1.png" target="_blank">b08bc38</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plotMotif(cormotif_all)</code></pre>
<p><img src="figure/Corrmotif_all.Rmd/Motif-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Motif-2">
Past versions of Motif-2.png
</button>
</p>
<div id="fig-Motif-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/b08bc389e65825d6f94786c53af77f69811dec5c/docs/figure/Corrmotif_all.Rmd/Motif-2.png" target="_blank">b08bc38</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="extract-gene-probabilities" class="section level2">
<h2><strong>ðŸ“Œ Extract Gene Probabilities</strong></h2>
<pre class="r"><code># Extract posterior probabilities for genes
gene_prob_all &lt;- cormotif_all$bestmotif$p.post
rownames(gene_prob_all) &lt;- rownames(Corrmotif_df)

# Define gene probability groups
prob_all_1  &lt;- rownames(gene_prob_all[(gene_prob_all[,1] &lt;0.5 &amp; gene_prob_all[,2] &lt;0.5 &amp; gene_prob_all[,3] &lt;0.5 &amp; gene_prob_all[,4] &lt;0.5 &amp; gene_prob_all[,5] &lt; 0.5 &amp; gene_prob_all[,6]&lt;0.5 &amp; gene_prob_all[,7]&lt;0.5 &amp; gene_prob_all[,8]&lt;0.5 &amp; gene_prob_all[,9]&lt;0.5 &amp; gene_prob_all[,10]&lt;0.5 &amp; gene_prob_all[,11]&lt;0.5 &amp; gene_prob_all[,12]&lt;0.5),])

length(prob_all_1)

prob_all_2  &lt;- rownames(gene_prob_all[(gene_prob_all[,1] &lt;0.5 &amp; gene_prob_all[,2] &lt;0.5 &amp; gene_prob_all[,3] &gt;0.5 &amp; gene_prob_all[,4] &gt;0.5 &amp; gene_prob_all[,5] &gt; 0.5 &amp; gene_prob_all[,6]&gt;0.5 &amp; gene_prob_all[,7]&lt;0.5 &amp; gene_prob_all[,8]&lt;0.5 &amp; gene_prob_all[,9]&gt;0.5 &amp; gene_prob_all[,10]&gt;0.1 &amp; gene_prob_all[,11]&gt;0.5 &amp; gene_prob_all[,12]&gt;0.1),])

length(prob_all_2)

prob_all_3  &lt;- rownames(gene_prob_all[(gene_prob_all[,1] &lt;0.5 &amp; gene_prob_all[,2] &lt;0.5 &amp; gene_prob_all[,3] &lt;0.5 &amp; gene_prob_all[,4] &lt;0.5 &amp; gene_prob_all[,5] &lt; 0.5 &amp; gene_prob_all[,6]&lt;0.5 &amp; gene_prob_all[,7]&lt;0.5 &amp; gene_prob_all[,8]&gt;=0.02 &amp; gene_prob_all[,9]&gt;0.5 &amp; gene_prob_all[,10]&gt;0.5 &amp; gene_prob_all[,11]&gt;0.5 &amp; gene_prob_all[,12]&gt;0.5),])

length(prob_all_3)

prob_all_4  &lt;- rownames(gene_prob_all[(gene_prob_all[,1] &lt;0.5 &amp; gene_prob_all[,2] &lt;0.5 &amp; gene_prob_all[,3] &lt;0.5 &amp; gene_prob_all[,4] &lt;0.5 &amp; gene_prob_all[,5] &lt; 0.5 &amp; gene_prob_all[,6]&lt;0.5 &amp; gene_prob_all[,7]&lt;0.5 &amp; gene_prob_all[,8]&lt;0.5 &amp; gene_prob_all[,9]&lt;0.5 &amp; gene_prob_all[,10]&gt;0.5 &amp; gene_prob_all[,11]&lt;0.5 &amp; gene_prob_all[,12]&gt;0.5),])

length(prob_all_4)</code></pre>
</div>
<div id="distribution-of-gene-clusters-identified-by-corrmotif"
class="section level2">
<h2><strong>ðŸ“Œ Distribution of Gene Clusters Identified by
Corrmotif</strong></h2>
<pre class="r"><code># Load necessary library
library(ggplot2)

# Data
data &lt;- data.frame(
  Category = c(&quot;Non response&quot;, &quot;CX_DOX shared late response&quot;,
               &quot;Dox specific response&quot;, &quot;Late high dose DOX specific response&quot;),
  Value = c(7031, 414, 1531, 4874)
)

# Define custom colors
custom_colors &lt;- c(&quot;Non response&quot; = &quot;#FF9999&quot;,
                   &quot;CX_DOX shared late response&quot; = &quot;#66B2FF&quot;,
                   &quot;Dox specific response&quot; = &quot;#99FF99&quot;,
                   &quot;Late high dose DOX specific response&quot; = &quot;#FFD700&quot;)

# Create pie chart
ggplot(data, aes(x = &quot;&quot;, y = Value, fill = Category)) +
  geom_bar(width = 1, stat = &quot;identity&quot;) +
  coord_polar(&quot;y&quot;, start = 0) +
  geom_text(aes(label = Value),
            position = position_stack(vjust = 0.5),
            size = 4, color = &quot;black&quot;) +
  labs(title = &quot;Distribution of Gene Clusters Identified by Corrmotif&quot;, x = NULL, y = NULL) +
  theme_void() +
  scale_fill_manual(values = custom_colors)</code></pre>
<p><img src="figure/Corrmotif_all.Rmd/pie-chart-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pie-chart-1">
Past versions of pie-chart-1.png
</button>
</p>
<div id="fig-pie-chart-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/b56a62ae3daececcbdeee937b8acc8e5fac7b4ff/docs/figure/Corrmotif_all.Rmd/pie-chart-1.png" target="_blank">b56a62a</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="corr-motif-boxplots" class="section level2">
<h2><strong>ðŸ“Œ Corr-motif Boxplots</strong></h2>
</div>
<div id="non-response-mrpl20" class="section level2">
<h2><strong>ðŸ“Œ Non response (MRPL20)</strong></h2>
<pre class="r"><code># Load the dataset from the data folder
# Load the dataset from the data folder
boxplot1 &lt;- read.csv(&quot;data/boxplot1.csv&quot;, check.names = FALSE)

boxplot1 &lt;- as.data.frame(boxplot1)

# Print first few column names to check structure
print(colnames(boxplot1))</code></pre>
<pre><code>  [1] &quot;ENTREZID&quot;            &quot;SYMBOL&quot;              &quot;GENENAME&quot;           
  [4] &quot;17.3_CX.5461_0.1_3&quot;  &quot;17.3_DOX_0.5_24&quot;     &quot;87.1_DOX_0.5_24&quot;    
  [7] &quot;87.1_VEH_0.1_24&quot;     &quot;87.1_VEH_0.5_24&quot;     &quot;87.1_CX.5461_0.1_48&quot;
 [10] &quot;87.1_CX.5461_0.5_48&quot; &quot;87.1_DOX_0.1_48&quot;     &quot;87.1_DOX_0.5_48&quot;    
 [13] &quot;87.1_VEH_0.1_48&quot;     &quot;87.1_VEH_0.5_48&quot;     &quot;17.3_VEH_0.1_24&quot;    
 [16] &quot;17.3_VEH_0.5_24&quot;     &quot;17.3_CX.5461_0.1_48&quot; &quot;17.3_CX.5461_0.5_48&quot;
 [19] &quot;17.3_DOX_0.1_48&quot;     &quot;17.3_DOX_0.5_48&quot;     &quot;17.3_VEH_0.1_48&quot;    
 [22] &quot;17.3_VEH_0.5_48&quot;     &quot;84.1_CX.5461_0.1_3&quot;  &quot;17.3_CX.5461_0.5_3&quot; 
 [25] &quot;84.1_CX.5461_0.5_3&quot;  &quot;84.1_DOX_0.1_3&quot;      &quot;84.1_DOX_0.5_3&quot;     
 [28] &quot;84.1_VEH_0.1_3&quot;      &quot;84.1_VEH_0.5_3&quot;      &quot;84.1_CX.5461_0.1_24&quot;
 [31] &quot;84.1_CX.5461_0.5_24&quot; &quot;84.1_DOX_0.1_24&quot;     &quot;84.1_DOX_0.5_24&quot;    
 [34] &quot;84.1_VEH_0.1_24&quot;     &quot;17.3_DOX_0.1_3&quot;      &quot;84.1_VEH_0.5_24&quot;    
 [37] &quot;84.1_CX.5461_0.1_48&quot; &quot;84.1_CX.5461_0.5_48&quot; &quot;84.1_DOX_0.1_48&quot;    
 [40] &quot;84.1_DOX_0.5_48&quot;     &quot;84.1_VEH_0.1_48&quot;     &quot;84.1_VEH_0.5_48&quot;    
 [43] &quot;90.1_CX.5461_0.1_3&quot;  &quot;90.1_CX.5461_0.5_3&quot;  &quot;90.1_DOX_0.1_3&quot;     
 [46] &quot;17.3_DOX_0.5_3&quot;      &quot;90.1_DOX_0.5_3&quot;      &quot;90.1_VEH_0.1_3&quot;     
 [49] &quot;90.1_VEH_0.5_3&quot;      &quot;90.1_CX.5461_0.1_24&quot; &quot;90.1_CX.5461_0.5_24&quot;
 [52] &quot;90.1_DOX_0.1_24&quot;     &quot;90.1_DOX_0.5_24&quot;     &quot;90.1_VEH_0.1_24&quot;    
 [55] &quot;90.1_VEH_0.5_24&quot;     &quot;90.1_CX.5461_0.1_48&quot; &quot;17.3_VEH_0.1_3&quot;     
 [58] &quot;90.1_CX.5461_0.5_48&quot; &quot;90.1_DOX_0.1_48&quot;     &quot;90.1_DOX_0.5_48&quot;    
 [61] &quot;90.1_VEH_0.1_48&quot;     &quot;90.1_VEH_0.5_48&quot;     &quot;75.1_CX.5461_0.1_3&quot; 
 [64] &quot;75.1_CX.5461_0.5_3&quot;  &quot;75.1_DOX_0.1_3&quot;      &quot;75.1_DOX_0.5_3&quot;     
 [67] &quot;75.1_VEH_0.1_3&quot;      &quot;17.3_VEH_0.5_3&quot;      &quot;75.1_VEH_0.5_3&quot;     
 [70] &quot;75.1_CX.5461_0.1_24&quot; &quot;75.1_CX.5461_0.5_24&quot; &quot;75.1_DOX_0.1_24&quot;    
 [73] &quot;75.1_DOX_0.5_24&quot;     &quot;75.1_VEH_0.1_24&quot;     &quot;75.1_VEH_0.5_24&quot;    
 [76] &quot;75.1_CX.5461_0.1_48&quot; &quot;75.1_CX.5461_0.5_48&quot; &quot;75.1_DOX_0.1_48&quot;    
 [79] &quot;17.3_CX.5461_0.1_24&quot; &quot;75.1_DOX_0.5_48&quot;     &quot;75.1_VEH_0.1_48&quot;    
 [82] &quot;75.1_VEH_0.5_48&quot;     &quot;78.1_CX.5461_0.1_3&quot;  &quot;78.1_CX.5461_0.5_3&quot; 
 [85] &quot;78.1_DOX_0.1_3&quot;      &quot;78.1_DOX_0.5_3&quot;      &quot;78.1_VEH_0.1_3&quot;     
 [88] &quot;78.1_VEH_0.5_3&quot;      &quot;78.1_CX.5461_0.1_24&quot; &quot;17.3_CX.5461_0.5_24&quot;
 [91] &quot;78.1_CX.5461_0.5_24&quot; &quot;78.1_DOX_0.1_24&quot;     &quot;78.1_DOX_0.5_24&quot;    
 [94] &quot;78.1_VEH_0.1_24&quot;     &quot;78.1_VEH_0.5_24&quot;     &quot;78.1_CX.5461_0.1_48&quot;
 [97] &quot;78.1_CX.5461_0.5_48&quot; &quot;78.1_DOX_0.1_48&quot;     &quot;78.1_DOX_0.5_48&quot;    
[100] &quot;78.1_VEH_0.1_48&quot;     &quot;17.3_DOX_0.1_24&quot;     &quot;78.1_VEH_0.5_48&quot;    
[103] &quot;87.1_CX.5461_0.1_3&quot;  &quot;87.1_CX.5461_0.5_3&quot;  &quot;87.1_DOX_0.1_3&quot;     
[106] &quot;87.1_DOX_0.5_3&quot;      &quot;87.1_VEH_0.1_3&quot;      &quot;87.1_VEH_0.5_3&quot;     
[109] &quot;87.1_CX.5461_0.1_24&quot; &quot;87.1_CX.5461_0.5_24&quot; &quot;87.1_DOX_0.1_24&quot;    </code></pre>
<pre class="r"><code># Print first few rows to check format
print(head(boxplot1[, 1:10]))</code></pre>
<pre><code>   ENTREZID       SYMBOL                                 GENENAME
1    653635       WASH7P        WASP family homolog 7, pseudogene
2    729737    LOC729737                uncharacterized LOC729737
3 102723897 LOC102723897 WAS protein family homolog 9, pseudogene
4 100132287 LOC100132287             uncharacterized LOC100132287
5 102465432         &lt;NA&gt;                                     &lt;NA&gt;
6 100133331         &lt;NA&gt;                                     &lt;NA&gt;
  17.3_CX.5461_0.1_3 17.3_DOX_0.5_24 87.1_DOX_0.5_24 87.1_VEH_0.1_24
1           3.499737        3.777758        4.051060        3.631547
2           4.018739        3.087263        2.799033        3.631547
3           3.581464        3.765087        4.210151        3.707275
4           3.229725        2.970710        1.422530        2.309608
5           6.910839        6.896427        5.596078        5.581451
6           3.044515        2.902929        1.971919        2.674101
  87.1_VEH_0.5_24 87.1_CX.5461_0.1_48 87.1_CX.5461_0.5_48
1        3.758155            3.982758            3.621628
2        4.399176            4.292059            3.859895
3        3.679166            4.047012            3.830959
4        2.553616            2.748309            2.223026
5        5.117157            5.156085            4.996110
6        2.861516            2.903876            2.576789</code></pre>
<pre class="r"><code># Filter data for the gene with Entrez ID: 55052
gene_data &lt;- boxplot1[boxplot1$ENTREZID == 55052, ]

# Check if gene_data is empty
if(nrow(gene_data) == 0) {
  stop(&quot;No data found for the selected gene ENTIREZID 55052.&quot;)
}

# Reshape data from wide to long format
gene_data_long &lt;- melt(gene_data,
                       id.vars = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;, &quot;GENENAME&quot;),
                       variable.name = &quot;Sample&quot;,
                       value.name = &quot;log2CPM&quot;)

# Print some sample names for debugging
print(unique(gene_data_long$Sample))</code></pre>
<pre><code>  [1] 17.3_CX.5461_0.1_3  17.3_DOX_0.5_24     87.1_DOX_0.5_24    
  [4] 87.1_VEH_0.1_24     87.1_VEH_0.5_24     87.1_CX.5461_0.1_48
  [7] 87.1_CX.5461_0.5_48 87.1_DOX_0.1_48     87.1_DOX_0.5_48    
 [10] 87.1_VEH_0.1_48     87.1_VEH_0.5_48     17.3_VEH_0.1_24    
 [13] 17.3_VEH_0.5_24     17.3_CX.5461_0.1_48 17.3_CX.5461_0.5_48
 [16] 17.3_DOX_0.1_48     17.3_DOX_0.5_48     17.3_VEH_0.1_48    
 [19] 17.3_VEH_0.5_48     84.1_CX.5461_0.1_3  17.3_CX.5461_0.5_3 
 [22] 84.1_CX.5461_0.5_3  84.1_DOX_0.1_3      84.1_DOX_0.5_3     
 [25] 84.1_VEH_0.1_3      84.1_VEH_0.5_3      84.1_CX.5461_0.1_24
 [28] 84.1_CX.5461_0.5_24 84.1_DOX_0.1_24     84.1_DOX_0.5_24    
 [31] 84.1_VEH_0.1_24     17.3_DOX_0.1_3      84.1_VEH_0.5_24    
 [34] 84.1_CX.5461_0.1_48 84.1_CX.5461_0.5_48 84.1_DOX_0.1_48    
 [37] 84.1_DOX_0.5_48     84.1_VEH_0.1_48     84.1_VEH_0.5_48    
 [40] 90.1_CX.5461_0.1_3  90.1_CX.5461_0.5_3  90.1_DOX_0.1_3     
 [43] 17.3_DOX_0.5_3      90.1_DOX_0.5_3      90.1_VEH_0.1_3     
 [46] 90.1_VEH_0.5_3      90.1_CX.5461_0.1_24 90.1_CX.5461_0.5_24
 [49] 90.1_DOX_0.1_24     90.1_DOX_0.5_24     90.1_VEH_0.1_24    
 [52] 90.1_VEH_0.5_24     90.1_CX.5461_0.1_48 17.3_VEH_0.1_3     
 [55] 90.1_CX.5461_0.5_48 90.1_DOX_0.1_48     90.1_DOX_0.5_48    
 [58] 90.1_VEH_0.1_48     90.1_VEH_0.5_48     75.1_CX.5461_0.1_3 
 [61] 75.1_CX.5461_0.5_3  75.1_DOX_0.1_3      75.1_DOX_0.5_3     
 [64] 75.1_VEH_0.1_3      17.3_VEH_0.5_3      75.1_VEH_0.5_3     
 [67] 75.1_CX.5461_0.1_24 75.1_CX.5461_0.5_24 75.1_DOX_0.1_24    
 [70] 75.1_DOX_0.5_24     75.1_VEH_0.1_24     75.1_VEH_0.5_24    
 [73] 75.1_CX.5461_0.1_48 75.1_CX.5461_0.5_48 75.1_DOX_0.1_48    
 [76] 17.3_CX.5461_0.1_24 75.1_DOX_0.5_48     75.1_VEH_0.1_48    
 [79] 75.1_VEH_0.5_48     78.1_CX.5461_0.1_3  78.1_CX.5461_0.5_3 
 [82] 78.1_DOX_0.1_3      78.1_DOX_0.5_3      78.1_VEH_0.1_3     
 [85] 78.1_VEH_0.5_3      78.1_CX.5461_0.1_24 17.3_CX.5461_0.5_24
 [88] 78.1_CX.5461_0.5_24 78.1_DOX_0.1_24     78.1_DOX_0.5_24    
 [91] 78.1_VEH_0.1_24     78.1_VEH_0.5_24     78.1_CX.5461_0.1_48
 [94] 78.1_CX.5461_0.5_48 78.1_DOX_0.1_48     78.1_DOX_0.5_48    
 [97] 78.1_VEH_0.1_48     17.3_DOX_0.1_24     78.1_VEH_0.5_48    
[100] 87.1_CX.5461_0.1_3  87.1_CX.5461_0.5_3  87.1_DOX_0.1_3     
[103] 87.1_DOX_0.5_3      87.1_VEH_0.1_3      87.1_VEH_0.5_3     
[106] 87.1_CX.5461_0.1_24 87.1_CX.5461_0.5_24 87.1_DOX_0.1_24    
108 Levels: 17.3_CX.5461_0.1_3 17.3_DOX_0.5_24 ... 87.1_DOX_0.1_24</code></pre>
<pre class="r"><code># Extract details from sample names
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Time = sub(&quot;.*_(\\d+)$&quot;, &quot;\\1&quot;, Sample),
    Concentration = sub(&quot;.*_(0\\.\\d)_\\d+$&quot;, &quot;\\1&quot;, Sample),
    Drug = sub(&quot;.*_(CX\\.5461|DOX|VEH)_.*&quot;, &quot;\\1&quot;, Sample),
    Indv = sub(&quot;^([0-9]+\\.[0-9]+)_.*&quot;, &quot;\\1&quot;, Sample)  # Ensure correct pattern extraction
  )

# Print extracted `Indv` column for debugging
print(unique(gene_data_long$Indv))</code></pre>
<pre><code>[1] &quot;17.3&quot; &quot;87.1&quot; &quot;84.1&quot; &quot;90.1&quot; &quot;75.1&quot; &quot;78.1&quot;</code></pre>
<pre class="r"><code># Convert extracted columns to factors
gene_data_long$Time &lt;- factor(gene_data_long$Time, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;))
gene_data_long$Concentration &lt;- factor(gene_data_long$Concentration, levels = c(&quot;0.1&quot;, &quot;0.5&quot;))

# Ensure each concentration facet only shows relevant data
gene_data_long &lt;- gene_data_long %&gt;%
  group_by(Concentration) %&gt;%
  filter(all(Concentration == &quot;0.1&quot;) | all(Concentration == &quot;0.5&quot;)) %&gt;%
  ungroup()

# Map individual IDs
indv_mapping &lt;- c(&quot;75.1&quot; = &quot;1&quot;, &quot;78.1&quot; = &quot;2&quot;, &quot;87.1&quot; = &quot;3&quot;, &quot;17.3&quot; = &quot;4&quot;, &quot;84.1&quot; = &quot;5&quot;, &quot;90.1&quot; = &quot;6&quot;)

# Ensure `Indv` values match the keys in `indv_mapping`
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], NA)
  )

# Print `Indv` column again to confirm mapping worked
print(unique(gene_data_long$Indv))</code></pre>
<pre><code>[1] &quot;4&quot; &quot;3&quot; &quot;5&quot; &quot;6&quot; &quot;1&quot; &quot;2&quot;</code></pre>
<pre class="r"><code># Check for missing values and handle them
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(Indv = ifelse(is.na(Indv), &quot;Unknown&quot;, Indv))  # Assign &quot;Unknown&quot; to unidentified individuals

# Define drug colors
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;, &quot;VEH&quot; = &quot;green&quot;)

# Extract the gene symbol for labeling
gene_symbol &lt;- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(Concentration ~ Time, labeller = label_both) +
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle(&quot;Non response&quot;) +
  labs(
    title = &quot;Non response&quot;,
    x = &quot;Drugs&quot;,
    y = paste(gene_symbol, &quot; log2CPM&quot;)
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_all.Rmd/box1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box1-1">
Past versions of box1-1.png
</button>
</p>
<div id="fig-box1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/e73d731caeb1fe0eb8f95488b5bcbd2d6049b253/docs/figure/Corrmotif_all.Rmd/box1-1.png" target="_blank">e73d731</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="non-response-faap20" class="section level2">
<h2><strong>ðŸ“Œ Non response (FAAP20)</strong></h2>
<pre class="r"><code># Load the dataset from the data folder
# Load the dataset from the data folder

# Filter data for the gene with Entrez ID: 199990
gene_data &lt;- boxplot1[boxplot1$ENTREZID == 199990, ]

# Check if gene_data is empty
if(nrow(gene_data) == 0) {
  stop(&quot;No data found for the selected gene ENTIREZID 199990.&quot;)
}

# Reshape data from wide to long format
gene_data_long &lt;- melt(gene_data,
                       id.vars = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;, &quot;GENENAME&quot;),
                       variable.name = &quot;Sample&quot;,
                       value.name = &quot;log2CPM&quot;)

# Print some sample names for debugging
print(unique(gene_data_long$Sample))</code></pre>
<pre><code>  [1] 17.3_CX.5461_0.1_3  17.3_DOX_0.5_24     87.1_DOX_0.5_24    
  [4] 87.1_VEH_0.1_24     87.1_VEH_0.5_24     87.1_CX.5461_0.1_48
  [7] 87.1_CX.5461_0.5_48 87.1_DOX_0.1_48     87.1_DOX_0.5_48    
 [10] 87.1_VEH_0.1_48     87.1_VEH_0.5_48     17.3_VEH_0.1_24    
 [13] 17.3_VEH_0.5_24     17.3_CX.5461_0.1_48 17.3_CX.5461_0.5_48
 [16] 17.3_DOX_0.1_48     17.3_DOX_0.5_48     17.3_VEH_0.1_48    
 [19] 17.3_VEH_0.5_48     84.1_CX.5461_0.1_3  17.3_CX.5461_0.5_3 
 [22] 84.1_CX.5461_0.5_3  84.1_DOX_0.1_3      84.1_DOX_0.5_3     
 [25] 84.1_VEH_0.1_3      84.1_VEH_0.5_3      84.1_CX.5461_0.1_24
 [28] 84.1_CX.5461_0.5_24 84.1_DOX_0.1_24     84.1_DOX_0.5_24    
 [31] 84.1_VEH_0.1_24     17.3_DOX_0.1_3      84.1_VEH_0.5_24    
 [34] 84.1_CX.5461_0.1_48 84.1_CX.5461_0.5_48 84.1_DOX_0.1_48    
 [37] 84.1_DOX_0.5_48     84.1_VEH_0.1_48     84.1_VEH_0.5_48    
 [40] 90.1_CX.5461_0.1_3  90.1_CX.5461_0.5_3  90.1_DOX_0.1_3     
 [43] 17.3_DOX_0.5_3      90.1_DOX_0.5_3      90.1_VEH_0.1_3     
 [46] 90.1_VEH_0.5_3      90.1_CX.5461_0.1_24 90.1_CX.5461_0.5_24
 [49] 90.1_DOX_0.1_24     90.1_DOX_0.5_24     90.1_VEH_0.1_24    
 [52] 90.1_VEH_0.5_24     90.1_CX.5461_0.1_48 17.3_VEH_0.1_3     
 [55] 90.1_CX.5461_0.5_48 90.1_DOX_0.1_48     90.1_DOX_0.5_48    
 [58] 90.1_VEH_0.1_48     90.1_VEH_0.5_48     75.1_CX.5461_0.1_3 
 [61] 75.1_CX.5461_0.5_3  75.1_DOX_0.1_3      75.1_DOX_0.5_3     
 [64] 75.1_VEH_0.1_3      17.3_VEH_0.5_3      75.1_VEH_0.5_3     
 [67] 75.1_CX.5461_0.1_24 75.1_CX.5461_0.5_24 75.1_DOX_0.1_24    
 [70] 75.1_DOX_0.5_24     75.1_VEH_0.1_24     75.1_VEH_0.5_24    
 [73] 75.1_CX.5461_0.1_48 75.1_CX.5461_0.5_48 75.1_DOX_0.1_48    
 [76] 17.3_CX.5461_0.1_24 75.1_DOX_0.5_48     75.1_VEH_0.1_48    
 [79] 75.1_VEH_0.5_48     78.1_CX.5461_0.1_3  78.1_CX.5461_0.5_3 
 [82] 78.1_DOX_0.1_3      78.1_DOX_0.5_3      78.1_VEH_0.1_3     
 [85] 78.1_VEH_0.5_3      78.1_CX.5461_0.1_24 17.3_CX.5461_0.5_24
 [88] 78.1_CX.5461_0.5_24 78.1_DOX_0.1_24     78.1_DOX_0.5_24    
 [91] 78.1_VEH_0.1_24     78.1_VEH_0.5_24     78.1_CX.5461_0.1_48
 [94] 78.1_CX.5461_0.5_48 78.1_DOX_0.1_48     78.1_DOX_0.5_48    
 [97] 78.1_VEH_0.1_48     17.3_DOX_0.1_24     78.1_VEH_0.5_48    
[100] 87.1_CX.5461_0.1_3  87.1_CX.5461_0.5_3  87.1_DOX_0.1_3     
[103] 87.1_DOX_0.5_3      87.1_VEH_0.1_3      87.1_VEH_0.5_3     
[106] 87.1_CX.5461_0.1_24 87.1_CX.5461_0.5_24 87.1_DOX_0.1_24    
108 Levels: 17.3_CX.5461_0.1_3 17.3_DOX_0.5_24 ... 87.1_DOX_0.1_24</code></pre>
<pre class="r"><code># Extract details from sample names
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Time = sub(&quot;.*_(\\d+)$&quot;, &quot;\\1&quot;, Sample),
    Concentration = sub(&quot;.*_(0\\.\\d)_\\d+$&quot;, &quot;\\1&quot;, Sample),
    Drug = sub(&quot;.*_(CX\\.5461|DOX|VEH)_.*&quot;, &quot;\\1&quot;, Sample),
    Indv = sub(&quot;^([0-9]+\\.[0-9]+)_.*&quot;, &quot;\\1&quot;, Sample)  # Ensure correct pattern extraction
  )

# Print extracted `Indv` column for debugging
print(unique(gene_data_long$Indv))</code></pre>
<pre><code>[1] &quot;17.3&quot; &quot;87.1&quot; &quot;84.1&quot; &quot;90.1&quot; &quot;75.1&quot; &quot;78.1&quot;</code></pre>
<pre class="r"><code># Convert extracted columns to factors
gene_data_long$Time &lt;- factor(gene_data_long$Time, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;))
gene_data_long$Concentration &lt;- factor(gene_data_long$Concentration, levels = c(&quot;0.1&quot;, &quot;0.5&quot;))

# Ensure each concentration facet only shows relevant data
gene_data_long &lt;- gene_data_long %&gt;%
  group_by(Concentration) %&gt;%
  filter(all(Concentration == &quot;0.1&quot;) | all(Concentration == &quot;0.5&quot;)) %&gt;%
  ungroup()

# Map individual IDs
indv_mapping &lt;- c(&quot;75.1&quot; = &quot;1&quot;, &quot;78.1&quot; = &quot;2&quot;, &quot;87.1&quot; = &quot;3&quot;, &quot;17.3&quot; = &quot;4&quot;, &quot;84.1&quot; = &quot;5&quot;, &quot;90.1&quot; = &quot;6&quot;)

# Ensure `Indv` values match the keys in `indv_mapping`
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], NA)
  )

# Print `Indv` column again to confirm mapping worked
print(unique(gene_data_long$Indv))</code></pre>
<pre><code>[1] &quot;4&quot; &quot;3&quot; &quot;5&quot; &quot;6&quot; &quot;1&quot; &quot;2&quot;</code></pre>
<pre class="r"><code># Check for missing values and handle them
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(Indv = ifelse(is.na(Indv), &quot;Unknown&quot;, Indv))  # Assign &quot;Unknown&quot; to unidentified individuals

# Define drug colors
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;, &quot;VEH&quot; = &quot;green&quot;)

# Extract the gene symbol for labeling
gene_symbol &lt;- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(Concentration ~ Time, labeller = label_both) +
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle(&quot;Non response&quot;) +
  labs(
    title = &quot;Non response&quot;,
    x = &quot;Drugs&quot;,
    y = paste(gene_symbol, &quot; log2CPM&quot;)
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_all.Rmd/box2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box2-1">
Past versions of box2-1.png
</button>
</p>
<div id="fig-box2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/e73d731caeb1fe0eb8f95488b5bcbd2d6049b253/docs/figure/Corrmotif_all.Rmd/box2-1.png" target="_blank">e73d731</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cx_dox-shared-late-response-mphosph9" class="section level2">
<h2><strong>ðŸ“Œ CX_DOX shared late response (MPHOSPH9)</strong></h2>
<pre class="r"><code># Load the dataset from the data folder
# Load the dataset from the data folder

# Filter data for the gene with Entrez ID: 10198
gene_data &lt;- boxplot1[boxplot1$ENTREZID == 10198, ]

# Check if gene_data is empty
if(nrow(gene_data) == 0) {
  stop(&quot;No data found for the selected gene ENTIREZID 10198.&quot;)
}

# Reshape data from wide to long format
gene_data_long &lt;- melt(gene_data,
                       id.vars = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;, &quot;GENENAME&quot;),
                       variable.name = &quot;Sample&quot;,
                       value.name = &quot;log2CPM&quot;)

# Print some sample names for debugging
print(unique(gene_data_long$Sample))</code></pre>
<pre><code>  [1] 17.3_CX.5461_0.1_3  17.3_DOX_0.5_24     87.1_DOX_0.5_24    
  [4] 87.1_VEH_0.1_24     87.1_VEH_0.5_24     87.1_CX.5461_0.1_48
  [7] 87.1_CX.5461_0.5_48 87.1_DOX_0.1_48     87.1_DOX_0.5_48    
 [10] 87.1_VEH_0.1_48     87.1_VEH_0.5_48     17.3_VEH_0.1_24    
 [13] 17.3_VEH_0.5_24     17.3_CX.5461_0.1_48 17.3_CX.5461_0.5_48
 [16] 17.3_DOX_0.1_48     17.3_DOX_0.5_48     17.3_VEH_0.1_48    
 [19] 17.3_VEH_0.5_48     84.1_CX.5461_0.1_3  17.3_CX.5461_0.5_3 
 [22] 84.1_CX.5461_0.5_3  84.1_DOX_0.1_3      84.1_DOX_0.5_3     
 [25] 84.1_VEH_0.1_3      84.1_VEH_0.5_3      84.1_CX.5461_0.1_24
 [28] 84.1_CX.5461_0.5_24 84.1_DOX_0.1_24     84.1_DOX_0.5_24    
 [31] 84.1_VEH_0.1_24     17.3_DOX_0.1_3      84.1_VEH_0.5_24    
 [34] 84.1_CX.5461_0.1_48 84.1_CX.5461_0.5_48 84.1_DOX_0.1_48    
 [37] 84.1_DOX_0.5_48     84.1_VEH_0.1_48     84.1_VEH_0.5_48    
 [40] 90.1_CX.5461_0.1_3  90.1_CX.5461_0.5_3  90.1_DOX_0.1_3     
 [43] 17.3_DOX_0.5_3      90.1_DOX_0.5_3      90.1_VEH_0.1_3     
 [46] 90.1_VEH_0.5_3      90.1_CX.5461_0.1_24 90.1_CX.5461_0.5_24
 [49] 90.1_DOX_0.1_24     90.1_DOX_0.5_24     90.1_VEH_0.1_24    
 [52] 90.1_VEH_0.5_24     90.1_CX.5461_0.1_48 17.3_VEH_0.1_3     
 [55] 90.1_CX.5461_0.5_48 90.1_DOX_0.1_48     90.1_DOX_0.5_48    
 [58] 90.1_VEH_0.1_48     90.1_VEH_0.5_48     75.1_CX.5461_0.1_3 
 [61] 75.1_CX.5461_0.5_3  75.1_DOX_0.1_3      75.1_DOX_0.5_3     
 [64] 75.1_VEH_0.1_3      17.3_VEH_0.5_3      75.1_VEH_0.5_3     
 [67] 75.1_CX.5461_0.1_24 75.1_CX.5461_0.5_24 75.1_DOX_0.1_24    
 [70] 75.1_DOX_0.5_24     75.1_VEH_0.1_24     75.1_VEH_0.5_24    
 [73] 75.1_CX.5461_0.1_48 75.1_CX.5461_0.5_48 75.1_DOX_0.1_48    
 [76] 17.3_CX.5461_0.1_24 75.1_DOX_0.5_48     75.1_VEH_0.1_48    
 [79] 75.1_VEH_0.5_48     78.1_CX.5461_0.1_3  78.1_CX.5461_0.5_3 
 [82] 78.1_DOX_0.1_3      78.1_DOX_0.5_3      78.1_VEH_0.1_3     
 [85] 78.1_VEH_0.5_3      78.1_CX.5461_0.1_24 17.3_CX.5461_0.5_24
 [88] 78.1_CX.5461_0.5_24 78.1_DOX_0.1_24     78.1_DOX_0.5_24    
 [91] 78.1_VEH_0.1_24     78.1_VEH_0.5_24     78.1_CX.5461_0.1_48
 [94] 78.1_CX.5461_0.5_48 78.1_DOX_0.1_48     78.1_DOX_0.5_48    
 [97] 78.1_VEH_0.1_48     17.3_DOX_0.1_24     78.1_VEH_0.5_48    
[100] 87.1_CX.5461_0.1_3  87.1_CX.5461_0.5_3  87.1_DOX_0.1_3     
[103] 87.1_DOX_0.5_3      87.1_VEH_0.1_3      87.1_VEH_0.5_3     
[106] 87.1_CX.5461_0.1_24 87.1_CX.5461_0.5_24 87.1_DOX_0.1_24    
108 Levels: 17.3_CX.5461_0.1_3 17.3_DOX_0.5_24 ... 87.1_DOX_0.1_24</code></pre>
<pre class="r"><code># Extract details from sample names
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Time = sub(&quot;.*_(\\d+)$&quot;, &quot;\\1&quot;, Sample),
    Concentration = sub(&quot;.*_(0\\.\\d)_\\d+$&quot;, &quot;\\1&quot;, Sample),
    Drug = sub(&quot;.*_(CX\\.5461|DOX|VEH)_.*&quot;, &quot;\\1&quot;, Sample),
    Indv = sub(&quot;^([0-9]+\\.[0-9]+)_.*&quot;, &quot;\\1&quot;, Sample)  # Ensure correct pattern extraction
  )

# Print extracted `Indv` column for debugging
print(unique(gene_data_long$Indv))</code></pre>
<pre><code>[1] &quot;17.3&quot; &quot;87.1&quot; &quot;84.1&quot; &quot;90.1&quot; &quot;75.1&quot; &quot;78.1&quot;</code></pre>
<pre class="r"><code># Convert extracted columns to factors
gene_data_long$Time &lt;- factor(gene_data_long$Time, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;))
gene_data_long$Concentration &lt;- factor(gene_data_long$Concentration, levels = c(&quot;0.1&quot;, &quot;0.5&quot;))

# Ensure each concentration facet only shows relevant data
gene_data_long &lt;- gene_data_long %&gt;%
  group_by(Concentration) %&gt;%
  filter(all(Concentration == &quot;0.1&quot;) | all(Concentration == &quot;0.5&quot;)) %&gt;%
  ungroup()

# Map individual IDs
indv_mapping &lt;- c(&quot;75.1&quot; = &quot;1&quot;, &quot;78.1&quot; = &quot;2&quot;, &quot;87.1&quot; = &quot;3&quot;, &quot;17.3&quot; = &quot;4&quot;, &quot;84.1&quot; = &quot;5&quot;, &quot;90.1&quot; = &quot;6&quot;)

# Ensure `Indv` values match the keys in `indv_mapping`
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], NA)
  )

# Print `Indv` column again to confirm mapping worked
print(unique(gene_data_long$Indv))</code></pre>
<pre><code>[1] &quot;4&quot; &quot;3&quot; &quot;5&quot; &quot;6&quot; &quot;1&quot; &quot;2&quot;</code></pre>
<pre class="r"><code># Check for missing values and handle them
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(Indv = ifelse(is.na(Indv), &quot;Unknown&quot;, Indv))  # Assign &quot;Unknown&quot; to unidentified individuals

# Define drug colors
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;, &quot;VEH&quot; = &quot;green&quot;)

# Extract the gene symbol for labeling
gene_symbol &lt;- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(Concentration ~ Time, labeller = label_both) +
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle(&quot;CX_DOX shared late response&quot;) +
  labs(
    title = &quot;CX_DOX shared late response&quot;,
    x = &quot;Drugs&quot;,
    y = paste(gene_symbol, &quot; log2CPM&quot;)
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_all.Rmd/box3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box3-1">
Past versions of box3-1.png
</button>
</p>
<div id="fig-box3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/b55709531c179d5260046632aa517bd5a94898ad/docs/figure/Corrmotif_all.Rmd/box3-1.png" target="_blank">b557095</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/e73d731caeb1fe0eb8f95488b5bcbd2d6049b253/docs/figure/Corrmotif_all.Rmd/box3-1.png" target="_blank">e73d731</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cx_dox-shared-late-response-svip" class="section level2">
<h2><strong>ðŸ“Œ CX_DOX shared late response (SVIP)</strong></h2>
<pre class="r"><code># Load the dataset from the data folder
# Load the dataset from the data folder

# Filter data for the gene with Entrez ID: 258010
gene_data &lt;- boxplot1[boxplot1$ENTREZID == 258010, ]

# Check if gene_data is empty
if(nrow(gene_data) == 0) {
  stop(&quot;No data found for the selected gene ENTIREZID 258010.&quot;)
}

# Reshape data from wide to long format
gene_data_long &lt;- melt(gene_data,
                       id.vars = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;, &quot;GENENAME&quot;),
                       variable.name = &quot;Sample&quot;,
                       value.name = &quot;log2CPM&quot;)

# Print some sample names for debugging
print(unique(gene_data_long$Sample))</code></pre>
<pre><code>  [1] 17.3_CX.5461_0.1_3  17.3_DOX_0.5_24     87.1_DOX_0.5_24    
  [4] 87.1_VEH_0.1_24     87.1_VEH_0.5_24     87.1_CX.5461_0.1_48
  [7] 87.1_CX.5461_0.5_48 87.1_DOX_0.1_48     87.1_DOX_0.5_48    
 [10] 87.1_VEH_0.1_48     87.1_VEH_0.5_48     17.3_VEH_0.1_24    
 [13] 17.3_VEH_0.5_24     17.3_CX.5461_0.1_48 17.3_CX.5461_0.5_48
 [16] 17.3_DOX_0.1_48     17.3_DOX_0.5_48     17.3_VEH_0.1_48    
 [19] 17.3_VEH_0.5_48     84.1_CX.5461_0.1_3  17.3_CX.5461_0.5_3 
 [22] 84.1_CX.5461_0.5_3  84.1_DOX_0.1_3      84.1_DOX_0.5_3     
 [25] 84.1_VEH_0.1_3      84.1_VEH_0.5_3      84.1_CX.5461_0.1_24
 [28] 84.1_CX.5461_0.5_24 84.1_DOX_0.1_24     84.1_DOX_0.5_24    
 [31] 84.1_VEH_0.1_24     17.3_DOX_0.1_3      84.1_VEH_0.5_24    
 [34] 84.1_CX.5461_0.1_48 84.1_CX.5461_0.5_48 84.1_DOX_0.1_48    
 [37] 84.1_DOX_0.5_48     84.1_VEH_0.1_48     84.1_VEH_0.5_48    
 [40] 90.1_CX.5461_0.1_3  90.1_CX.5461_0.5_3  90.1_DOX_0.1_3     
 [43] 17.3_DOX_0.5_3      90.1_DOX_0.5_3      90.1_VEH_0.1_3     
 [46] 90.1_VEH_0.5_3      90.1_CX.5461_0.1_24 90.1_CX.5461_0.5_24
 [49] 90.1_DOX_0.1_24     90.1_DOX_0.5_24     90.1_VEH_0.1_24    
 [52] 90.1_VEH_0.5_24     90.1_CX.5461_0.1_48 17.3_VEH_0.1_3     
 [55] 90.1_CX.5461_0.5_48 90.1_DOX_0.1_48     90.1_DOX_0.5_48    
 [58] 90.1_VEH_0.1_48     90.1_VEH_0.5_48     75.1_CX.5461_0.1_3 
 [61] 75.1_CX.5461_0.5_3  75.1_DOX_0.1_3      75.1_DOX_0.5_3     
 [64] 75.1_VEH_0.1_3      17.3_VEH_0.5_3      75.1_VEH_0.5_3     
 [67] 75.1_CX.5461_0.1_24 75.1_CX.5461_0.5_24 75.1_DOX_0.1_24    
 [70] 75.1_DOX_0.5_24     75.1_VEH_0.1_24     75.1_VEH_0.5_24    
 [73] 75.1_CX.5461_0.1_48 75.1_CX.5461_0.5_48 75.1_DOX_0.1_48    
 [76] 17.3_CX.5461_0.1_24 75.1_DOX_0.5_48     75.1_VEH_0.1_48    
 [79] 75.1_VEH_0.5_48     78.1_CX.5461_0.1_3  78.1_CX.5461_0.5_3 
 [82] 78.1_DOX_0.1_3      78.1_DOX_0.5_3      78.1_VEH_0.1_3     
 [85] 78.1_VEH_0.5_3      78.1_CX.5461_0.1_24 17.3_CX.5461_0.5_24
 [88] 78.1_CX.5461_0.5_24 78.1_DOX_0.1_24     78.1_DOX_0.5_24    
 [91] 78.1_VEH_0.1_24     78.1_VEH_0.5_24     78.1_CX.5461_0.1_48
 [94] 78.1_CX.5461_0.5_48 78.1_DOX_0.1_48     78.1_DOX_0.5_48    
 [97] 78.1_VEH_0.1_48     17.3_DOX_0.1_24     78.1_VEH_0.5_48    
[100] 87.1_CX.5461_0.1_3  87.1_CX.5461_0.5_3  87.1_DOX_0.1_3     
[103] 87.1_DOX_0.5_3      87.1_VEH_0.1_3      87.1_VEH_0.5_3     
[106] 87.1_CX.5461_0.1_24 87.1_CX.5461_0.5_24 87.1_DOX_0.1_24    
108 Levels: 17.3_CX.5461_0.1_3 17.3_DOX_0.5_24 ... 87.1_DOX_0.1_24</code></pre>
<pre class="r"><code># Extract details from sample names
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Time = sub(&quot;.*_(\\d+)$&quot;, &quot;\\1&quot;, Sample),
    Concentration = sub(&quot;.*_(0\\.\\d)_\\d+$&quot;, &quot;\\1&quot;, Sample),
    Drug = sub(&quot;.*_(CX\\.5461|DOX|VEH)_.*&quot;, &quot;\\1&quot;, Sample),
    Indv = sub(&quot;^([0-9]+\\.[0-9]+)_.*&quot;, &quot;\\1&quot;, Sample)  # Ensure correct pattern extraction
  )

# Print extracted `Indv` column for debugging
print(unique(gene_data_long$Indv))</code></pre>
<pre><code>[1] &quot;17.3&quot; &quot;87.1&quot; &quot;84.1&quot; &quot;90.1&quot; &quot;75.1&quot; &quot;78.1&quot;</code></pre>
<pre class="r"><code># Convert extracted columns to factors
gene_data_long$Time &lt;- factor(gene_data_long$Time, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;))
gene_data_long$Concentration &lt;- factor(gene_data_long$Concentration, levels = c(&quot;0.1&quot;, &quot;0.5&quot;))

# Ensure each concentration facet only shows relevant data
gene_data_long &lt;- gene_data_long %&gt;%
  group_by(Concentration) %&gt;%
  filter(all(Concentration == &quot;0.1&quot;) | all(Concentration == &quot;0.5&quot;)) %&gt;%
  ungroup()

# Map individual IDs
indv_mapping &lt;- c(&quot;75.1&quot; = &quot;1&quot;, &quot;78.1&quot; = &quot;2&quot;, &quot;87.1&quot; = &quot;3&quot;, &quot;17.3&quot; = &quot;4&quot;, &quot;84.1&quot; = &quot;5&quot;, &quot;90.1&quot; = &quot;6&quot;)

# Ensure `Indv` values match the keys in `indv_mapping`
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], NA)
  )

# Print `Indv` column again to confirm mapping worked
print(unique(gene_data_long$Indv))</code></pre>
<pre><code>[1] &quot;4&quot; &quot;3&quot; &quot;5&quot; &quot;6&quot; &quot;1&quot; &quot;2&quot;</code></pre>
<pre class="r"><code># Check for missing values and handle them
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(Indv = ifelse(is.na(Indv), &quot;Unknown&quot;, Indv))  # Assign &quot;Unknown&quot; to unidentified individuals

# Define drug colors
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;, &quot;VEH&quot; = &quot;green&quot;)

# Extract the gene symbol for labeling
gene_symbol &lt;- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(Concentration ~ Time, labeller = label_both) +
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle(&quot;CX_DOX shared late response&quot;) +
  labs(
    title = &quot;CX_DOX shared late response&quot;,
    x = &quot;Drugs&quot;,
    y = paste(gene_symbol, &quot; log2CPM&quot;)
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_all.Rmd/box5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box5-1">
Past versions of box5-1.png
</button>
</p>
<div id="fig-box5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/ebfddcd2544a6364ccb5f8df72be2a583bc7ab4b/docs/figure/Corrmotif_all.Rmd/box5-1.png" target="_blank">ebfddcd</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dox-specific-response-espn" class="section level2">
<h2><strong>ðŸ“Œ DOX-Specific Response (ESPN)</strong></h2>
<pre class="r"><code># Load the dataset from the data folder
# Load the dataset from the data folder

# Filter data for the gene with Entrez ID:  83715
gene_data &lt;- boxplot1[boxplot1$ENTREZID ==  83715, ]

# Check if gene_data is empty
if(nrow(gene_data) == 0) {
  stop(&quot;No data found for the selected gene ENTIREZID   
83715.&quot;)
}

# Reshape data from wide to long format
gene_data_long &lt;- melt(gene_data,
                       id.vars = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;, &quot;GENENAME&quot;),
                       variable.name = &quot;Sample&quot;,
                       value.name = &quot;log2CPM&quot;)

# Print some sample names for debugging
print(unique(gene_data_long$Sample))</code></pre>
<pre><code>  [1] 17.3_CX.5461_0.1_3  17.3_DOX_0.5_24     87.1_DOX_0.5_24    
  [4] 87.1_VEH_0.1_24     87.1_VEH_0.5_24     87.1_CX.5461_0.1_48
  [7] 87.1_CX.5461_0.5_48 87.1_DOX_0.1_48     87.1_DOX_0.5_48    
 [10] 87.1_VEH_0.1_48     87.1_VEH_0.5_48     17.3_VEH_0.1_24    
 [13] 17.3_VEH_0.5_24     17.3_CX.5461_0.1_48 17.3_CX.5461_0.5_48
 [16] 17.3_DOX_0.1_48     17.3_DOX_0.5_48     17.3_VEH_0.1_48    
 [19] 17.3_VEH_0.5_48     84.1_CX.5461_0.1_3  17.3_CX.5461_0.5_3 
 [22] 84.1_CX.5461_0.5_3  84.1_DOX_0.1_3      84.1_DOX_0.5_3     
 [25] 84.1_VEH_0.1_3      84.1_VEH_0.5_3      84.1_CX.5461_0.1_24
 [28] 84.1_CX.5461_0.5_24 84.1_DOX_0.1_24     84.1_DOX_0.5_24    
 [31] 84.1_VEH_0.1_24     17.3_DOX_0.1_3      84.1_VEH_0.5_24    
 [34] 84.1_CX.5461_0.1_48 84.1_CX.5461_0.5_48 84.1_DOX_0.1_48    
 [37] 84.1_DOX_0.5_48     84.1_VEH_0.1_48     84.1_VEH_0.5_48    
 [40] 90.1_CX.5461_0.1_3  90.1_CX.5461_0.5_3  90.1_DOX_0.1_3     
 [43] 17.3_DOX_0.5_3      90.1_DOX_0.5_3      90.1_VEH_0.1_3     
 [46] 90.1_VEH_0.5_3      90.1_CX.5461_0.1_24 90.1_CX.5461_0.5_24
 [49] 90.1_DOX_0.1_24     90.1_DOX_0.5_24     90.1_VEH_0.1_24    
 [52] 90.1_VEH_0.5_24     90.1_CX.5461_0.1_48 17.3_VEH_0.1_3     
 [55] 90.1_CX.5461_0.5_48 90.1_DOX_0.1_48     90.1_DOX_0.5_48    
 [58] 90.1_VEH_0.1_48     90.1_VEH_0.5_48     75.1_CX.5461_0.1_3 
 [61] 75.1_CX.5461_0.5_3  75.1_DOX_0.1_3      75.1_DOX_0.5_3     
 [64] 75.1_VEH_0.1_3      17.3_VEH_0.5_3      75.1_VEH_0.5_3     
 [67] 75.1_CX.5461_0.1_24 75.1_CX.5461_0.5_24 75.1_DOX_0.1_24    
 [70] 75.1_DOX_0.5_24     75.1_VEH_0.1_24     75.1_VEH_0.5_24    
 [73] 75.1_CX.5461_0.1_48 75.1_CX.5461_0.5_48 75.1_DOX_0.1_48    
 [76] 17.3_CX.5461_0.1_24 75.1_DOX_0.5_48     75.1_VEH_0.1_48    
 [79] 75.1_VEH_0.5_48     78.1_CX.5461_0.1_3  78.1_CX.5461_0.5_3 
 [82] 78.1_DOX_0.1_3      78.1_DOX_0.5_3      78.1_VEH_0.1_3     
 [85] 78.1_VEH_0.5_3      78.1_CX.5461_0.1_24 17.3_CX.5461_0.5_24
 [88] 78.1_CX.5461_0.5_24 78.1_DOX_0.1_24     78.1_DOX_0.5_24    
 [91] 78.1_VEH_0.1_24     78.1_VEH_0.5_24     78.1_CX.5461_0.1_48
 [94] 78.1_CX.5461_0.5_48 78.1_DOX_0.1_48     78.1_DOX_0.5_48    
 [97] 78.1_VEH_0.1_48     17.3_DOX_0.1_24     78.1_VEH_0.5_48    
[100] 87.1_CX.5461_0.1_3  87.1_CX.5461_0.5_3  87.1_DOX_0.1_3     
[103] 87.1_DOX_0.5_3      87.1_VEH_0.1_3      87.1_VEH_0.5_3     
[106] 87.1_CX.5461_0.1_24 87.1_CX.5461_0.5_24 87.1_DOX_0.1_24    
108 Levels: 17.3_CX.5461_0.1_3 17.3_DOX_0.5_24 ... 87.1_DOX_0.1_24</code></pre>
<pre class="r"><code># Extract details from sample names
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Time = sub(&quot;.*_(\\d+)$&quot;, &quot;\\1&quot;, Sample),
    Concentration = sub(&quot;.*_(0\\.\\d)_\\d+$&quot;, &quot;\\1&quot;, Sample),
    Drug = sub(&quot;.*_(CX\\.5461|DOX|VEH)_.*&quot;, &quot;\\1&quot;, Sample),
    Indv = sub(&quot;^([0-9]+\\.[0-9]+)_.*&quot;, &quot;\\1&quot;, Sample)  # Ensure correct pattern extraction
  )

# Print extracted `Indv` column for debugging
print(unique(gene_data_long$Indv))</code></pre>
<pre><code>[1] &quot;17.3&quot; &quot;87.1&quot; &quot;84.1&quot; &quot;90.1&quot; &quot;75.1&quot; &quot;78.1&quot;</code></pre>
<pre class="r"><code># Convert extracted columns to factors
gene_data_long$Time &lt;- factor(gene_data_long$Time, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;))
gene_data_long$Concentration &lt;- factor(gene_data_long$Concentration, levels = c(&quot;0.1&quot;, &quot;0.5&quot;))

# Ensure each concentration facet only shows relevant data
gene_data_long &lt;- gene_data_long %&gt;%
  group_by(Concentration) %&gt;%
  filter(all(Concentration == &quot;0.1&quot;) | all(Concentration == &quot;0.5&quot;)) %&gt;%
  ungroup()

# Map individual IDs
indv_mapping &lt;- c(&quot;75.1&quot; = &quot;1&quot;, &quot;78.1&quot; = &quot;2&quot;, &quot;87.1&quot; = &quot;3&quot;, &quot;17.3&quot; = &quot;4&quot;, &quot;84.1&quot; = &quot;5&quot;, &quot;90.1&quot; = &quot;6&quot;)

# Ensure `Indv` values match the keys in `indv_mapping`
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], NA)
  )

# Print `Indv` column again to confirm mapping worked
print(unique(gene_data_long$Indv))</code></pre>
<pre><code>[1] &quot;4&quot; &quot;3&quot; &quot;5&quot; &quot;6&quot; &quot;1&quot; &quot;2&quot;</code></pre>
<pre class="r"><code># Check for missing values and handle them
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(Indv = ifelse(is.na(Indv), &quot;Unknown&quot;, Indv))  # Assign &quot;Unknown&quot; to unidentified individuals

# Define drug colors
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;, &quot;VEH&quot; = &quot;green&quot;)

# Extract the gene symbol for labeling
gene_symbol &lt;- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(Concentration ~ Time, labeller = label_both) +
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle(&quot;DOX-Specific Response&quot;) +
  labs(
    title = &quot;DOX-Specific Response&quot;,
    x = &quot;Drugs&quot;,
    y = paste(gene_symbol, &quot; log2CPM&quot;)
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_all.Rmd/box6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box6-1">
Past versions of box6-1.png
</button>
</p>
<div id="fig-box6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/ebfddcd2544a6364ccb5f8df72be2a583bc7ab4b/docs/figure/Corrmotif_all.Rmd/box6-1.png" target="_blank">ebfddcd</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dox-specific-response-fbxo2" class="section level2">
<h2><strong>ðŸ“Œ DOX-Specific Response (FBXO2)</strong></h2>
<pre class="r"><code># Load the dataset from the data folder
# Load the dataset from the data folder

# Filter data for the gene with Entrez ID:  26232
gene_data &lt;- boxplot1[boxplot1$ENTREZID ==  26232, ]

# Check if gene_data is empty
if(nrow(gene_data) == 0) {
  stop(&quot;No data found for the selected gene ENTIREZID   
26232.&quot;)
}

# Reshape data from wide to long format
gene_data_long &lt;- melt(gene_data,
                       id.vars = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;, &quot;GENENAME&quot;),
                       variable.name = &quot;Sample&quot;,
                       value.name = &quot;log2CPM&quot;)

# Print some sample names for debugging
print(unique(gene_data_long$Sample))</code></pre>
<pre><code>  [1] 17.3_CX.5461_0.1_3  17.3_DOX_0.5_24     87.1_DOX_0.5_24    
  [4] 87.1_VEH_0.1_24     87.1_VEH_0.5_24     87.1_CX.5461_0.1_48
  [7] 87.1_CX.5461_0.5_48 87.1_DOX_0.1_48     87.1_DOX_0.5_48    
 [10] 87.1_VEH_0.1_48     87.1_VEH_0.5_48     17.3_VEH_0.1_24    
 [13] 17.3_VEH_0.5_24     17.3_CX.5461_0.1_48 17.3_CX.5461_0.5_48
 [16] 17.3_DOX_0.1_48     17.3_DOX_0.5_48     17.3_VEH_0.1_48    
 [19] 17.3_VEH_0.5_48     84.1_CX.5461_0.1_3  17.3_CX.5461_0.5_3 
 [22] 84.1_CX.5461_0.5_3  84.1_DOX_0.1_3      84.1_DOX_0.5_3     
 [25] 84.1_VEH_0.1_3      84.1_VEH_0.5_3      84.1_CX.5461_0.1_24
 [28] 84.1_CX.5461_0.5_24 84.1_DOX_0.1_24     84.1_DOX_0.5_24    
 [31] 84.1_VEH_0.1_24     17.3_DOX_0.1_3      84.1_VEH_0.5_24    
 [34] 84.1_CX.5461_0.1_48 84.1_CX.5461_0.5_48 84.1_DOX_0.1_48    
 [37] 84.1_DOX_0.5_48     84.1_VEH_0.1_48     84.1_VEH_0.5_48    
 [40] 90.1_CX.5461_0.1_3  90.1_CX.5461_0.5_3  90.1_DOX_0.1_3     
 [43] 17.3_DOX_0.5_3      90.1_DOX_0.5_3      90.1_VEH_0.1_3     
 [46] 90.1_VEH_0.5_3      90.1_CX.5461_0.1_24 90.1_CX.5461_0.5_24
 [49] 90.1_DOX_0.1_24     90.1_DOX_0.5_24     90.1_VEH_0.1_24    
 [52] 90.1_VEH_0.5_24     90.1_CX.5461_0.1_48 17.3_VEH_0.1_3     
 [55] 90.1_CX.5461_0.5_48 90.1_DOX_0.1_48     90.1_DOX_0.5_48    
 [58] 90.1_VEH_0.1_48     90.1_VEH_0.5_48     75.1_CX.5461_0.1_3 
 [61] 75.1_CX.5461_0.5_3  75.1_DOX_0.1_3      75.1_DOX_0.5_3     
 [64] 75.1_VEH_0.1_3      17.3_VEH_0.5_3      75.1_VEH_0.5_3     
 [67] 75.1_CX.5461_0.1_24 75.1_CX.5461_0.5_24 75.1_DOX_0.1_24    
 [70] 75.1_DOX_0.5_24     75.1_VEH_0.1_24     75.1_VEH_0.5_24    
 [73] 75.1_CX.5461_0.1_48 75.1_CX.5461_0.5_48 75.1_DOX_0.1_48    
 [76] 17.3_CX.5461_0.1_24 75.1_DOX_0.5_48     75.1_VEH_0.1_48    
 [79] 75.1_VEH_0.5_48     78.1_CX.5461_0.1_3  78.1_CX.5461_0.5_3 
 [82] 78.1_DOX_0.1_3      78.1_DOX_0.5_3      78.1_VEH_0.1_3     
 [85] 78.1_VEH_0.5_3      78.1_CX.5461_0.1_24 17.3_CX.5461_0.5_24
 [88] 78.1_CX.5461_0.5_24 78.1_DOX_0.1_24     78.1_DOX_0.5_24    
 [91] 78.1_VEH_0.1_24     78.1_VEH_0.5_24     78.1_CX.5461_0.1_48
 [94] 78.1_CX.5461_0.5_48 78.1_DOX_0.1_48     78.1_DOX_0.5_48    
 [97] 78.1_VEH_0.1_48     17.3_DOX_0.1_24     78.1_VEH_0.5_48    
[100] 87.1_CX.5461_0.1_3  87.1_CX.5461_0.5_3  87.1_DOX_0.1_3     
[103] 87.1_DOX_0.5_3      87.1_VEH_0.1_3      87.1_VEH_0.5_3     
[106] 87.1_CX.5461_0.1_24 87.1_CX.5461_0.5_24 87.1_DOX_0.1_24    
108 Levels: 17.3_CX.5461_0.1_3 17.3_DOX_0.5_24 ... 87.1_DOX_0.1_24</code></pre>
<pre class="r"><code># Extract details from sample names
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Time = sub(&quot;.*_(\\d+)$&quot;, &quot;\\1&quot;, Sample),
    Concentration = sub(&quot;.*_(0\\.\\d)_\\d+$&quot;, &quot;\\1&quot;, Sample),
    Drug = sub(&quot;.*_(CX\\.5461|DOX|VEH)_.*&quot;, &quot;\\1&quot;, Sample),
    Indv = sub(&quot;^([0-9]+\\.[0-9]+)_.*&quot;, &quot;\\1&quot;, Sample)  # Ensure correct pattern extraction
  )

# Print extracted `Indv` column for debugging
print(unique(gene_data_long$Indv))</code></pre>
<pre><code>[1] &quot;17.3&quot; &quot;87.1&quot; &quot;84.1&quot; &quot;90.1&quot; &quot;75.1&quot; &quot;78.1&quot;</code></pre>
<pre class="r"><code># Convert extracted columns to factors
gene_data_long$Time &lt;- factor(gene_data_long$Time, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;))
gene_data_long$Concentration &lt;- factor(gene_data_long$Concentration, levels = c(&quot;0.1&quot;, &quot;0.5&quot;))

# Ensure each concentration facet only shows relevant data
gene_data_long &lt;- gene_data_long %&gt;%
  group_by(Concentration) %&gt;%
  filter(all(Concentration == &quot;0.1&quot;) | all(Concentration == &quot;0.5&quot;)) %&gt;%
  ungroup()

# Map individual IDs
indv_mapping &lt;- c(&quot;75.1&quot; = &quot;1&quot;, &quot;78.1&quot; = &quot;2&quot;, &quot;87.1&quot; = &quot;3&quot;, &quot;17.3&quot; = &quot;4&quot;, &quot;84.1&quot; = &quot;5&quot;, &quot;90.1&quot; = &quot;6&quot;)

# Ensure `Indv` values match the keys in `indv_mapping`
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], NA)
  )

# Print `Indv` column again to confirm mapping worked
print(unique(gene_data_long$Indv))</code></pre>
<pre><code>[1] &quot;4&quot; &quot;3&quot; &quot;5&quot; &quot;6&quot; &quot;1&quot; &quot;2&quot;</code></pre>
<pre class="r"><code># Check for missing values and handle them
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(Indv = ifelse(is.na(Indv), &quot;Unknown&quot;, Indv))  # Assign &quot;Unknown&quot; to unidentified individuals

# Define drug colors
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;, &quot;VEH&quot; = &quot;green&quot;)

# Extract the gene symbol for labeling
gene_symbol &lt;- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(Concentration ~ Time, labeller = label_both) +
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle(&quot;DOX-Specific Response&quot;) +
  labs(
    title = &quot;DOX-Specific Response&quot;,
    x = &quot;Drugs&quot;,
    y = paste(gene_symbol, &quot; log2CPM&quot;)
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_all.Rmd/box7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box7-1">
Past versions of box7-1.png
</button>
</p>
<div id="fig-box7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/ebfddcd2544a6364ccb5f8df72be2a583bc7ab4b/docs/figure/Corrmotif_all.Rmd/box7-1.png" target="_blank">ebfddcd</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="late-high-dose-dox-specific-response-ephx1"
class="section level2">
<h2><strong>ðŸ“Œ Late high dose DOX-specific response
(EPHX1)</strong></h2>
<pre class="r"><code># Load the dataset from the data folder
# Load the dataset from the data folder

# Filter data for the gene with Entrez ID:  2052
gene_data &lt;- boxplot1[boxplot1$ENTREZID ==  2052, ]

# Check if gene_data is empty
if(nrow(gene_data) == 0) {
  stop(&quot;No data found for the selected gene ENTIREZID   
2052.&quot;)
}

# Reshape data from wide to long format
gene_data_long &lt;- melt(gene_data,
                       id.vars = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;, &quot;GENENAME&quot;),
                       variable.name = &quot;Sample&quot;,
                       value.name = &quot;log2CPM&quot;)

# Print some sample names for debugging
print(unique(gene_data_long$Sample))</code></pre>
<pre><code>  [1] 17.3_CX.5461_0.1_3  17.3_DOX_0.5_24     87.1_DOX_0.5_24    
  [4] 87.1_VEH_0.1_24     87.1_VEH_0.5_24     87.1_CX.5461_0.1_48
  [7] 87.1_CX.5461_0.5_48 87.1_DOX_0.1_48     87.1_DOX_0.5_48    
 [10] 87.1_VEH_0.1_48     87.1_VEH_0.5_48     17.3_VEH_0.1_24    
 [13] 17.3_VEH_0.5_24     17.3_CX.5461_0.1_48 17.3_CX.5461_0.5_48
 [16] 17.3_DOX_0.1_48     17.3_DOX_0.5_48     17.3_VEH_0.1_48    
 [19] 17.3_VEH_0.5_48     84.1_CX.5461_0.1_3  17.3_CX.5461_0.5_3 
 [22] 84.1_CX.5461_0.5_3  84.1_DOX_0.1_3      84.1_DOX_0.5_3     
 [25] 84.1_VEH_0.1_3      84.1_VEH_0.5_3      84.1_CX.5461_0.1_24
 [28] 84.1_CX.5461_0.5_24 84.1_DOX_0.1_24     84.1_DOX_0.5_24    
 [31] 84.1_VEH_0.1_24     17.3_DOX_0.1_3      84.1_VEH_0.5_24    
 [34] 84.1_CX.5461_0.1_48 84.1_CX.5461_0.5_48 84.1_DOX_0.1_48    
 [37] 84.1_DOX_0.5_48     84.1_VEH_0.1_48     84.1_VEH_0.5_48    
 [40] 90.1_CX.5461_0.1_3  90.1_CX.5461_0.5_3  90.1_DOX_0.1_3     
 [43] 17.3_DOX_0.5_3      90.1_DOX_0.5_3      90.1_VEH_0.1_3     
 [46] 90.1_VEH_0.5_3      90.1_CX.5461_0.1_24 90.1_CX.5461_0.5_24
 [49] 90.1_DOX_0.1_24     90.1_DOX_0.5_24     90.1_VEH_0.1_24    
 [52] 90.1_VEH_0.5_24     90.1_CX.5461_0.1_48 17.3_VEH_0.1_3     
 [55] 90.1_CX.5461_0.5_48 90.1_DOX_0.1_48     90.1_DOX_0.5_48    
 [58] 90.1_VEH_0.1_48     90.1_VEH_0.5_48     75.1_CX.5461_0.1_3 
 [61] 75.1_CX.5461_0.5_3  75.1_DOX_0.1_3      75.1_DOX_0.5_3     
 [64] 75.1_VEH_0.1_3      17.3_VEH_0.5_3      75.1_VEH_0.5_3     
 [67] 75.1_CX.5461_0.1_24 75.1_CX.5461_0.5_24 75.1_DOX_0.1_24    
 [70] 75.1_DOX_0.5_24     75.1_VEH_0.1_24     75.1_VEH_0.5_24    
 [73] 75.1_CX.5461_0.1_48 75.1_CX.5461_0.5_48 75.1_DOX_0.1_48    
 [76] 17.3_CX.5461_0.1_24 75.1_DOX_0.5_48     75.1_VEH_0.1_48    
 [79] 75.1_VEH_0.5_48     78.1_CX.5461_0.1_3  78.1_CX.5461_0.5_3 
 [82] 78.1_DOX_0.1_3      78.1_DOX_0.5_3      78.1_VEH_0.1_3     
 [85] 78.1_VEH_0.5_3      78.1_CX.5461_0.1_24 17.3_CX.5461_0.5_24
 [88] 78.1_CX.5461_0.5_24 78.1_DOX_0.1_24     78.1_DOX_0.5_24    
 [91] 78.1_VEH_0.1_24     78.1_VEH_0.5_24     78.1_CX.5461_0.1_48
 [94] 78.1_CX.5461_0.5_48 78.1_DOX_0.1_48     78.1_DOX_0.5_48    
 [97] 78.1_VEH_0.1_48     17.3_DOX_0.1_24     78.1_VEH_0.5_48    
[100] 87.1_CX.5461_0.1_3  87.1_CX.5461_0.5_3  87.1_DOX_0.1_3     
[103] 87.1_DOX_0.5_3      87.1_VEH_0.1_3      87.1_VEH_0.5_3     
[106] 87.1_CX.5461_0.1_24 87.1_CX.5461_0.5_24 87.1_DOX_0.1_24    
108 Levels: 17.3_CX.5461_0.1_3 17.3_DOX_0.5_24 ... 87.1_DOX_0.1_24</code></pre>
<pre class="r"><code># Extract details from sample names
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Time = sub(&quot;.*_(\\d+)$&quot;, &quot;\\1&quot;, Sample),
    Concentration = sub(&quot;.*_(0\\.\\d)_\\d+$&quot;, &quot;\\1&quot;, Sample),
    Drug = sub(&quot;.*_(CX\\.5461|DOX|VEH)_.*&quot;, &quot;\\1&quot;, Sample),
    Indv = sub(&quot;^([0-9]+\\.[0-9]+)_.*&quot;, &quot;\\1&quot;, Sample)  # Ensure correct pattern extraction
  )

# Print extracted `Indv` column for debugging
print(unique(gene_data_long$Indv))</code></pre>
<pre><code>[1] &quot;17.3&quot; &quot;87.1&quot; &quot;84.1&quot; &quot;90.1&quot; &quot;75.1&quot; &quot;78.1&quot;</code></pre>
<pre class="r"><code># Convert extracted columns to factors
gene_data_long$Time &lt;- factor(gene_data_long$Time, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;))
gene_data_long$Concentration &lt;- factor(gene_data_long$Concentration, levels = c(&quot;0.1&quot;, &quot;0.5&quot;))

# Ensure each concentration facet only shows relevant data
gene_data_long &lt;- gene_data_long %&gt;%
  group_by(Concentration) %&gt;%
  filter(all(Concentration == &quot;0.1&quot;) | all(Concentration == &quot;0.5&quot;)) %&gt;%
  ungroup()

# Map individual IDs
indv_mapping &lt;- c(&quot;75.1&quot; = &quot;1&quot;, &quot;78.1&quot; = &quot;2&quot;, &quot;87.1&quot; = &quot;3&quot;, &quot;17.3&quot; = &quot;4&quot;, &quot;84.1&quot; = &quot;5&quot;, &quot;90.1&quot; = &quot;6&quot;)

# Ensure `Indv` values match the keys in `indv_mapping`
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], NA)
  )

# Print `Indv` column again to confirm mapping worked
print(unique(gene_data_long$Indv))</code></pre>
<pre><code>[1] &quot;4&quot; &quot;3&quot; &quot;5&quot; &quot;6&quot; &quot;1&quot; &quot;2&quot;</code></pre>
<pre class="r"><code># Check for missing values and handle them
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(Indv = ifelse(is.na(Indv), &quot;Unknown&quot;, Indv))  # Assign &quot;Unknown&quot; to unidentified individuals

# Define drug colors
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;, &quot;VEH&quot; = &quot;green&quot;)

# Extract the gene symbol for labeling
gene_symbol &lt;- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(Concentration ~ Time, labeller = label_both) +
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle(&quot;Late high dose DOX-specific response&quot;) +
  labs(
    title = &quot;Late high dose DOX-specific response&quot;,
    x = &quot;Drugs&quot;,
    y = paste(gene_symbol, &quot; log2CPM&quot;)
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_all.Rmd/box8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box8-1">
Past versions of box8-1.png
</button>
</p>
<div id="fig-box8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/ebfddcd2544a6364ccb5f8df72be2a583bc7ab4b/docs/figure/Corrmotif_all.Rmd/box8-1.png" target="_blank">ebfddcd</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="late-high-dose-dox-specific-response-obscn"
class="section level2">
<h2><strong>ðŸ“Œ Late high dose DOX-specific response
(OBSCN)</strong></h2>
<pre class="r"><code># Load the dataset from the data folder
# Load the dataset from the data folder

# Filter data for the gene with Entrez ID:  84033
gene_data &lt;- boxplot1[boxplot1$ENTREZID ==  84033, ]

# Check if gene_data is empty
if(nrow(gene_data) == 0) {
  stop(&quot;No data found for the selected gene ENTIREZID   
84033.&quot;)
}

# Reshape data from wide to long format
gene_data_long &lt;- melt(gene_data,
                       id.vars = c(&quot;ENTREZID&quot;, &quot;SYMBOL&quot;, &quot;GENENAME&quot;),
                       variable.name = &quot;Sample&quot;,
                       value.name = &quot;log2CPM&quot;)

# Print some sample names for debugging
print(unique(gene_data_long$Sample))</code></pre>
<pre><code>  [1] 17.3_CX.5461_0.1_3  17.3_DOX_0.5_24     87.1_DOX_0.5_24    
  [4] 87.1_VEH_0.1_24     87.1_VEH_0.5_24     87.1_CX.5461_0.1_48
  [7] 87.1_CX.5461_0.5_48 87.1_DOX_0.1_48     87.1_DOX_0.5_48    
 [10] 87.1_VEH_0.1_48     87.1_VEH_0.5_48     17.3_VEH_0.1_24    
 [13] 17.3_VEH_0.5_24     17.3_CX.5461_0.1_48 17.3_CX.5461_0.5_48
 [16] 17.3_DOX_0.1_48     17.3_DOX_0.5_48     17.3_VEH_0.1_48    
 [19] 17.3_VEH_0.5_48     84.1_CX.5461_0.1_3  17.3_CX.5461_0.5_3 
 [22] 84.1_CX.5461_0.5_3  84.1_DOX_0.1_3      84.1_DOX_0.5_3     
 [25] 84.1_VEH_0.1_3      84.1_VEH_0.5_3      84.1_CX.5461_0.1_24
 [28] 84.1_CX.5461_0.5_24 84.1_DOX_0.1_24     84.1_DOX_0.5_24    
 [31] 84.1_VEH_0.1_24     17.3_DOX_0.1_3      84.1_VEH_0.5_24    
 [34] 84.1_CX.5461_0.1_48 84.1_CX.5461_0.5_48 84.1_DOX_0.1_48    
 [37] 84.1_DOX_0.5_48     84.1_VEH_0.1_48     84.1_VEH_0.5_48    
 [40] 90.1_CX.5461_0.1_3  90.1_CX.5461_0.5_3  90.1_DOX_0.1_3     
 [43] 17.3_DOX_0.5_3      90.1_DOX_0.5_3      90.1_VEH_0.1_3     
 [46] 90.1_VEH_0.5_3      90.1_CX.5461_0.1_24 90.1_CX.5461_0.5_24
 [49] 90.1_DOX_0.1_24     90.1_DOX_0.5_24     90.1_VEH_0.1_24    
 [52] 90.1_VEH_0.5_24     90.1_CX.5461_0.1_48 17.3_VEH_0.1_3     
 [55] 90.1_CX.5461_0.5_48 90.1_DOX_0.1_48     90.1_DOX_0.5_48    
 [58] 90.1_VEH_0.1_48     90.1_VEH_0.5_48     75.1_CX.5461_0.1_3 
 [61] 75.1_CX.5461_0.5_3  75.1_DOX_0.1_3      75.1_DOX_0.5_3     
 [64] 75.1_VEH_0.1_3      17.3_VEH_0.5_3      75.1_VEH_0.5_3     
 [67] 75.1_CX.5461_0.1_24 75.1_CX.5461_0.5_24 75.1_DOX_0.1_24    
 [70] 75.1_DOX_0.5_24     75.1_VEH_0.1_24     75.1_VEH_0.5_24    
 [73] 75.1_CX.5461_0.1_48 75.1_CX.5461_0.5_48 75.1_DOX_0.1_48    
 [76] 17.3_CX.5461_0.1_24 75.1_DOX_0.5_48     75.1_VEH_0.1_48    
 [79] 75.1_VEH_0.5_48     78.1_CX.5461_0.1_3  78.1_CX.5461_0.5_3 
 [82] 78.1_DOX_0.1_3      78.1_DOX_0.5_3      78.1_VEH_0.1_3     
 [85] 78.1_VEH_0.5_3      78.1_CX.5461_0.1_24 17.3_CX.5461_0.5_24
 [88] 78.1_CX.5461_0.5_24 78.1_DOX_0.1_24     78.1_DOX_0.5_24    
 [91] 78.1_VEH_0.1_24     78.1_VEH_0.5_24     78.1_CX.5461_0.1_48
 [94] 78.1_CX.5461_0.5_48 78.1_DOX_0.1_48     78.1_DOX_0.5_48    
 [97] 78.1_VEH_0.1_48     17.3_DOX_0.1_24     78.1_VEH_0.5_48    
[100] 87.1_CX.5461_0.1_3  87.1_CX.5461_0.5_3  87.1_DOX_0.1_3     
[103] 87.1_DOX_0.5_3      87.1_VEH_0.1_3      87.1_VEH_0.5_3     
[106] 87.1_CX.5461_0.1_24 87.1_CX.5461_0.5_24 87.1_DOX_0.1_24    
108 Levels: 17.3_CX.5461_0.1_3 17.3_DOX_0.5_24 ... 87.1_DOX_0.1_24</code></pre>
<pre class="r"><code># Extract details from sample names
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Time = sub(&quot;.*_(\\d+)$&quot;, &quot;\\1&quot;, Sample),
    Concentration = sub(&quot;.*_(0\\.\\d)_\\d+$&quot;, &quot;\\1&quot;, Sample),
    Drug = sub(&quot;.*_(CX\\.5461|DOX|VEH)_.*&quot;, &quot;\\1&quot;, Sample),
    Indv = sub(&quot;^([0-9]+\\.[0-9]+)_.*&quot;, &quot;\\1&quot;, Sample)  # Ensure correct pattern extraction
  )

# Print extracted `Indv` column for debugging
print(unique(gene_data_long$Indv))</code></pre>
<pre><code>[1] &quot;17.3&quot; &quot;87.1&quot; &quot;84.1&quot; &quot;90.1&quot; &quot;75.1&quot; &quot;78.1&quot;</code></pre>
<pre class="r"><code># Convert extracted columns to factors
gene_data_long$Time &lt;- factor(gene_data_long$Time, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;))
gene_data_long$Concentration &lt;- factor(gene_data_long$Concentration, levels = c(&quot;0.1&quot;, &quot;0.5&quot;))

# Ensure each concentration facet only shows relevant data
gene_data_long &lt;- gene_data_long %&gt;%
  group_by(Concentration) %&gt;%
  filter(all(Concentration == &quot;0.1&quot;) | all(Concentration == &quot;0.5&quot;)) %&gt;%
  ungroup()

# Map individual IDs
indv_mapping &lt;- c(&quot;75.1&quot; = &quot;1&quot;, &quot;78.1&quot; = &quot;2&quot;, &quot;87.1&quot; = &quot;3&quot;, &quot;17.3&quot; = &quot;4&quot;, &quot;84.1&quot; = &quot;5&quot;, &quot;90.1&quot; = &quot;6&quot;)

# Ensure `Indv` values match the keys in `indv_mapping`
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(
    Indv = ifelse(Indv %in% names(indv_mapping), indv_mapping[Indv], NA)
  )

# Print `Indv` column again to confirm mapping worked
print(unique(gene_data_long$Indv))</code></pre>
<pre><code>[1] &quot;4&quot; &quot;3&quot; &quot;5&quot; &quot;6&quot; &quot;1&quot; &quot;2&quot;</code></pre>
<pre class="r"><code># Check for missing values and handle them
gene_data_long &lt;- gene_data_long %&gt;%
  mutate(Indv = ifelse(is.na(Indv), &quot;Unknown&quot;, Indv))  # Assign &quot;Unknown&quot; to unidentified individuals

# Define drug colors
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;, &quot;VEH&quot; = &quot;green&quot;)

# Extract the gene symbol for labeling
gene_symbol &lt;- unique(gene_data_long$SYMBOL)[1]

# Create the boxplot
ggplot(gene_data_long, aes(x = Drug, y = log2CPM, fill = Drug)) +
  geom_boxplot(outlier.shape = NA) +
  scale_fill_manual(values = drug_palette) +
  facet_grid(Concentration ~ Time, labeller = label_both) +
  geom_point(aes(color = Indv), size = 2, alpha = 0.5,
             position = position_jitter(width = -0.3, height = 0)) +
  ggtitle(&quot;Late high dose DOX-specific response&quot;) +
  labs(
    title = &quot;Late high dose DOX-specific response&quot;,
    x = &quot;Drugs&quot;,
    y = paste(gene_symbol, &quot; log2CPM&quot;)
  ) +
  ylim(0, NA) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 45, hjust = 1),
    strip.text = element_text(size = 12, face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Corrmotif_all.Rmd/box9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box9-1">
Past versions of box9-1.png
</button>
</p>
<div id="fig-box9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/ebfddcd2544a6364ccb5f8df72be2a583bc7ab4b/docs/figure/Corrmotif_all.Rmd/box9-1.png" target="_blank">ebfddcd</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="corr-motif-set-abs-logfc-boxplots" class="section level2">
<h2><strong>ðŸ“Œ Corr-motif set Abs logFC boxplots</strong></h2>
</div>
<div id="read-and-process-deg-data" class="section level2">
<h2><strong>ðŸ“Œ Read and Process DEG Data</strong></h2>
<pre class="r"><code># Load DEGs Data
CX_0.1_3 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_3.csv&quot;)
CX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_24.csv&quot;)
CX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.1_48.csv&quot;)
CX_0.5_3 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_3.csv&quot;)
CX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_24.csv&quot;)
CX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_CX_0.5_48.csv&quot;)

DOX_0.1_3 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_3.csv&quot;)
DOX_0.1_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_24.csv&quot;)
DOX_0.1_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.1_48.csv&quot;)
DOX_0.5_3 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_3.csv&quot;)
DOX_0.5_24 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_24.csv&quot;)
DOX_0.5_48 &lt;- read.csv(&quot;data/DEGs/Toptable_DOX_0.5_48.csv&quot;)

Toptable_CX_0.1_3_df&lt;- data.frame(CX_0.1_3)
Toptable_CX_0.1_24_df&lt;- data.frame(CX_0.1_24)
Toptable_CX_0.1_48_df&lt;- data.frame(CX_0.1_48)
Toptable_CX_0.5_3_df&lt;- data.frame(CX_0.5_3)
Toptable_CX_0.5_24_df&lt;- data.frame(CX_0.5_24)
Toptable_CX_0.5_48_df&lt;- data.frame(CX_0.5_48)
Toptable_DOX_0.1_3_df&lt;- data.frame(DOX_0.1_3)
Toptable_DOX_0.1_24_df&lt;- data.frame(DOX_0.1_24)
Toptable_DOX_0.1_48_df&lt;- data.frame(DOX_0.1_48)
Toptable_DOX_0.5_3_df&lt;- data.frame(DOX_0.5_3)
Toptable_DOX_0.5_24_df&lt;- data.frame(DOX_0.5_24)
Toptable_DOX_0.5_48_df&lt;- data.frame(DOX_0.5_48)</code></pre>
</div>
<div id="non-response-abs-logfc-boxplots" class="section level2">
<h2><strong>ðŸ“Œ Non response abs logFC boxplots</strong></h2>
<pre class="r"><code># Extract Entrez_IDs from set1
non_response_entrez_ids &lt;- prob_all_1

all_toptables &lt;- bind_rows(
  Toptable_CX_0.1_3_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;3&quot;),
  Toptable_CX_0.1_24_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;24&quot;),
  Toptable_CX_0.1_48_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;48&quot;),
  Toptable_CX_0.5_3_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;3&quot;),
  Toptable_CX_0.5_24_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;24&quot;),
  Toptable_CX_0.5_48_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;48&quot;),
  Toptable_DOX_0.1_3_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;3&quot;),
  Toptable_DOX_0.1_24_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;24&quot;),
  Toptable_DOX_0.1_48_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;48&quot;),
  Toptable_DOX_0.5_3_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;3&quot;),
  Toptable_DOX_0.5_24_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;24&quot;),
  Toptable_DOX_0.5_48_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;48&quot;)
)

# Filter and visualize data for shared late response genes
filtered_toptables &lt;- all_toptables %&gt;%
  filter(Entrez_ID %in% non_response_entrez_ids)

# Filter and visualize data with exact same font size and bold styling for concentration and timepoints
filtered_toptables %&gt;%
  mutate(absFC = abs(logFC)) %&gt;%  # Compute absolute log fold change
  mutate(
    Drug = factor(Drug, levels = c(&quot;CX.5461&quot;, &quot;DOX&quot;)),  # Factorize Drug with CX.5461 first
    Timepoint = factor(Timepoint, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;), labels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Concentration = factor(Concentration, levels = c(0.1, 0.5), labels = c(&quot;Concentration: 0.1&quot;, &quot;Concentration: 0.5&quot;))
  ) %&gt;% 
  ggplot(aes(x = Drug, y = absFC)) +
  geom_boxplot(aes(fill = Drug)) +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +  # Updated color palette
  guides(fill = guide_legend(title = &quot;Treatment&quot;)) +
  facet_grid(Concentration ~ Timepoint) +  # Facet by Concentration and Timepoint
  theme_bw() +
  xlab(&quot;Drugs&quot;) +
  ylab(&quot;|Log Fold Change|&quot;) +
  ggtitle(&quot;|Log Fold| for Non-Response Genes&quot;) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.line = element_line(linewidth = 1.5),
    strip.background = element_rect(fill = &quot;gray&quot;),  # Set strip background to gray
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),  # Unified styling for all facet labels
    axis.text.x = element_text(size = 8, color = &quot;black&quot;, angle = 15)
  )</code></pre>
<p><img src="figure/Corrmotif_all.Rmd/box10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box10-1">
Past versions of box10-1.png
</button>
</p>
<div id="fig-box10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/2862c42bbe271925d87c00cd50b08289692b2517/docs/figure/Corrmotif_all.Rmd/box10-1.png" target="_blank">2862c42</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cx_dox-shared-late-response-abs-logfc-boxplots"
class="section level2">
<h2><strong>ðŸ“Œ CX_DOX shared late response abs logFC
boxplots</strong></h2>
<pre class="r"><code># Extract Entrez_IDs from set2
shared_late_response_entrez_ids &lt;- prob_all_2

# Combine all subset_toptables into a single data frame with additional metadata columns
all_toptables &lt;- bind_rows(
  Toptable_CX_0.1_3_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;3&quot;),
  Toptable_CX_0.1_24_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;24&quot;),
  Toptable_CX_0.1_48_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;48&quot;),
  Toptable_CX_0.5_3_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;3&quot;),
  Toptable_CX_0.5_24_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;24&quot;),
  Toptable_CX_0.5_48_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;48&quot;),
  Toptable_DOX_0.1_3_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;3&quot;),
  Toptable_DOX_0.1_24_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;24&quot;),
  Toptable_DOX_0.1_48_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;48&quot;),
  Toptable_DOX_0.5_3_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;3&quot;),
  Toptable_DOX_0.5_24_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;24&quot;),
  Toptable_DOX_0.5_48_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;48&quot;)
)

# Filter and visualize data for shared late response genes
filtered_toptables &lt;- all_toptables %&gt;%
  filter(Entrez_ID %in% shared_late_response_entrez_ids)

filtered_toptables %&gt;%
  mutate(absFC = abs(logFC)) %&gt;%  # Compute absolute log fold change
  mutate(
    Drug = factor(Drug, levels = c(&quot;CX.5461&quot;, &quot;DOX&quot;)),  # Factorize Drug with CX.5461 first
    Timepoint = factor(Timepoint, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;), labels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Concentration = factor(Concentration, levels = c(0.1, 0.5), labels = c(&quot;Concentration: 0.1&quot;, &quot;Concentration: 0.5&quot;))
  ) %&gt;%
  ggplot(aes(x = Drug, y = absFC)) +
  geom_boxplot(aes(fill = Drug)) +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +  # Updated color palette
  guides(fill = guide_legend(title = &quot;Treatment&quot;)) +
  facet_grid(Concentration ~ Timepoint) +  # Facet by Concentration and Timepoint
  theme_bw() +
  xlab(&quot;Drugs&quot;) +
  ylab(&quot;|Log Fold Change|&quot;) +
  ggtitle(&quot;|Log Fold| for CX_DOX shared late response Genes&quot;) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.line = element_line(linewidth = 1.5),
    strip.background = element_rect(fill = &quot;gray&quot;),  # Set strip background to gray
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),  # Unified styling for all facet labels
    axis.text.x = element_text(size = 8, color = &quot;black&quot;, angle = 15)
  )</code></pre>
<p><img src="figure/Corrmotif_all.Rmd/box11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box11-1">
Past versions of box11-1.png
</button>
</p>
<div id="fig-box11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/2862c42bbe271925d87c00cd50b08289692b2517/docs/figure/Corrmotif_all.Rmd/box11-1.png" target="_blank">2862c42</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dox-specific-response-abs-logfc-boxplots"
class="section level2">
<h2><strong>ðŸ“Œ DOX Specific response abs logFC boxplots</strong></h2>
<pre class="r"><code># Extract Entrez_IDs from set3
dox_specific_response_entrez_ids &lt;- prob_all_3

# Combine all subset_toptables into a single data frame with additional metadata columns
all_toptables &lt;- bind_rows(
  Toptable_CX_0.1_3_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;3&quot;),
  Toptable_CX_0.1_24_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;24&quot;),
  Toptable_CX_0.1_48_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;48&quot;),
  Toptable_CX_0.5_3_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;3&quot;),
  Toptable_CX_0.5_24_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;24&quot;),
  Toptable_CX_0.5_48_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;48&quot;),
  Toptable_DOX_0.1_3_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;3&quot;),
  Toptable_DOX_0.1_24_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;24&quot;),
  Toptable_DOX_0.1_48_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;48&quot;),
  Toptable_DOX_0.5_3_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;3&quot;),
  Toptable_DOX_0.5_24_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;24&quot;),
  Toptable_DOX_0.5_48_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;48&quot;)
)

# Filter and visualize data for DOX-specific response genes
filtered_toptables &lt;- all_toptables %&gt;%
  filter(Entrez_ID %in% dox_specific_response_entrez_ids)

filtered_toptables %&gt;%
  mutate(absFC = abs(logFC)) %&gt;%  # Compute absolute log fold change
  mutate(
    Drug = factor(Drug, levels = c(&quot;CX.5461&quot;, &quot;DOX&quot;)),  # Factorize Drug with CX.5461 first
    Timepoint = factor(Timepoint, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;), labels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Concentration = factor(Concentration, levels = c(0.1, 0.5), labels = c(&quot;Concentration: 0.1&quot;, &quot;Concentration: 0.5&quot;))
  ) %&gt;%
  ggplot(aes(x = Drug, y = absFC)) +
  geom_boxplot(aes(fill = Drug)) +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +  # Updated color palette
  guides(fill = guide_legend(title = &quot;Treatment&quot;)) +
  facet_grid(Concentration ~ Timepoint) +  # Facet by Concentration and Timepoint
  theme_bw() +
  xlab(&quot;Drugs&quot;) +
  ylab(&quot;|Log Fold Change|&quot;) +
  ggtitle(&quot;|Log Fold| for DOX-Specific Response Genes&quot;) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.line = element_line(linewidth = 1.5),
    strip.background = element_rect(fill = &quot;gray&quot;),  # Set strip background to gray
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),  # Unified styling for all facet labels
    axis.text.x = element_text(size = 8, color = &quot;black&quot;, angle = 15)
  )</code></pre>
<p><img src="figure/Corrmotif_all.Rmd/box12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box12-1">
Past versions of box12-1.png
</button>
</p>
<div id="fig-box12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/2862c42bbe271925d87c00cd50b08289692b2517/docs/figure/Corrmotif_all.Rmd/box12-1.png" target="_blank">2862c42</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="late-high-dose-dox-specific-response-abs-logfc-boxplots"
class="section level2">
<h2><strong>ðŸ“Œ Late high dose DOX-specific response abs logFC
boxplots</strong></h2>
<pre class="r"><code># Extract Entrez_IDs from set4
late_high_dose_dox_entrez_ids &lt;- prob_all_4

# Combine all subset_toptables into a single data frame with additional metadata columns
all_toptables &lt;- bind_rows(
  Toptable_CX_0.1_3_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;3&quot;),
  Toptable_CX_0.1_24_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;24&quot;),
  Toptable_CX_0.1_48_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;48&quot;),
  Toptable_CX_0.5_3_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;3&quot;),
  Toptable_CX_0.5_24_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;24&quot;),
  Toptable_CX_0.5_48_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;48&quot;),
  Toptable_DOX_0.1_3_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;3&quot;),
  Toptable_DOX_0.1_24_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;24&quot;),
  Toptable_DOX_0.1_48_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;48&quot;),
  Toptable_DOX_0.5_3_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;3&quot;),
  Toptable_DOX_0.5_24_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;24&quot;),
  Toptable_DOX_0.5_48_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;48&quot;)
)

# Filter and visualize data for late high dose DOX-specific response genes
filtered_toptables &lt;- all_toptables %&gt;%
  filter(Entrez_ID %in% late_high_dose_dox_entrez_ids)

filtered_toptables %&gt;%
  mutate(absFC = abs(logFC)) %&gt;%  # Compute absolute log fold change
  mutate(
    Drug = factor(Drug, levels = c(&quot;CX.5461&quot;, &quot;DOX&quot;)),  # Factorize Drug with CX.5461 first
    Timepoint = factor(Timepoint, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;), labels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Concentration = factor(Concentration, levels = c(0.1, 0.5), labels = c(&quot;Concentration: 0.1&quot;, &quot;Concentration: 0.5&quot;))
  ) %&gt;%
  ggplot(aes(x = Drug, y = absFC)) +
  geom_boxplot(aes(fill = Drug)) +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +  # Updated color palette
  guides(fill = guide_legend(title = &quot;Treatment&quot;)) +
  facet_grid(Concentration ~ Timepoint) +  # Facet by Concentration and Timepoint
  theme_bw() +
  xlab(&quot;Drugs&quot;) +
  ylab(&quot;|Log Fold Change|&quot;) +
  ggtitle(&quot;|Log Fold| for Late High Dose DOX-Specific Response Genes&quot;) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.line = element_line(linewidth = 1.5),
    strip.background = element_rect(fill = &quot;gray&quot;),  # Set strip background to gray
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),  # Unified styling for all facet labels
    axis.text.x = element_text(size = 8, color = &quot;black&quot;, angle = 15)
  )</code></pre>
<p><img src="figure/Corrmotif_all.Rmd/box13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box13-1">
Past versions of box13-1.png
</button>
</p>
<div id="fig-box13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/2862c42bbe271925d87c00cd50b08289692b2517/docs/figure/Corrmotif_all.Rmd/box13-1.png" target="_blank">2862c42</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="corr-motif-set-mean-abs-logfc-across-timepoints"
class="section level2">
<h2><strong>ðŸ“Œ Corr-motif set mean (Abs logFC) across
timepoints</strong></h2>
<pre class="r"><code>all_toptables &lt;- bind_rows(
  Toptable_CX_0.1_3_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;3&quot;),
  Toptable_CX_0.1_24_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;24&quot;),
  Toptable_CX_0.1_48_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;48&quot;),
  Toptable_CX_0.5_3_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;3&quot;),
  Toptable_CX_0.5_24_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;24&quot;),
  Toptable_CX_0.5_48_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;48&quot;),
  Toptable_DOX_0.1_3_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;3&quot;),
  Toptable_DOX_0.1_24_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;24&quot;),
  Toptable_DOX_0.1_48_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;48&quot;),
  Toptable_DOX_0.5_3_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;3&quot;),
  Toptable_DOX_0.5_24_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;24&quot;),
  Toptable_DOX_0.5_48_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;48&quot;)
)

# Convert Entrez_ID in `all_toptables` to character for proper matching
all_toptables &lt;- all_toptables %&gt;% mutate(Entrez_ID = as.character(Entrez_ID))

# Define datasets for filtering (lists of Entrez IDs)
data_sets &lt;- list(
  &quot;Non Response&quot; = prob_all_1,
  &quot;CX_DOX Shared\nLate Response&quot; = prob_all_2,
  &quot;DOX-Specific Response&quot; = prob_all_3,
  &quot;Late High Dose\nDOX-Specific Response&quot; = prob_all_4
)

# Filter datasets and add labels
data_combined &lt;- bind_rows(lapply(names(data_sets), function(label) {
  all_toptables %&gt;%
    filter(Entrez_ID %in% data_sets[[label]]) %&gt;%  # Use list directly for filtering
    mutate(Dataset = label)
}))

# Ensure `logFC` column exists before computing mean absolute logFC
if (!&quot;logFC&quot; %in% colnames(data_combined)) {
  stop(&quot;Column `logFC` is missing in `data_combined`. Ensure that `all_toptables` has `logFC`.&quot;)
}

# Compute mean absolute logFC grouped by Dataset, Drug, Concentration, and Timepoint
data_summary &lt;- data_combined %&gt;%
  mutate(abs_logFC = abs(logFC)) %&gt;%  # Take absolute log fold change
  group_by(Dataset, Drug, Concentration, Timepoint) %&gt;%
  dplyr::summarize(mean_abs_logFC = mean(abs_logFC, na.rm = TRUE), .groups = &quot;drop&quot;) %&gt;%
  as.data.frame()

# Convert factors for plotting
data_summary &lt;- data_summary %&gt;%
  mutate(
    Timepoint = factor(Timepoint, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;), labels = c(&quot;3 hours&quot;, &quot;24 hours&quot;, &quot;48 hours&quot;)),
    Concentration = factor(Concentration, levels = c(0.1, 0.5)),
    Dataset = factor(Dataset, levels = c(&quot;Non Response&quot;, &quot;CX_DOX Shared\nLate Response&quot;, &quot;DOX-Specific Response&quot;, &quot;Late High Dose\nDOX-Specific Response&quot;))
  )

# Define custom drug palette
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)

# Plot the data for absolute logFC
ggplot(data_summary, aes(x = Timepoint, y = mean_abs_logFC, group = Drug, color = Drug)) +
  geom_point(size = 3) +
  geom_line(size = 1.2) +
  scale_color_manual(values = drug_palette) +
  ylim(0, 2) +  # Adjust the Y-axis for better visualization
  facet_grid(Dataset ~ Concentration) +
  theme_bw() +
  labs(
    x = &quot;Timepoints&quot;,
    y = &quot;Mean |Log Fold Change|&quot;,
    title = &quot;Average |Log Fold Change| Across Motifs and Concentrations&quot;,
    color = &quot;Drug&quot;
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text = element_text(size = 12, color = &quot;black&quot;),
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_all.Rmd/box14-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box14-1">
Past versions of box14-1.png
</button>
</p>
<div id="fig-box14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/2862c42bbe271925d87c00cd50b08289692b2517/docs/figure/Corrmotif_all.Rmd/box14-1.png" target="_blank">2862c42</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="corr-motif-set-logfc-boxplots" class="section level2">
<h2><strong>ðŸ“Œ Corr-motif set logFC boxplots</strong></h2>
</div>
<div id="non-response-logfc-boxplots" class="section level2">
<h2><strong>ðŸ“Œ Non response logFC boxplots</strong></h2>
<pre class="r"><code># Extract Entrez_IDs from set1
non_response_entrez_ids &lt;- prob_all_1

# Combine all subset_toptables into a single data frame with additional metadata columns
all_toptables &lt;- bind_rows(
  Toptable_CX_0.1_3_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;3&quot;),
  Toptable_CX_0.1_24_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;24&quot;),
  Toptable_CX_0.1_48_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;48&quot;),
  Toptable_CX_0.5_3_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;3&quot;),
  Toptable_CX_0.5_24_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;24&quot;),
  Toptable_CX_0.5_48_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;48&quot;),
  Toptable_DOX_0.1_3_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;3&quot;),
  Toptable_DOX_0.1_24_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;24&quot;),
  Toptable_DOX_0.1_48_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;48&quot;),
  Toptable_DOX_0.5_3_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;3&quot;),
  Toptable_DOX_0.5_24_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;24&quot;),
  Toptable_DOX_0.5_48_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;48&quot;)
)


# Filter and visualize data for shared late response genes
filtered_toptables &lt;- all_toptables %&gt;%
  filter(Entrez_ID %in% non_response_entrez_ids)

# Visualization of logFC for non-response genes
filtered_toptables %&gt;%
  mutate(
    Drug = factor(Drug, levels = c(&quot;CX.5461&quot;, &quot;DOX&quot;)),  # Factorize Drug with CX.5461 first
    Timepoint = factor(Timepoint, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;), labels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Concentration = factor(Concentration, levels = c(0.1, 0.5), labels = c(&quot;Concentration: 0.1&quot;, &quot;Concentration: 0.5&quot;))
  ) %&gt;% 
  ggplot(aes(x = Drug, y = logFC)) +  # Plot logFC directly
  geom_boxplot(aes(fill = Drug)) +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +  # Updated color palette
  guides(fill = guide_legend(title = &quot;Treatment&quot;)) +
  facet_grid(Concentration ~ Timepoint) +  # Facet by Concentration and Timepoint
  theme_bw() +
  xlab(&quot;Drugs&quot;) +
  ylab(&quot;Log Fold Change&quot;) +  # Updated y-axis label
  ggtitle(&quot;Log Fold Change for Non-Response Genes&quot;) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.line = element_line(linewidth = 1.5),
    strip.background = element_rect(fill = &quot;gray&quot;),  # Set strip background to gray
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),  # Unified styling for all facet labels
    axis.text.x = element_text(size = 8, color = &quot;black&quot;, angle = 15)
  )</code></pre>
<p><img src="figure/Corrmotif_all.Rmd/box15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box15-1">
Past versions of box15-1.png
</button>
</p>
<div id="fig-box15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/2862c42bbe271925d87c00cd50b08289692b2517/docs/figure/Corrmotif_all.Rmd/box15-1.png" target="_blank">2862c42</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cx_dox-shared-late-response-logfc-boxplots"
class="section level2">
<h2><strong>ðŸ“Œ CX_DOX Shared Late Response logFC boxplots</strong></h2>
<pre class="r"><code># Extract Entrez_IDs from set2
cx_dox_shared_entrez_ids &lt;- prob_all_2

# Combine all subset_toptables into a single data frame with additional metadata columns
all_toptables &lt;- bind_rows(
  Toptable_CX_0.1_3_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;3&quot;),
  Toptable_CX_0.1_24_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;24&quot;),
  Toptable_CX_0.1_48_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;48&quot;),
  Toptable_CX_0.5_3_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;3&quot;),
  Toptable_CX_0.5_24_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;24&quot;),
  Toptable_CX_0.5_48_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;48&quot;),
  Toptable_DOX_0.1_3_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;3&quot;),
  Toptable_DOX_0.1_24_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;24&quot;),
  Toptable_DOX_0.1_48_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;48&quot;),
  Toptable_DOX_0.5_3_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;3&quot;),
  Toptable_DOX_0.5_24_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;24&quot;),
  Toptable_DOX_0.5_48_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;48&quot;)
)

# Filter and visualize data for CX_DOX shared late response genes
filtered_toptables &lt;- all_toptables %&gt;%
  filter(Entrez_ID %in% cx_dox_shared_entrez_ids)

# Visualization of logFC for CX_DOX shared late response genes
filtered_toptables %&gt;%
  mutate(
    Drug = factor(Drug, levels = c(&quot;CX.5461&quot;, &quot;DOX&quot;)),  # Factorize Drug with CX.5461 first
    Timepoint = factor(Timepoint, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;), labels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Concentration = factor(Concentration, levels = c(0.1, 0.5), labels = c(&quot;Concentration: 0.1&quot;, &quot;Concentration: 0.5&quot;))
  ) %&gt;% 
  ggplot(aes(x = Drug, y = logFC)) +  # Plot logFC directly
  geom_boxplot(aes(fill = Drug)) +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +  # Updated color palette
  guides(fill = guide_legend(title = &quot;Treatment&quot;)) +
  facet_grid(Concentration ~ Timepoint) +  # Facet by Concentration and Timepoint
  theme_bw() +
  theme_bw() +
  xlab(&quot;Drugs&quot;) +
  ylab(&quot;Log Fold Change&quot;) +  # Updated y-axis label
  ggtitle(&quot;Log Fold Change for CX_DOX Shared Late Response Genes&quot;) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.line = element_line(linewidth = 1.5),
    strip.background = element_rect(fill = &quot;gray&quot;),  # Set strip background to gray
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),  # Unified styling for all facet labels
    axis.text.x = element_text(size = 8, color = &quot;black&quot;, angle = 15)
  )</code></pre>
<p><img src="figure/Corrmotif_all.Rmd/box16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box16-1">
Past versions of box16-1.png
</button>
</p>
<div id="fig-box16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/2862c42bbe271925d87c00cd50b08289692b2517/docs/figure/Corrmotif_all.Rmd/box16-1.png" target="_blank">2862c42</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dox-specific-response-logfc-boxplots" class="section level2">
<h2><strong>ðŸ“Œ DOX-Specific Response logFC boxplots</strong></h2>
<pre class="r"><code># Extract Entrez_IDs from set3
dox_specific_entrez_ids &lt;- prob_all_3

# Combine all subset_toptables into a single data frame with additional metadata columns
all_toptables &lt;- bind_rows(
  Toptable_CX_0.1_3_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;3&quot;),
  Toptable_CX_0.1_24_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;24&quot;),
  Toptable_CX_0.1_48_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;48&quot;),
  Toptable_CX_0.5_3_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;3&quot;),
  Toptable_CX_0.5_24_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;24&quot;),
  Toptable_CX_0.5_48_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;48&quot;),
  Toptable_DOX_0.1_3_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;3&quot;),
  Toptable_DOX_0.1_24_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;24&quot;),
  Toptable_DOX_0.1_48_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;48&quot;),
  Toptable_DOX_0.5_3_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;3&quot;),
  Toptable_DOX_0.5_24_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;24&quot;),
  Toptable_DOX_0.5_48_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;48&quot;)
)

# Filter and visualize data for DOX-specific response genes
filtered_toptables &lt;- all_toptables %&gt;%
  filter(Entrez_ID %in% dox_specific_entrez_ids)

# Visualization of logFC for DOX-specific response genes
filtered_toptables %&gt;%
  mutate(
    Drug = factor(Drug, levels = c(&quot;CX.5461&quot;, &quot;DOX&quot;)),  # Factorize Drug with CX.5461 first
    Timepoint = factor(Timepoint, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;), labels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Concentration = factor(Concentration, levels = c(0.1, 0.5), labels = c(&quot;Concentration: 0.1&quot;, &quot;Concentration: 0.5&quot;))
  ) %&gt;% 
  ggplot(aes(x = Drug, y = logFC)) +  # Plot logFC directly
  geom_boxplot(aes(fill = Drug)) +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +  # Updated color palette
  guides(fill = guide_legend(title = &quot;Treatment&quot;)) +
  facet_grid(Concentration ~ Timepoint) +  # Facet by Concentration and Timepoint
  theme_bw() +
  xlab(&quot;Drugs&quot;) +
  ylab(&quot;Log Fold Change&quot;) +  # Updated y-axis label
  ggtitle(&quot;Log Fold Change for DOX-Specific Response Genes&quot;) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.line = element_line(linewidth = 1.5),
    strip.background = element_rect(fill = &quot;gray&quot;),  # Set strip background to gray
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),  # Unified styling for all facet labels
    axis.text.x = element_text(size = 8, color = &quot;black&quot;, angle = 15)
  )</code></pre>
<p><img src="figure/Corrmotif_all.Rmd/box17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box17-1">
Past versions of box17-1.png
</button>
</p>
<div id="fig-box17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/2862c42bbe271925d87c00cd50b08289692b2517/docs/figure/Corrmotif_all.Rmd/box17-1.png" target="_blank">2862c42</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="late-high-dose-dox-specific-response-logfc-boxplots"
class="section level2">
<h2><strong>ðŸ“Œ Late High Dose DOX-Specific Response logFC
boxplots</strong></h2>
<pre class="r"><code># Extract Entrez_IDs from set4
late_high_dose_dox_specific_entrez_ids &lt;- prob_all_4

# Combine all subset_toptables into a single data frame with additional metadata columns
all_toptables &lt;- bind_rows(
  Toptable_CX_0.1_3_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;3&quot;),
  Toptable_CX_0.1_24_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;24&quot;),
  Toptable_CX_0.1_48_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;48&quot;),
  Toptable_CX_0.5_3_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;3&quot;),
  Toptable_CX_0.5_24_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;24&quot;),
  Toptable_CX_0.5_48_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;48&quot;),
  Toptable_DOX_0.1_3_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;3&quot;),
  Toptable_DOX_0.1_24_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;24&quot;),
  Toptable_DOX_0.1_48_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;48&quot;),
  Toptable_DOX_0.5_3_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;3&quot;),
  Toptable_DOX_0.5_24_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;24&quot;),
  Toptable_DOX_0.5_48_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;48&quot;)
)

# Filter and visualize data for Late High Dose DOX-Specific Response Genes
filtered_toptables &lt;- all_toptables %&gt;%
  filter(Entrez_ID %in% late_high_dose_dox_specific_entrez_ids)

# Visualization of logFC for Late High Dose DOX-Specific Response Genes
filtered_toptables %&gt;%
  mutate(
    Drug = factor(Drug, levels = c(&quot;CX.5461&quot;, &quot;DOX&quot;)),  # Factorize Drug with CX.5461 first
    Timepoint = factor(Timepoint, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;), labels = c(&quot;Timepoint: 3 hours&quot;, &quot;Timepoint: 24 hours&quot;, &quot;Timepoint: 48 hours&quot;)),
    Concentration = factor(Concentration, levels = c(0.1, 0.5), labels = c(&quot;Concentration: 0.1&quot;, &quot;Concentration: 0.5&quot;))
  ) %&gt;% 
  ggplot(aes(x = Drug, y = logFC)) +  # Plot logFC directly
  geom_boxplot(aes(fill = Drug)) +
  scale_fill_manual(values = c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)) +  # Updated color palette
  guides(fill = guide_legend(title = &quot;Treatment&quot;)) +
  facet_grid(Concentration ~ Timepoint) +  # Facet by Concentration and Timepoint
  theme_bw() +
  xlab(&quot;Drugs&quot;) +
  ylab(&quot;Log Fold Change&quot;) +  # Updated y-axis label
  ggtitle(&quot;Log Fold Change for Late High Dose DOX-Specific Response Genes&quot;) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.line = element_line(linewidth = 1.5),
    strip.background = element_rect(fill = &quot;gray&quot;),  # Set strip background to gray
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),  # Unified styling for all facet labels
    axis.text.x = element_text(size = 8, color = &quot;black&quot;, angle = 15)
  )</code></pre>
<p><img src="figure/Corrmotif_all.Rmd/box18-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box18-1">
Past versions of box18-1.png
</button>
</p>
<div id="fig-box18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/2862c42bbe271925d87c00cd50b08289692b2517/docs/figure/Corrmotif_all.Rmd/box18-1.png" target="_blank">2862c42</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="corr-motif-set-mean-logfc-across-timepoints"
class="section level2">
<h2><strong>ðŸ“Œ Corr-motif set mean logFC across timepoints</strong></h2>
<pre class="r"><code># Combine all subset_toptables into a single data frame with additional metadata columns
all_toptables &lt;- bind_rows(
  Toptable_CX_0.1_3_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;3&quot;),
  Toptable_CX_0.1_24_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;24&quot;),
  Toptable_CX_0.1_48_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.1, Timepoint = &quot;48&quot;),
  Toptable_CX_0.5_3_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;3&quot;),
  Toptable_CX_0.5_24_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;24&quot;),
  Toptable_CX_0.5_48_df %&gt;% mutate(Drug = &quot;CX.5461&quot;, Concentration = 0.5, Timepoint = &quot;48&quot;),
  Toptable_DOX_0.1_3_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;3&quot;),
  Toptable_DOX_0.1_24_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;24&quot;),
  Toptable_DOX_0.1_48_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.1, Timepoint = &quot;48&quot;),
  Toptable_DOX_0.5_3_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;3&quot;),
  Toptable_DOX_0.5_24_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;24&quot;),
  Toptable_DOX_0.5_48_df %&gt;% mutate(Drug = &quot;DOX&quot;, Concentration = 0.5, Timepoint = &quot;48&quot;)
)

# Convert Entrez_ID in `all_toptables` to character for proper matching
all_toptables &lt;- all_toptables %&gt;% mutate(Entrez_ID = as.character(Entrez_ID))

# Define datasets for filtering (lists of Entrez IDs)
data_sets &lt;- list(
  &quot;Non Response&quot; = prob_all_1,
  &quot;CX_DOX Shared\nLate Response&quot; = prob_all_2,
  &quot;DOX-Specific Response&quot; = prob_all_3,
  &quot;Late High Dose\nDOX-Specific Response&quot; = prob_all_4
)

# Filter datasets and add labels
data_combined &lt;- bind_rows(lapply(names(data_sets), function(label) {
  all_toptables %&gt;%
    filter(Entrez_ID %in% data_sets[[label]]) %&gt;%  # Use list directly for filtering
    mutate(Dataset = label)
}))

# Ensure `logFC` column exists before computing mean
if (!&quot;logFC&quot; %in% colnames(data_combined)) {
  stop(&quot;Column `logFC` is missing in `data_combined`. Ensure that `all_toptables` has `logFC`.&quot;)
}

# Calculate the average logFC grouped by Dataset, Drug, Concentration, and Timepoint
data_summary &lt;- data_combined %&gt;%
  group_by(Dataset, Drug, Concentration, Timepoint) %&gt;%
  dplyr::summarize(avg_logFC = mean(logFC, na.rm = TRUE), .groups = &quot;drop&quot;) %&gt;%
  as.data.frame()

# Convert factors for plotting
data_summary &lt;- data_summary %&gt;%
  mutate(
    Timepoint = factor(Timepoint, levels = c(&quot;3&quot;, &quot;24&quot;, &quot;48&quot;), labels = c(&quot;3 hours&quot;, &quot;24 hours&quot;, &quot;48 hours&quot;)),
    Concentration = factor(Concentration, levels = c(0.1, 0.5)),
    Dataset = factor(Dataset, levels = c(&quot;Non Response&quot;, &quot;CX_DOX Shared\nLate Response&quot;, &quot;DOX-Specific Response&quot;, &quot;Late High Dose\nDOX-Specific Response&quot;))
  )

# Define custom drug palette
drug_palette &lt;- c(&quot;CX.5461&quot; = &quot;blue&quot;, &quot;DOX&quot; = &quot;red&quot;)

# Plot the data
ggplot(data_summary, aes(x = Timepoint, y = avg_logFC, group = Drug, color = Drug)) +
  geom_point(size = 3) +
  geom_line(size = 1.2) +
  scale_color_manual(values = drug_palette) +
  ylim(-2, 2) +  # Set the Y-axis scale
  facet_grid(Dataset ~ Concentration) +
  theme_bw() +
  labs(
    x = &quot;Timepoints&quot;,
    y = &quot;Mean Log Fold Change&quot;,
    title = &quot;Average Log Fold Change Across Motifs and Concentrations&quot;,
    color = &quot;Drug&quot;
  ) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text = element_text(size = 12, color = &quot;black&quot;),
    strip.text = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),
    legend.title = element_text(size = 14),
    legend.text = element_text(size = 12)
  )</code></pre>
<p><img src="figure/Corrmotif_all.Rmd/box19-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-box19-1">
Past versions of box19-1.png
</button>
</p>
<div id="fig-box19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/2862c42bbe271925d87c00cd50b08289692b2517/docs/figure/Corrmotif_all.Rmd/box19-1.png" target="_blank">2862c42</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="save-corr-motif-datasets-for-gene-ontology-analysis"
class="section level2">
<h2><strong>ðŸ“Œ Save Corr-motif datasets for Gene Ontology
analysis</strong></h2>
<pre class="r"><code>write.csv(data.frame(Entrez_ID = prob_all_1), &quot;data/prob_all_1.csv&quot;, row.names = FALSE)
write.csv(data.frame(Entrez_ID = prob_all_2), &quot;data/prob_all_2.csv&quot;, row.names = FALSE)
write.csv(data.frame(Entrez_ID = prob_all_3), &quot;data/prob_all_3.csv&quot;, row.names = FALSE)
write.csv(data.frame(Entrez_ID = prob_all_4), &quot;data/prob_all_4.csv&quot;, row.names = FALSE)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 26100)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] BiocParallel_1.36.0    Rfast_2.1.5.1          RcppParallel_5.1.10   
 [4] zigg_0.0.2             Rcpp_1.0.12            Cormotif_1.48.0       
 [7] limma_3.58.1           affy_1.80.0            corrplot_0.95         
[10] ggpubr_0.6.0           pheatmap_1.0.12        gprofiler2_0.2.3      
[13] RColorBrewer_1.1-3     org.Hs.eg.db_3.18.0    AnnotationDbi_1.64.1  
[16] IRanges_2.36.0         S4Vectors_0.40.2       Biobase_2.62.0        
[19] BiocGenerics_0.48.1    clusterProfiler_4.10.1 reshape2_1.4.4        
[22] lubridate_1.9.4        forcats_1.0.0          stringr_1.5.1         
[25] dplyr_1.1.4            purrr_1.0.4            readr_2.1.5           
[28] tidyr_1.3.1            tibble_3.2.1           ggplot2_3.5.2         
[31] tidyverse_2.0.0        workflowr_1.7.1       

loaded via a namespace (and not attached):
  [1] rstudioapi_0.17.1       jsonlite_2.0.0          magrittr_2.0.3         
  [4] farver_2.1.2            rmarkdown_2.29          fs_1.6.3               
  [7] zlibbioc_1.48.2         vctrs_0.6.5             memoise_2.0.1          
 [10] RCurl_1.98-1.17         ggtree_3.10.1           rstatix_0.7.2          
 [13] htmltools_0.5.8.1       broom_1.0.8             Formula_1.2-5          
 [16] gridGraphics_0.5-1      sass_0.4.10             bslib_0.9.0            
 [19] htmlwidgets_1.6.4       plyr_1.8.9              plotly_4.10.4          
 [22] cachem_1.1.0            whisker_0.4.1           igraph_2.1.4           
 [25] lifecycle_1.0.4         pkgconfig_2.0.3         Matrix_1.6-1.1         
 [28] R6_2.6.1                fastmap_1.2.0           gson_0.1.0             
 [31] GenomeInfoDbData_1.2.11 digest_0.6.34           aplot_0.2.5            
 [34] enrichplot_1.22.0       colorspace_2.1-0        patchwork_1.3.0        
 [37] ps_1.8.1                rprojroot_2.0.4         RSQLite_2.3.9          
 [40] labeling_0.4.3          timechange_0.3.0        abind_1.4-8            
 [43] httr_1.4.7              polyclip_1.10-7         compiler_4.3.0         
 [46] bit64_4.6.0-1           withr_3.0.2             backports_1.5.0        
 [49] carData_3.0-5           viridis_0.6.5           DBI_1.2.3              
 [52] ggforce_0.4.2           ggsignif_0.6.4          MASS_7.3-60            
 [55] HDO.db_0.99.1           tools_4.3.0             ape_5.8-1              
 [58] scatterpie_0.2.4        httpuv_1.6.15           glue_1.7.0             
 [61] callr_3.7.6             nlme_3.1-168            GOSemSim_2.28.1        
 [64] promises_1.3.2          grid_4.3.0              shadowtext_0.1.4       
 [67] getPass_0.2-4           fgsea_1.28.0            generics_0.1.3         
 [70] gtable_0.3.6            tzdb_0.5.0              preprocessCore_1.64.0  
 [73] data.table_1.17.0       hms_1.1.3               car_3.1-3              
 [76] tidygraph_1.3.1         XVector_0.42.0          ggrepel_0.9.6          
 [79] pillar_1.10.2           yulab.utils_0.2.0       later_1.3.2            
 [82] splines_4.3.0           tweenr_2.0.3            treeio_1.26.0          
 [85] lattice_0.22-7          bit_4.6.0               tidyselect_1.2.1       
 [88] GO.db_3.18.0            Biostrings_2.70.3       knitr_1.50             
 [91] git2r_0.36.2            gridExtra_2.3           xfun_0.52              
 [94] graphlayouts_1.2.2      statmod_1.5.0           stringi_1.8.3          
 [97] lazyeval_0.2.2          ggfun_0.1.8             yaml_2.3.10            
[100] evaluate_1.0.3          codetools_0.2-20        ggraph_2.2.1           
[103] qvalue_2.34.0           BiocManager_1.30.25     affyio_1.72.0          
[106] ggplotify_0.1.2         cli_3.6.1               munsell_0.5.1          
[109] processx_3.8.6          jquerylib_0.1.4         GenomeInfoDb_1.38.8    
[112] png_0.1-8               parallel_4.3.0          blob_1.2.4             
[115] DOSE_3.28.2             bitops_1.0-9            viridisLite_0.4.2      
[118] tidytree_0.4.6          scales_1.3.0            crayon_1.5.3           
[121] rlang_1.1.3             cowplot_1.1.3           fastmatch_1.1-6        
[124] KEGGREST_1.42.0        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
