<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Your Name" />

<meta name="date" content="2025-01-30" />

<title>Total Reads and Mapped Reads</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Paul_CX_2025</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Total Reads and Mapped Reads</h1>
<h4 class="author">Your Name</h4>
<h4 class="date">2025-01-30</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-01-30
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Paul_CX_2025/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it‚Äôs best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250129code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250129)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250129code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250129)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttree650faeb4af3ae016bda9a96abf2c6668a433b6c7targetblank650faeba">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/tree/650faeb4af3ae016bda9a96abf2c6668a433b6c7" target="_blank">650faeb</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsayanpaul01CX5461Projecttree650faeb4af3ae016bda9a96abf2c6668a433b6c7targetblank650faeba"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sayanpaul01/Paul_CX_2025/tree/650faeb4af3ae016bda9a96abf2c6668a433b6c7" target="_blank">650faeb</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

</code></pre>
<p>
Note that any generated files, e.g.¬†HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/total_reads_and_mapped_reads.Rmd</code>) and HTML
(<code>docs/total_reads_and_mapped_reads.html</code>) files. If you‚Äôve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/650faeb4af3ae016bda9a96abf2c6668a433b6c7/analysis/total_reads_and_mapped_reads.Rmd" target="_blank">650faeb</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-01-30
</td>
<td>
Updated total reads and mapped reads with code visibility and plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/21029509cdbf503797e9a0c715cb7e9ffc3598a1/docs/total_reads_and_mapped_reads.html" target="_blank">2102950</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-01-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/aab567b908229dcc8ff07faa68f2451e58d6be1b/analysis/total_reads_and_mapped_reads.Rmd" target="_blank">aab567b</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-01-30
</td>
<td>
Updated total reads and mapped reads with code visibility and plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/2830835ac00b1449a957bc043d954b9b52a7f74c/docs/total_reads_and_mapped_reads.html" target="_blank">2830835</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-01-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/7924924cc1ee331f04641bb871f734c72613c731/docs/total_reads_and_mapped_reads.html" target="_blank">7924924</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-01-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/03eba2031e39f8d57d01bdfa9e572fcb2152a818/analysis/total_reads_and_mapped_reads.Rmd" target="_blank">03eba20</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-01-30
</td>
<td>
Added Total Reads and Mapped Reads visualization
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/6b6d0573967e87daba5f3de77ba78f2b3e3df872/docs/total_reads_and_mapped_reads.html" target="_blank">6b6d057</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-01-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/ea621dc9e01c6027445d706fcb9485ce1f88069b/analysis/total_reads_and_mapped_reads.Rmd" target="_blank">ea621dc</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-01-30
</td>
<td>
Added Total Reads and Mapped Reads visualization
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sayanpaul01/Paul_CX_2025/d90515359bd42be2831bc03dc82c104d5a1173b1/docs/total_reads_and_mapped_reads.html" target="_blank">d905153</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-01-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/e3b04c6f40746397a087ec88971955374dd8d51c/analysis/total_reads_and_mapped_reads.Rmd" target="_blank">e3b04c6</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-01-30
</td>
<td>
Added Total Reads and Mapped Reads visualization
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="total-reads-by-sample" class="section level2">
<h2><strong>üìå Total Reads by Sample</strong></h2>
<p>This section visualizes the total RNA-sequencing reads across
samples.</p>
<hr />
<div id="load-required-libraries" class="section level3">
<h3><strong>Load Required Libraries</strong></h3>
<pre class="r"><code># Load necessary R packages
library(limma)</code></pre>
<pre><code>Warning: package &#39;limma&#39; was built under R version 4.3.1</code></pre>
<pre class="r"><code>library(RColorBrewer)
library(data.table)</code></pre>
<pre><code>Warning: package &#39;data.table&#39; was built under R version 4.3.2</code></pre>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>Warning: package &#39;tidyverse&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;ggplot2&#39; was built under R version 4.3.3</code></pre>
<pre><code>Warning: package &#39;tidyr&#39; was built under R version 4.3.3</code></pre>
<pre><code>Warning: package &#39;readr&#39; was built under R version 4.3.3</code></pre>
<pre><code>Warning: package &#39;purrr&#39; was built under R version 4.3.1</code></pre>
<pre><code>Warning: package &#39;dplyr&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;stringr&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;lubridate&#39; was built under R version 4.3.1</code></pre>
<pre><code>‚îÄ‚îÄ Attaching core tidyverse packages ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ tidyverse 2.0.0 ‚îÄ‚îÄ
‚úî dplyr     1.1.4     ‚úî readr     2.1.5
‚úî forcats   1.0.0     ‚úî stringr   1.5.1
‚úî ggplot2   3.5.1     ‚úî tibble    3.2.1
‚úî lubridate 1.9.3     ‚úî tidyr     1.3.1
‚úî purrr     1.0.2     
‚îÄ‚îÄ Conflicts ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ‚îÄ tidyverse_conflicts() ‚îÄ‚îÄ
‚úñ dplyr::between()     masks data.table::between()
‚úñ dplyr::filter()      masks stats::filter()
‚úñ dplyr::first()       masks data.table::first()
‚úñ lubridate::hour()    masks data.table::hour()
‚úñ lubridate::isoweek() masks data.table::isoweek()
‚úñ dplyr::lag()         masks stats::lag()
‚úñ dplyr::last()        masks data.table::last()
‚úñ lubridate::mday()    masks data.table::mday()
‚úñ lubridate::minute()  masks data.table::minute()
‚úñ lubridate::month()   masks data.table::month()
‚úñ lubridate::quarter() masks data.table::quarter()
‚úñ lubridate::second()  masks data.table::second()
‚úñ purrr::transpose()   masks data.table::transpose()
‚úñ lubridate::wday()    masks data.table::wday()
‚úñ lubridate::week()    masks data.table::week()
‚úñ lubridate::yday()    masks data.table::yday()
‚úñ lubridate::year()    masks data.table::year()
‚Ñπ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<pre class="r"><code>library(scales)</code></pre>
<pre><code>Warning: package &#39;scales&#39; was built under R version 4.3.2</code></pre>
<pre><code>
Attaching package: &#39;scales&#39;

The following object is masked from &#39;package:purrr&#39;:

    discard

The following object is masked from &#39;package:readr&#39;:

    col_factor</code></pre>
<pre class="r"><code>library(ggplot2)
library(dplyr)</code></pre>
</div>
</div>
<div id="load-data" class="section level2">
<h2><strong>üìç 2. Load Data</strong></h2>
<pre class="r"><code># Load the dataset containing the total reads per sample
align &lt;- read.csv(&quot;data/Total_number_of_reads_by_sample.csv&quot;)  # Ensure the file is in the &#39;data/&#39; folder
map &lt;- data.frame(align)</code></pre>
</div>
<div id="define-color-palettes" class="section level2">
<h2><strong>üìç 3. Define Color Palettes</strong></h2>
<pre class="r"><code># Define color palettes for plots
drug_palc &lt;- c(&quot;#8B006D&quot;,&quot;#DF707E&quot;,&quot;#F1B72B&quot;, &quot;#3386DD&quot;,&quot;#707031&quot;,&quot;#41B333&quot;)
Ind_palc &lt;- c(&quot;#ffbe0b&quot;,&quot;#ff006e&quot;,&quot;#fb5607&quot;, &quot;#8338ec&quot;,&quot;#3a86ff&quot;,&quot;#4a4e69&quot;)
Time_palc &lt;- c(&quot;#0000FF&quot;,&quot;#80FF00&quot;, &quot;#FF00FF&quot;)
Combined_palc &lt;- c(&quot;#FF0000&quot;,&quot;#00FF00&quot;,&quot;#0000FF&quot;,&quot;#FFFF00&quot;,&quot;#FF00FF&quot;,&quot;#00FFFF&quot;,
                   &quot;#FFA500&quot;,&quot;#800080&quot;,&quot;#FFC0CB&quot;,&quot;#A52A2A&quot;,&quot;#808080&quot;,&quot;#FFD700&quot;,
                   &quot;#008080&quot;,&quot;#000080&quot;,&quot;#FFFFFF&quot;,&quot;#000000&quot;,&quot;#D2691E&quot;,&quot;#ADFF2F&quot;)</code></pre>
</div>
<div id="prepare-data" class="section level2">
<h2><strong>üìç 4. Prepare Data</strong></h2>
<pre class="r"><code># Factor Sample_name to maintain order
map$Sample_name &lt;- factor(map$Sample_name, levels = map$Sample_name)</code></pre>
</div>
<div id="plot-total-reads-by-sample" class="section level2">
<h2><strong>üìç 5. Plot Total Reads by Sample</strong></h2>
<pre class="r"><code># Generate the bar plot
p &lt;- ggplot(map, aes(x = Sample_name, y = Counts, fill = Condition)) +
  geom_col() +
  scale_fill_manual(values = drug_palc) +
  scale_y_continuous(labels = function(x) paste0(x / 1e6, &quot;M&quot;)) +
  ggtitle(expression(&quot;Total number of reads by sample&quot;)) +
  xlab(&quot;&quot;) +
  ylab(expression(&quot;RNA-sequencing reads&quot;)) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(2), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;, angle = 0, hjust = 0.8, vjust = 0.5),
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 90, hjust = 1, vjust = 0.2)
  )

# Save the plot as an image
ggsave(&quot;output/total_reads_by_sample_plot.png&quot;, p)

# Display the plot in the document
p</code></pre>
<p><img src="figure/total_reads_and_mapped_reads.Rmd/plot-total-reads-1.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-total-reads-1">
Past versions of plot-total-reads-1.png
</button>
</p>
<div id="fig-plot-total-reads-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/d90515359bd42be2831bc03dc82c104d5a1173b1/docs/figure/total_reads_and_mapped_reads.Rmd/plot-total-reads-1.png" target="_blank">d905153</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="total-reads-by-treatment" class="section level2">
<h2><strong>üìå Total Reads by Treatment</strong></h2>
</div>
<div id="define-color-palettes-1" class="section level2">
<h2><strong>üìç Define Color Palettes</strong></h2>
<pre class="r"><code># Define color palettes for plots
drug_palc &lt;- c(&quot;#8B006D&quot;,&quot;#DF707E&quot;,&quot;#F1B72B&quot;, &quot;#3386DD&quot;,&quot;#707031&quot;,&quot;#41B333&quot;)
Ind_palc &lt;- c(&quot;#ffbe0b&quot;,&quot;#ff006e&quot;,&quot;#fb5607&quot;, &quot;#8338ec&quot;,&quot;#3a86ff&quot;,&quot;#4a4e69&quot;)
Time_palc &lt;- c(&quot;#0000FF&quot;,&quot;#80FF00&quot;, &quot;#FF00FF&quot;)
Combined_palc &lt;- c(&quot;#FF0000&quot;,&quot;#00FF00&quot;,&quot;#0000FF&quot;,&quot;#FFFF00&quot;,&quot;#FF00FF&quot;,&quot;#00FFFF&quot;,
                   &quot;#FFA500&quot;,&quot;#800080&quot;,&quot;#FFC0CB&quot;,&quot;#A52A2A&quot;,&quot;#808080&quot;,&quot;#FFD700&quot;,
                   &quot;#008080&quot;,&quot;#000080&quot;,&quot;#FFFFFF&quot;,&quot;#000000&quot;,&quot;#D2691E&quot;,&quot;#ADFF2F&quot;)</code></pre>
</div>
<div id="load-dataset-for-total-reads-by-treatment"
class="section level2">
<h2><strong>üìç Load dataset for total reads by treatment</strong></h2>
<pre class="r"><code>align1 &lt;- read.csv(&quot;data/Total_number_of_reads_by_treatment.csv&quot;)  # Ensure this file is inside the &#39;data/&#39; folder
map1 &lt;- data.frame(align1)</code></pre>
</div>
<div id="generate-the-boxplot" class="section level2">
<h2><strong>üìç Generate the boxplot</strong></h2>
<pre class="r"><code>p_treatment &lt;- ggplot(map1, aes(x = Condition, y = Counts, fill = Condition)) +
  geom_boxplot() +
  scale_fill_manual(values = drug_palc) +
  scale_y_continuous(
    limits = c(0, 40000000),  # Set y-axis range
    labels = function(x) paste0(x / 1e6, &quot;M&quot;)  # Display labels in millions
  ) +
  ggtitle(expression(&quot;Total number of reads by treatment&quot;)) +
  xlab(&quot;&quot;) +
  ylab(expression(&quot;RNA-sequencing reads&quot;)) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(2), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, angle = 90, hjust = 1, vjust = 0.2)
  )

# Save the plot as an image
ggsave(&quot;output/total_reads_by_treatment_plot.png&quot;, p_treatment)

# Display the plot in the document
p_treatment</code></pre>
<p><img src="figure/total_reads_and_mapped_reads.Rmd/generate_boxplot%20treatment-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-generate_boxplot-treatment-1">
Past versions of "generate_boxplot treatment-1.png"
</button>
</p>
<div id="fig-no-spaces-generate_boxplot-treatment-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/d90515359bd42be2831bc03dc82c104d5a1173b1/docs/figure/total_reads_and_mapped_reads.Rmd/generate_boxplot treatment-1.png" target="_blank">d905153</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="total-reads-by-individuals" class="section level2">
<h2><strong>üìå Total Reads by Individuals</strong></h2>
</div>
<div id="define-color-palettes-2" class="section level2">
<h2><strong>üìç Define Color Palettes</strong></h2>
<pre class="r"><code># Define color palette for individuals
Ind_palc &lt;- c(&quot;#ffbe0b&quot;,&quot;#ff006e&quot;,&quot;#fb5607&quot;, &quot;#8338ec&quot;,&quot;#3a86ff&quot;,&quot;#4a4e69&quot;)</code></pre>
</div>
<div id="load-dataset-for-total-reads-by-individual"
class="section level2">
<h2><strong>üìç Load dataset for total reads by individual</strong></h2>
<pre class="r"><code># Load dataset for total reads by individual
align2 &lt;- read.csv(&quot;data/Total_number_of_reads_by_Individuals.csv&quot;)  # Ensure this file is inside the &#39;data/&#39; folder
map2 &lt;- data.frame(align2)</code></pre>
</div>
<div id="generate-the-boxplot-1" class="section level2">
<h2><strong>üìç Generate the boxplot</strong></h2>
<pre class="r"><code># Generate the boxplot
p_individual &lt;- ggplot(map2, aes(x = Ind, y = Counts, fill = Individual)) +
  geom_boxplot() +
  scale_fill_manual(values = Ind_palc) +
  scale_y_continuous(
    limits = c(0, 40000000),  # Set y-axis range
    labels = function(x) paste0(x / 1e6, &quot;M&quot;)  # Display labels in millions
  ) +
  ggtitle(expression(&quot;Total number of reads by individual&quot;)) +
  xlab(&quot;&quot;) +
  ylab(expression(&quot;RNA-sequencing reads&quot;)) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(2), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, angle = 90, hjust = 1, vjust = 0.2)
  )

# Save the plot as an image
ggsave(&quot;output/total_reads_by_individual_plot.png&quot;, p_individual)

# Display the plot in the document
p_individual</code></pre>
<p><img src="figure/total_reads_and_mapped_reads.Rmd/generate_boxplot%20indv-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-generate_boxplot-indv-1">
Past versions of "generate_boxplot indv-1.png"
</button>
</p>
<div id="fig-no-spaces-generate_boxplot-indv-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/d90515359bd42be2831bc03dc82c104d5a1173b1/docs/figure/total_reads_and_mapped_reads.Rmd/generate_boxplot indv-1.png" target="_blank">d905153</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="total-reads-by-time" class="section level2">
<h2><strong>üìå Total Reads by Time</strong></h2>
</div>
<div id="define-color-palettes-3" class="section level2">
<h2><strong>üìç Define Color Palettes</strong></h2>
<pre class="r"><code># Define color palette for time points
Time_palc &lt;- c(&quot;#0000FF&quot;,&quot;#80FF00&quot;, &quot;#FF00FF&quot;)</code></pre>
</div>
<div id="load-dataset-for-total-reads-by-time" class="section level2">
<h2><strong>üìç Load dataset for total reads by time</strong></h2>
<pre class="r"><code># Load dataset for total reads by time
align3 &lt;- read.csv(&quot;data/Total_number_of_reads_by_time.csv&quot;)  # Ensure this file is inside the &#39;data/&#39; folder
map3 &lt;- data.frame(align3)</code></pre>
</div>
<div id="generate-the-boxplot-2" class="section level2">
<h2><strong>üìç Generate the boxplot</strong></h2>
<pre class="r"><code># Generate the boxplot
p_time &lt;- ggplot(map3, aes(x = Condition, y = Counts, fill = Time)) +
  geom_boxplot() +
  scale_fill_manual(values = Time_palc) +
  scale_y_continuous(
    limits = c(0, 40000000),  # Set y-axis range
    labels = function(x) paste0(x / 1e6, &quot;M&quot;)  # Display labels in millions
  ) +
  ggtitle(expression(&quot;Total number of reads by time&quot;)) +
  xlab(&quot;&quot;) +
  ylab(expression(&quot;RNA-sequencing reads&quot;)) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(2), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, angle = 90, hjust = 1, vjust = 0.2)
  )

# Save the plot as an image
ggsave(&quot;output/total_reads_by_time_plot.png&quot;, p_time)

# Display the plot in the document
p_time</code></pre>
<p><img src="figure/total_reads_and_mapped_reads.Rmd/generate_boxplot%20time-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-generate_boxplot-time-1">
Past versions of "generate_boxplot time-1.png"
</button>
</p>
<div id="fig-no-spaces-generate_boxplot-time-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/d90515359bd42be2831bc03dc82c104d5a1173b1/docs/figure/total_reads_and_mapped_reads.Rmd/generate_boxplot time-1.png" target="_blank">d905153</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="mapped-reads-by-sample" class="section level2">
<h2><strong>üìå Mapped Reads by Sample</strong></h2>
</div>
<div id="load-data-1" class="section level2">
<h2><strong>üìç 2. Load Data</strong></h2>
<pre class="r"><code># Load dataset for total mapped reads by sample
align_mapped_sample &lt;- read.csv(&quot;data/Total_number_of_mapped_reads_by_sample.csv&quot;)  
map_mapped_sample &lt;- data.frame(align_mapped_sample)

# Factor Sample Name
map_mapped_sample$Sample_name &lt;- factor(map_mapped_sample$Sample_name, levels = map_mapped_sample$Sample_name)</code></pre>
</div>
<div id="define-color-palettes-4" class="section level2">
<h2><strong>üìç 3. Define Color Palettes</strong></h2>
<pre class="r"><code># Define color palettes for plots
drug_palc &lt;- c(&quot;#8B006D&quot;,&quot;#DF707E&quot;,&quot;#F1B72B&quot;, &quot;#3386DD&quot;,&quot;#707031&quot;,&quot;#41B333&quot;)
Ind_palc &lt;- c(&quot;#ffbe0b&quot;,&quot;#ff006e&quot;,&quot;#fb5607&quot;, &quot;#8338ec&quot;,&quot;#3a86ff&quot;,&quot;#4a4e69&quot;)
Time_palc &lt;- c(&quot;#0000FF&quot;,&quot;#80FF00&quot;, &quot;#FF00FF&quot;)
Map_palc &lt;- c(&quot;#9b19f5&quot;,&quot;#e6d800&quot;, &quot;#b3d4ff&quot;)
Combined_palc &lt;- c(&quot;#FF0000&quot;,&quot;#00FF00&quot;,&quot;#0000FF&quot;,&quot;#FFFF00&quot;,&quot;#FF00FF&quot;,&quot;#00FFFF&quot;,
                   &quot;#FFA500&quot;,&quot;#800080&quot;,&quot;#FFC0CB&quot;,&quot;#A52A2A&quot;,&quot;#808080&quot;,&quot;#FFD700&quot;,
                   &quot;#008080&quot;,&quot;#000080&quot;,&quot;#FFFFFF&quot;,&quot;#000000&quot;,&quot;#D2691E&quot;,&quot;#ADFF2F&quot;)</code></pre>
</div>
<div id="plot-mapped-reads-by-sample" class="section level2">
<h2><strong>üìç 5. Plot Mapped Reads by Sample</strong></h2>
<pre class="r"><code># Generate the bar plot
p_mapped &lt;- ggplot(map_mapped_sample, aes(x = Sample_name, y = Counts, fill = Condition)) +
  geom_col() +
  scale_fill_manual(values = drug_palc) +
  scale_y_continuous(labels = function(x) paste0(x / 1e6, &quot;M&quot;)) +
  ggtitle(expression(&quot;Total number of mapped reads by sample&quot;)) +
  xlab(&quot;&quot;) +
  ylab(expression(&quot;RNA-sequencing reads&quot;)) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(2), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(size = 10, color = &quot;black&quot;, angle = 0, hjust = 0.8, vjust = 0.5),
    axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 90, hjust = 1, vjust = 0.2)
  )

# Save the plot as an image
ggsave(&quot;output/total_mapped_reads_by_sample_plot.png&quot;, p_mapped)

# Display the plot in the document
p_mapped</code></pre>
<p><img src="figure/total_reads_and_mapped_reads.Rmd/plot-mapped-reads-1.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-mapped-reads-1">
Past versions of plot-mapped-reads-1.png
</button>
</p>
<div id="fig-plot-mapped-reads-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/7924924cc1ee331f04641bb871f734c72613c731/docs/figure/total_reads_and_mapped_reads.Rmd/plot-mapped-reads-1.png" target="_blank">7924924</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-01-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/6b6d0573967e87daba5f3de77ba78f2b3e3df872/docs/figure/total_reads_and_mapped_reads.Rmd/plot-mapped-reads-1.png" target="_blank">6b6d057</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="mapped-reads-by-treatment" class="section level2">
<h2><strong>üìå Mapped Reads by Treatment</strong></h2>
</div>
<div id="define-color-palettes-5" class="section level2">
<h2><strong>üìç Define Color Palettes</strong></h2>
<pre class="r"><code># Define color palettes for plots
drug_palc &lt;- c(&quot;#8B006D&quot;,&quot;#DF707E&quot;,&quot;#F1B72B&quot;, &quot;#3386DD&quot;,&quot;#707031&quot;,&quot;#41B333&quot;)
Ind_palc &lt;- c(&quot;#ffbe0b&quot;,&quot;#ff006e&quot;,&quot;#fb5607&quot;, &quot;#8338ec&quot;,&quot;#3a86ff&quot;,&quot;#4a4e69&quot;)
Time_palc &lt;- c(&quot;#0000FF&quot;,&quot;#80FF00&quot;, &quot;#FF00FF&quot;)
Combined_palc &lt;- c(&quot;#FF0000&quot;,&quot;#00FF00&quot;,&quot;#0000FF&quot;,&quot;#FFFF00&quot;,&quot;#FF00FF&quot;,&quot;#00FFFF&quot;,
                   &quot;#FFA500&quot;,&quot;#800080&quot;,&quot;#FFC0CB&quot;,&quot;#A52A2A&quot;,&quot;#808080&quot;,&quot;#FFD700&quot;,
                   &quot;#008080&quot;,&quot;#000080&quot;,&quot;#FFFFFF&quot;,&quot;#000000&quot;,&quot;#D2691E&quot;,&quot;#ADFF2F&quot;)</code></pre>
</div>
<div id="load-dataset-for-mapped-reads-by-treatment"
class="section level2">
<h2><strong>üìç Load dataset for Mapped reads by treatment</strong></h2>
<pre class="r"><code>align4 &lt;- read.csv(&quot;data/Total_number_of_mapped_reads_by_treatment.csv&quot;)  # Ensure this file is inside the &#39;data/&#39; folder
map4 &lt;- data.frame(align4)</code></pre>
</div>
<div id="generate-the-boxplot-3" class="section level2">
<h2><strong>üìç Generate the boxplot</strong></h2>
<pre class="r"><code>p__mapped_treatment &lt;- ggplot(map4, aes(x = Condition, y = Counts, fill = Condition)) +
  geom_boxplot() +
  scale_fill_manual(values = drug_palc) +
  scale_y_continuous(
    limits = c(0, 40000000),  # Set y-axis range
    labels = function(x) paste0(x / 1e6, &quot;M&quot;)  # Display labels in millions
  ) +
  ggtitle(expression(&quot;Total number of mapped reads by treatment&quot;)) +
  xlab(&quot;&quot;) +
  ylab(expression(&quot;RNA-sequencing reads&quot;)) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(2), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, angle = 90, hjust = 1, vjust = 0.2)
  )
# Save the plot as an image
ggsave(&quot;output/Mapped_reads_by_treatment_plot.png&quot;, p__mapped_treatment)

# Display the plot in the document
p__mapped_treatment</code></pre>
<p><img src="figure/total_reads_and_mapped_reads.Rmd/generate_boxplot%20mapped%20treatment-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-generate_boxplot-mapped-treatment-1">
Past versions of "generate_boxplot mapped treatment-1.png"
</button>
</p>
<div id="fig-no-spaces-generate_boxplot-mapped-treatment-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/6b6d0573967e87daba5f3de77ba78f2b3e3df872/docs/figure/total_reads_and_mapped_reads.Rmd/generate_boxplot mapped treatment-1.png" target="_blank">6b6d057</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="mapped-reads-by-individuals" class="section level2">
<h2><strong>üìå Mapped Reads by Individuals</strong></h2>
</div>
<div id="define-color-palettes-6" class="section level2">
<h2><strong>üìç Define Color Palettes</strong></h2>
<pre class="r"><code># Define color palette for individuals
Ind_palc &lt;- c(&quot;#ffbe0b&quot;,&quot;#ff006e&quot;,&quot;#fb5607&quot;, &quot;#8338ec&quot;,&quot;#3a86ff&quot;,&quot;#4a4e69&quot;)</code></pre>
</div>
<div id="load-dataset-for-mapped-reads-by-individual"
class="section level2">
<h2><strong>üìç Load dataset for Mapped reads by individual</strong></h2>
<pre class="r"><code># Load dataset for Mapped reads by individual
align5 &lt;- read.csv(&quot;data/Total_number_of_Mapped_reads_by_Individuals.csv&quot;)  # Ensure this file is inside the &#39;data/&#39; folder
map5 &lt;- data.frame(align5)</code></pre>
</div>
<div id="generate-the-boxplot-4" class="section level2">
<h2><strong>üìç Generate the boxplot</strong></h2>
<pre class="r"><code># Generate the boxplot
p_mapped_individual &lt;- ggplot(map5, aes(x = Ind, y = Counts, fill = Individual)) +
  geom_boxplot() +
  scale_fill_manual(values = Ind_palc) +
  scale_y_continuous(
    limits = c(0, 40000000),  # Set y-axis range
    labels = function(x) paste0(x / 1e6, &quot;M&quot;)  # Display labels in millions
  ) +
  ggtitle(expression(&quot;Total number of mapped number of reads by individual&quot;)) +
  xlab(&quot;&quot;) +
  ylab(expression(&quot;RNA-sequencing reads&quot;)) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(2), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, angle = 90, hjust = 1, vjust = 0.2)
  )

# Save the plot as an image
ggsave(&quot;output/Mapped_reads_by_individual_plot.png&quot;, p_mapped_individual)

# Display the plot in the document
p_mapped_individual</code></pre>
<p><img src="figure/total_reads_and_mapped_reads.Rmd/generate_boxplot%20mapped%20indv-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-generate_boxplot-mapped-indv-1">
Past versions of "generate_boxplot mapped indv-1.png"
</button>
</p>
<div id="fig-no-spaces-generate_boxplot-mapped-indv-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/6b6d0573967e87daba5f3de77ba78f2b3e3df872/docs/figure/total_reads_and_mapped_reads.Rmd/generate_boxplot mapped indv-1.png" target="_blank">6b6d057</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="mapped-reads-by-time" class="section level2">
<h2><strong>üìå Mapped Reads by Time</strong></h2>
</div>
<div id="define-color-palettes-7" class="section level2">
<h2><strong>üìç Define Color Palettes</strong></h2>
<pre class="r"><code># Define color palette for time points
Time_palc &lt;- c(&quot;#0000FF&quot;,&quot;#80FF00&quot;, &quot;#FF00FF&quot;)</code></pre>
</div>
<div id="load-dataset-for-mapped-reads-by-time" class="section level2">
<h2><strong>üìç Load dataset for Mapped reads by time</strong></h2>
<pre class="r"><code># Load dataset for Mapped reads by time
align6 &lt;- read.csv(&quot;data/Total_number_of_mapped_reads_by_time.csv&quot;)  # Ensure this file is inside the &#39;data/&#39; folder
map6 &lt;- data.frame(align6)</code></pre>
</div>
<div id="generate-the-boxplot-5" class="section level2">
<h2><strong>üìç Generate the boxplot</strong></h2>
<pre class="r"><code># Generate the boxplot
p_mapped_time &lt;- ggplot(map6, aes(x = Condition, y = Counts, fill = Time)) +
  geom_boxplot() +
  scale_fill_manual(values = Time_palc) +
  scale_y_continuous(
    limits = c(0, 40000000),  # Set y-axis range
    labels = function(x) paste0(x / 1e6, &quot;M&quot;)  # Display labels in millions
  ) +
  ggtitle(expression(&quot;Total number of mapped reads by time&quot;)) +
  xlab(&quot;&quot;) +
  ylab(expression(&quot;RNA-sequencing reads&quot;)) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(2), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text.x = element_text(size = 10, angle = 90, hjust = 1, vjust = 0.2)
  )

# Save the plot as an image
ggsave(&quot;output/total_reads_by_time_plot.png&quot;, p_mapped_time)

# Display the plot in the document
p_mapped_time</code></pre>
<p><img src="figure/total_reads_and_mapped_reads.Rmd/generate_boxplot%20mapped%20time-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-generate_boxplot-mapped-time-1">
Past versions of "generate_boxplot mapped time-1.png"
</button>
</p>
<div id="fig-no-spaces-generate_boxplot-mapped-time-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/7924924cc1ee331f04641bb871f734c72613c731/docs/figure/total_reads_and_mapped_reads.Rmd/generate_boxplot mapped time-1.png" target="_blank">7924924</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-01-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/6b6d0573967e87daba5f3de77ba78f2b3e3df872/docs/figure/total_reads_and_mapped_reads.Rmd/generate_boxplot mapped time-1.png" target="_blank">6b6d057</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="read-mapping-summary" class="section level3">
<h3><strong>üîπ Read Mapping Summary</strong></h3>
<pre class="r"><code># Load dataset for mapping summary
align_mapping &lt;- read.csv(&quot;data/Comparison.csv&quot;)  
map_mapping &lt;- data.frame(align_mapping)</code></pre>
</div>
<div id="read-mapping-across-samples" class="section level3">
<h3><strong>üìå Read Mapping Across Samples</strong></h3>
<pre class="r"><code>p_mapping_samples &lt;- ggplot(map_mapping, aes(x = Sample_name, y = Total, fill = Mapping)) +
  geom_bar(position=&quot;dodge&quot;, stat=&quot;identity&quot;)+
  #geom_hline(aes(yintercept=20000000))+
  scale_fill_manual(values=Map_palc)+
  scale_y_continuous(labels = label_number(suffix = &quot; M&quot;, scale = 1e-6))+
  ggtitle(expression(&quot;Read mapping summary across samples&quot;))+
  xlab(&quot;&quot;)+
  ylab(expression(&quot;RNA -sequencing reads&quot;))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = &quot;black&quot;),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = &quot;black&quot;, angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = &quot;black&quot;, angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = &quot;black&quot;, face = &quot;bold&quot;),
        strip.text.y = element_text(color = &quot;white&quot;))

p_mapping_samples</code></pre>
<p><img src="figure/total_reads_and_mapped_reads.Rmd/generate_boxplot%20Read%20mapping%20samples-1.png" width="2688" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-generate_boxplot-Read-mapping-samples-1">
Past versions of "generate_boxplot Read mapping samples-1.png"
</button>
</p>
<div id="fig-no-spaces-generate_boxplot-Read-mapping-samples-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/21029509cdbf503797e9a0c715cb7e9ffc3598a1/docs/figure/total_reads_and_mapped_reads.Rmd/generate_boxplot Read mapping samples-1.png" target="_blank">2102950</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-01-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/7924924cc1ee331f04641bb871f734c72613c731/docs/figure/total_reads_and_mapped_reads.Rmd/generate_boxplot Read mapping samples-1.png" target="_blank">7924924</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="read-mapping-across-treatments" class="section level3">
<h3><strong>üìå Read Mapping Across Treatments</strong></h3>
<pre class="r"><code>p_mapping_treatments &lt;- ggplot(map_mapping, aes(x = Mapping, y = Total, fill = Condition)) +
  geom_boxplot() +
  scale_fill_manual(values=drug_palc) +
  scale_y_continuous(labels = function(x) paste0(x / 1e6, &quot;M&quot;)) +
  ggtitle(&quot;Read Mapping Summary Across Treatments&quot;) +
  theme_bw()

p_mapping_treatments</code></pre>
<p><img src="figure/total_reads_and_mapped_reads.Rmd/generate_boxplot%20Read%20mapping%20treatment-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-generate_boxplot-Read-mapping-treatment-1">
Past versions of "generate_boxplot Read mapping treatment-1.png"
</button>
</p>
<div id="fig-no-spaces-generate_boxplot-Read-mapping-treatment-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/7924924cc1ee331f04641bb871f734c72613c731/docs/figure/total_reads_and_mapped_reads.Rmd/generate_boxplot Read mapping treatment-1.png" target="_blank">7924924</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="read-mapping-across-individuals" class="section level3">
<h3><strong>üìå Read Mapping Across Individuals</strong></h3>
<pre class="r"><code>p_mapping_individuals &lt;- ggplot(map_mapping, aes(x = Mapping, y = Total, fill = Individual)) +
  geom_boxplot() +
  scale_fill_manual(values=Ind_palc) +
  scale_y_continuous(labels = function(x) paste0(x / 1e6, &quot;M&quot;)) +
  ggtitle(&quot;Read Mapping Summary Across Individuals&quot;) +
  theme_bw()

p_mapping_individuals</code></pre>
<p><img src="figure/total_reads_and_mapped_reads.Rmd/generate_boxplot%20Read%20mapping%20Individuals-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-generate_boxplot-Read-mapping-Individuals-1">
Past versions of "generate_boxplot Read mapping Individuals-1.png"
</button>
</p>
<div id="fig-no-spaces-generate_boxplot-Read-mapping-Individuals-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/7924924cc1ee331f04641bb871f734c72613c731/docs/figure/total_reads_and_mapped_reads.Rmd/generate_boxplot Read mapping Individuals-1.png" target="_blank">7924924</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="read-mapping-across-timepoints" class="section level3">
<h3><strong>üìå Read Mapping Across Timepoints</strong></h3>
<pre class="r"><code>p_mapping_time &lt;- ggplot(map_mapping, aes(x = Mapping, y = Total, fill = Time)) +
  geom_boxplot() +
  scale_fill_manual(values=Ind_palc) +
  scale_y_continuous(labels = function(x) paste0(x / 1e6, &quot;M&quot;)) +
  ggtitle(&quot;Read Mapping Summary Across Timepoints&quot;) +
  theme_bw()

p_mapping_time</code></pre>
<p><img src="figure/total_reads_and_mapped_reads.Rmd/generate_boxplot%20Read%20mapping%20Timepoints-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-generate_boxplot-Read-mapping-Timepoints-1">
Past versions of "generate_boxplot Read mapping Timepoints-1.png"
</button>
</p>
<div id="fig-no-spaces-generate_boxplot-Read-mapping-Timepoints-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/7924924cc1ee331f04641bb871f734c72613c731/docs/figure/total_reads_and_mapped_reads.Rmd/generate_boxplot Read mapping Timepoints-1.png" target="_blank">7924924</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="read-mapping-across-treatment-concentration-and-timepoints"
class="section level3">
<h3><strong>üìå Read Mapping Across treatment, concentration, and
timepoints</strong></h3>
<pre class="r"><code>map_mapping %&gt;%
  ggplot(., aes (x = Mapping, y = Total, fill = Treat_Cond)) +
  geom_boxplot() +
  scale_fill_manual(values = Combined_palc) +
  scale_y_continuous(labels = label_number(suffix = &quot; M&quot;, scale = 1e-6)) +
  ggtitle(expression(&quot;Read mapping summary across treatment, Conc., timepoints&quot;)) +
  xlab(&quot;&quot;) +
  ylab(expression(&quot;RNA -sequencing reads&quot;)) +
  theme_bw() +
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = &quot;black&quot;),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text.y = element_text(size = 10, color = &quot;black&quot;, angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size = 10, color = &quot;black&quot;, angle = 90, hjust = 1, vjust = 0.2),
        strip.text.y = element_text(color = &quot;white&quot;))</code></pre>
<p><img src="figure/total_reads_and_mapped_reads.Rmd/plot-mapping-summary-treatment-cond-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-mapping-summary-treatment-cond-1">
Past versions of plot-mapping-summary-treatment-cond-1.png
</button>
</p>
<div id="fig-plot-mapping-summary-treatment-cond-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sayanpaul01/Paul_CX_2025/blob/7924924cc1ee331f04641bb871f734c72613c731/docs/figure/total_reads_and_mapped_reads.Rmd/plot-mapping-summary-treatment-cond-1.png" target="_blank">7924924</a>
</td>
<td>
sayanpaul01
</td>
<td>
2025-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 22631)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] ggpubr_0.6.0       Hmisc_5.2-0        corrplot_0.95      ggrepel_0.9.6     
 [5] cowplot_1.1.3      biomaRt_2.58.2     reshape2_1.4.4     gridExtra_2.3     
 [9] edgeR_4.0.1        scales_1.3.0       lubridate_1.9.3    forcats_1.0.0     
[13] stringr_1.5.1      dplyr_1.1.4        purrr_1.0.2        readr_2.1.5       
[17] tidyr_1.3.1        tibble_3.2.1       ggplot2_3.5.1      tidyverse_2.0.0   
[21] data.table_1.14.10 RColorBrewer_1.1-3 limma_3.58.1       workflowr_1.7.1   

loaded via a namespace (and not attached):
  [1] DBI_1.2.3               bitops_1.0-7            rlang_1.1.3            
  [4] magrittr_2.0.3          git2r_0.35.0            compiler_4.3.0         
  [7] RSQLite_2.3.3           getPass_0.2-4           png_0.1-8              
 [10] systemfonts_1.1.0       callr_3.7.6             vctrs_0.6.5            
 [13] pkgconfig_2.0.3         crayon_1.5.3            fastmap_1.1.1          
 [16] backports_1.5.0         dbplyr_2.5.0            XVector_0.42.0         
 [19] labeling_0.4.3          promises_1.3.0          rmarkdown_2.29         
 [22] tzdb_0.4.0              ps_1.8.1                ragg_1.3.3             
 [25] bit_4.0.5               xfun_0.50               zlibbioc_1.48.0        
 [28] cachem_1.0.8            GenomeInfoDb_1.38.8     jsonlite_1.8.9         
 [31] progress_1.2.3          blob_1.2.4              later_1.3.2            
 [34] broom_1.0.7             cluster_2.1.6           prettyunits_1.2.0      
 [37] R6_2.5.1                bslib_0.8.0             stringi_1.8.3          
 [40] car_3.1-3               rpart_4.1.23            jquerylib_0.1.4        
 [43] Rcpp_1.0.12             knitr_1.49              base64enc_0.1-3        
 [46] IRanges_2.36.0          nnet_7.3-19             httpuv_1.6.15          
 [49] timechange_0.3.0        tidyselect_1.2.1        abind_1.4-8            
 [52] rstudioapi_0.17.1       yaml_2.3.10             curl_6.0.1             
 [55] processx_3.8.5          lattice_0.22-5          plyr_1.8.9             
 [58] Biobase_2.62.0          withr_3.0.2             KEGGREST_1.42.0        
 [61] evaluate_1.0.3          foreign_0.8-87          BiocFileCache_2.10.2   
 [64] xml2_1.3.6              Biostrings_2.70.1       pillar_1.10.1          
 [67] filelock_1.0.3          carData_3.0-5           whisker_0.4.1          
 [70] checkmate_2.3.2         stats4_4.3.0            generics_0.1.3         
 [73] rprojroot_2.0.4         RCurl_1.98-1.13         S4Vectors_0.40.1       
 [76] hms_1.1.3               munsell_0.5.1           glue_1.7.0             
 [79] tools_4.3.0             ggsignif_0.6.4          locfit_1.5-9.8         
 [82] fs_1.6.3                XML_3.99-0.17           grid_4.3.0             
 [85] AnnotationDbi_1.64.1    colorspace_2.1-0        GenomeInfoDbData_1.2.11
 [88] htmlTable_2.4.3         Formula_1.2-5           cli_3.6.1              
 [91] rappdirs_0.3.3          textshaping_0.4.1       gtable_0.3.6           
 [94] rstatix_0.7.2           sass_0.4.9              digest_0.6.34          
 [97] BiocGenerics_0.48.1     htmlwidgets_1.6.4       farver_2.1.2           
[100] memoise_2.0.1           htmltools_0.5.8.1       lifecycle_1.0.4        
[103] httr_1.4.7              statmod_1.5.0           bit64_4.0.5            </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
